diff --git a/PKGBUILD b/PKGBUILD
index 5f5a3d0..e25dde0 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -83,6 +83,8 @@ build() {
     -DCALM_DEPRECATION=ON
     -DENABLE_WEBSOCKET=ON
     -Wno-dev
+    -DENABLE_QSV11=OFF
+    -DENABLE_BROWSER=OFF
   )
   cmake "${cmake_options[@]}"
   cmake --build build
@@ -91,3 +93,5 @@ build() {
 package() {
   DESTDIR="$pkgdir" cmake --install build
 }
+
+depends=($(printf "%s\n" "${depends[@]}" | grep -Ev '^(cef)$'))
