--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,15 +6,10 @@ pkgname=(
   'bootconfig'
   'bpf'
   'cpupower'
-  'hyperv'
-  'intel-speed-select'
-  'kcpuid'
   'linux-tools-meta'
   'perf'
   'tmon'
-  'turbostat'
   'usbip'
-  'x86_energy_perf_policy'
 )
 pkgver=7.0.3
 pkgrel=1
@@ -39,11 +34,11 @@ makedepends+=('ncurses')
 # bpf deps
 makedepends+=('readline' 'zlib' 'libelf' 'libcap' 'python-docutils')
 # turbostat deps
-makedepends+=('libcap')
+# makedepends+=('libcap')
 # bpftool
 makedepends+=('llvm' 'clang')
 # intel-speed-select
-makedepends+=('libnl')
+# makedepends+=('libnl')
 groups=("$pkgbase")
 source=(https://cdn.kernel.org/pub/linux/kernel/v${pkgver%%.*}.x/${_srcname}.tar.{xz,sign}
         'usbipd.service'
@@ -73,6 +68,7 @@ prepare() {
     echo ":: Applying patch $src"
     patch -p1 -N -i "$srcdir/$src"
   done
+  sed -i '/perf-util-$(CONFIG_LIBDW_DWARF_UNWIND) += unwind-libdw.o/d' tools/perf/arch/loongarch/util/Build
   # force our perf version
   cat > tools/perf/util/PERF-VERSION-GEN << EOF
 #!/bin/sh
@@ -105,10 +101,13 @@ build() {
   make VERSION=$pkgver-$pkgrel
   popd
 
+  # Use a "multi-line comment" to keep patch from rotting
+  : <<COMMENT_SEPARATOR
   echo ':: x86_energy_perf_policy'
   pushd "$_srcname"/tools/power/x86/x86_energy_perf_policy
   make
   popd
+COMMENT_SEPARATOR
 
   echo ':: usbip'
   pushd "$_srcname"/tools/usb/usbip
@@ -124,6 +123,7 @@ build() {
   make
   popd
 
+  : <<COMMENT_SEPARATOR
   echo ':: turbostat'
   pushd "$_srcname"/tools/power/x86/turbostat
   make
@@ -133,6 +133,7 @@ build() {
   pushd "$_srcname"/tools/hv
   CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hypervkvpd/\"' make
   popd
+COMMENT_SEPARATOR
 
   echo ':: bpf'
   pushd "$_srcname"/tools/bpf
@@ -149,6 +150,7 @@ build() {
   make
   popd
 
+  : <<COMMENT_SEPARATOR
   echo ':: intel-speed-select'
   pushd "$_srcname"/tools/power/x86/intel-speed-select
   make
@@ -158,6 +160,7 @@ build() {
   pushd "$_srcname"/tools/arch/x86/kcpuid
   make
   popd
+COMMENT_SEPARATOR
 }
 
 package_linux-tools-meta() {
@@ -167,14 +170,9 @@ package_linux-tools-meta() {
     'bootconfig'
     'bpf'
     'cpupower'
-    'hyperv'
-    'intel-speed-select'
-    'kcpuid'
     'perf'
     'tmon'
-    'turbostat'
     'usbip'
-    'x86_energy_perf_policy'
   )
   conflicts=(
     'acpidump'
@@ -192,7 +190,7 @@ package_perf() {
     prefix=/usr \
     lib=lib/perf \
     perfexecdir=lib/perf \
-    EXTRA_CFLAGS=' -Wno-error=bad-function-cast -Wno-error=declaration-after-statement -Wno-error=switch-enum -Wno-error=discarded-qualifiers' \
+    EXTRA_CFLAGS=' -Wno-error=bad-function-cast -Wno-error=declaration-after-statement -Wno-error=switch-enum -Wno-error=discarded-qualifiers -Wno-error=aggressive-loop-optimizations -Wno-error=unterminated-string-initialization' \
     NO_SDT=1 \
     BUILD_BPF_SKEL=1 \
     PYTHON=python \
