--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,13 +5,13 @@ pkgname=(
   'bootconfig'
   'bpf'
   'cpupower'
-  'hyperv'
+#  'hyperv'
   'linux-tools-meta'
   'perf'
   'tmon'
-  'turbostat'
+#  'turbostat'
   'usbip'
-  'x86_energy_perf_policy'
+# 'x86_energy_perf_policy'
 )
 pkgver=6.10
 pkgrel=2
@@ -100,11 +100,6 @@ build() {
   make VERSION=$pkgver-$pkgrel
   popd
 
-  echo ':: x86_energy_perf_policy'
-  pushd linux/tools/power/x86/x86_energy_perf_policy
-  make
-  popd
-
   echo ':: usbip'
   pushd linux/tools/usb/usbip
   # Fix gcc compilation
@@ -119,15 +114,10 @@ build() {
   make
   popd
 
-  echo ':: turbostat'
-  pushd linux/tools/power/x86/turbostat
-  make
-  popd
-
-  echo ':: hv'
-  pushd linux/tools/hv
-  CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hypervkvpd/\"' make
-  popd
+#  echo ':: hv'
+#  pushd linux/tools/hv
+#  CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hypervkvpd/\"' make
+#  popd
 
   echo ':: bpf'
   pushd linux/tools/bpf
@@ -151,12 +141,12 @@ package_linux-tools-meta() {
     'bootconfig'
     'bpf'
     'cpupower'
-    'hyperv'
+#    'hyperv'
     'perf'
     'tmon'
-    'turbostat'
+#    'turbostat'
     'usbip'
-    'x86_energy_perf_policy'
+#    'x86_energy_perf_policy'
   )
   conflicts=(
     'acpidump'
