--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,15 +6,10 @@ pkgname=(
   'bootconfig'
   'bpf'
   'cpupower'
-  'hyperv'
-  'intel-speed-select'
-  'kcpuid'
   'linux-tools-meta'
   'perf'
   'tmon'
-  'turbostat'
   'usbip'
-  'x86_energy_perf_policy'
 )
 pkgver=7.1.7
 pkgrel=1
@@ -73,6 +68,7 @@ prepare() {
     echo ":: Applying patch $src"
     patch -p1 -N -i "$srcdir/$src"
   done
+  sed -i '/perf-util-$(CONFIG_LIBDW_DWARF_UNWIND) += unwind-libdw.o/d' tools/perf/arch/loongarch/util/Build
   # force our perf version
   cat > tools/perf/util/PERF-VERSION-GEN << EOF
 #!/bin/sh
@@ -132,11 +128,12 @@ build() {
     make
     popd
   fi
-
+  : <<COMMENT_SEPARATOR
   echo ':: hv'
   pushd "$_srcname"/tools/hv
   CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hypervkvpd/\"' make
   popd
+COMMENT_SEPARATOR
 
   echo ':: bpf'
   pushd "$_srcname"/tools/bpf
@@ -175,7 +172,6 @@ package_linux-tools-meta() {
     'bootconfig'
     'bpf'
     'cpupower'
-    'hyperv'
     'perf'
     'tmon'
     'usbip'
@@ -202,7 +198,7 @@ package_perf() {
     prefix=/usr \
     lib=lib/perf \
     perfexecdir=lib/perf \
-    EXTRA_CFLAGS=' -Wno-error=bad-function-cast -Wno-error=declaration-after-statement -Wno-error=switch-enum -Wno-error=discarded-qualifiers' \
+    EXTRA_CFLAGS=' -Wno-error=bad-function-cast -Wno-error=declaration-after-statement -Wno-error=switch-enum -Wno-error=discarded-qualifiers -Wno-error=aggressive-loop-optimizations -Wno-error=unterminated-string-initialization' \
     NO_SDT=1 \
     BUILD_BPF_SKEL=1 \
     PYTHON=python \
