diff --git a/PKGBUILD b/PKGBUILD
index 497acd4..fff6e5b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,15 +5,10 @@ pkgname=(
   'bootconfig'
   'bpf'
   'cpupower'
-  'hyperv'
-  'intel-speed-select'
-  'kcpuid'
   'linux-tools-meta'
   'perf'
   'tmon'
-  'turbostat'
   'usbip'
-  'x86_energy_perf_policy'
 )
 pkgver=6.14
 pkgrel=1
@@ -105,10 +100,13 @@ build() {
   make VERSION=$pkgver-$pkgrel
   popd
 
+  # Use a "multi-line comment" to keep patch from rotting
+  : <<COMMENT_SEPARATOR
   echo ':: x86_energy_perf_policy'
   pushd linux/tools/power/x86/x86_energy_perf_policy
   make
   popd
+COMMENT_SEPARATOR
 
   echo ':: usbip'
   pushd linux/tools/usb/usbip
@@ -124,6 +122,7 @@ build() {
   make
   popd
 
+  : <<COMMENT_SEPARATOR
   echo ':: turbostat'
   pushd linux/tools/power/x86/turbostat
   make
@@ -133,6 +132,7 @@ build() {
   pushd linux/tools/hv
   CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hypervkvpd/\"' make
   popd
+COMMENT_SEPARATOR
 
   echo ':: bpf'
   pushd linux/tools/bpf
@@ -148,6 +148,7 @@ build() {
   make
   popd
 
+  : <<COMMENT_SEPARATOR
   echo ':: intel-speed-select'
   pushd linux/tools/power/x86/intel-speed-select
   make
@@ -157,6 +158,7 @@ build() {
   pushd linux/tools/arch/x86/kcpuid
   make
   popd
+COMMENT_SEPARATOR
 }
 
 package_linux-tools-meta() {
@@ -166,14 +168,9 @@ package_linux-tools-meta() {
     'bootconfig'
     'bpf'
     'cpupower'
-    'hyperv'
-    'intel-speed-select'
-    'kcpuid'
     'perf'
     'tmon'
-    'turbostat'
     'usbip'
-    'x86_energy_perf_policy'
   )
   conflicts=(
     'acpidump'
@@ -191,7 +188,7 @@ package_perf() {
     prefix=/usr \
     lib=lib/perf \
     perfexecdir=lib/perf \
-    EXTRA_CFLAGS=' -Wno-error=bad-function-cast -Wno-error=declaration-after-statement -Wno-error=switch-enum' \
+    EXTRA_CFLAGS=' -Wno-error=bad-function-cast -Wno-error=declaration-after-statement -Wno-error=switch-enum -Wno-error=aggressive-loop-optimizations -Wno-error=unterminated-string-initialization' \
     NO_SDT=1 \
     BUILD_BPF_SKEL=1 \
     PYTHON=python \
@@ -322,4 +319,7 @@ package_kcpuid() {
   make BINDIR=/usr/bin HWDATADIR="/usr/share/misc" DESTDIR="$pkgdir" -C linux/tools/arch/x86/kcpuid install
 }
 
+source+=("fix-perf-in-loong64.patch")
+sha256sums+=('7e5c86eb2c487ec799f789bd51f1fe1a7466a351a5b8db70235c43883cb82ab2')
+
 # vim:set ts=2 sw=2 et:
