diff --git a/PKGBUILD b/PKGBUILD
index 9455ade..108a794 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,15 +5,10 @@ pkgname=(
   'bootconfig'
   'bpf'
   'cpupower'
-  'hyperv'
-  'intel-speed-select'
-  'kcpuid'
   'linux-tools-meta'
   'perf'
   'tmon'
-  'turbostat'
   'usbip'
-  'x86_energy_perf_policy'
 )
 pkgver=6.12
 pkgrel=11
@@ -108,10 +103,13 @@ build() {
   make VERSION=$pkgver-$pkgrel
   popd
 
+  # Use a "multi-line comment" to keep patch from rotting
+  : <<COMMENT_SEPARATOR
   echo ':: x86_energy_perf_policy'
   pushd linux/tools/power/x86/x86_energy_perf_policy
   make
   popd
+COMMENT_SEPARATOR
 
   echo ':: usbip'
   pushd linux/tools/usb/usbip
@@ -127,6 +125,7 @@ build() {
   make
   popd
 
+  : <<COMMENT_SEPARATOR
   echo ':: turbostat'
   pushd linux/tools/power/x86/turbostat
   make
@@ -136,6 +135,7 @@ build() {
   pushd linux/tools/hv
   CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hypervkvpd/\"' make
   popd
+COMMENT_SEPARATOR
 
   echo ':: bpf'
   pushd linux/tools/bpf
@@ -151,6 +151,7 @@ build() {
   make
   popd
 
+  : <<COMMENT_SEPARATOR
   echo ':: intel-speed-select'
   pushd linux/tools/power/x86/intel-speed-select
   make
@@ -160,6 +161,7 @@ build() {
   pushd linux/tools/arch/x86/kcpuid
   make
   popd
+COMMENT_SEPARATOR
 }
 
 package_linux-tools-meta() {
@@ -169,14 +171,9 @@ package_linux-tools-meta() {
     'bootconfig'
     'bpf'
     'cpupower'
-    'hyperv'
-    'intel-speed-select'
-    'kcpuid'
     'perf'
     'tmon'
-    'turbostat'
     'usbip'
-    'x86_energy_perf_policy'
   )
   conflicts=(
     'acpidump'
