diff --git a/PKGBUILD b/PKGBUILD
index edfbadc..7582569 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -69,6 +69,8 @@ prepare() {
 
   # Remove CMake find module for zstd; breaks if out of sync with upstream zstd
   rm cmake/modules/Findzstd.cmake
+
+  patch -p2 -i "${srcdir}/LoongArch-Allow-f16-codegen-with-expansion-to-libcalls.patch"
 }
 
 build() {
@@ -158,4 +160,7 @@ package_llvm-libs() {
     "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
 
+source+=("LoongArch-Allow-f16-codegen-with-expansion-to-libcalls.patch::https://github.com/llvm/llvm-project/pull/94456/commits/bbb1854a5a76a8199c50d60a2ae5809163f9d1c5.patch")
+sha256sums+=('c384b21354a23f9b8ad7d376edc1d21f8f2d67fae693d0be7281323aeca61885')
+
 # vim:set ts=2 sw=2 et:
