--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,6 +23,9 @@ sha256sums=('f0e8bb1f9b7eb0b01285495a2699df3a4b766784c1765a8f1aeedf63c0806369'
 
 options=('!makeflags')
 
+source+=('fix-config.patch')
+sha256sums+=('120b080fbef489bd3c2328b7431fd30eb7be1b9f9105f2556a6d6f8200dd17f8')
+
 prepare() {
 	cd "${srcdir}/${pkgname}${_pkgver}"
 	sed -e "/^CFLAGS_NOOPT =/s/\$/ $CPPFLAGS $CFLAGS/" -i unix/Makefile
@@ -31,6 +34,7 @@ prepare() {
 	patch -p1 -i ../zip-3.0-exec-shield.patch
 	patch -p1 -i ../zip-3.0-format-security.patch
 	patch -p1 -i ../zipnote.patch
+	patch -p1 -i ../fix-config.patch
 }
 
 build() {
