diff --git a/PKGBUILD b/PKGBUILD
index a202625..eb68cb8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,6 +23,10 @@ validpgpkeys=('FFF3E01444FED7C316A3545A895F5BC123A02740')
 build() {
     cd "${srcdir}/${pkgname}_v${pkgver}_src-deps"
 
+    # Update jacobsa-crypto to v1.1.0 to support loong64
+    go mod edit -replace github.com/aperturerobotics/jacobsa-crypto=github.com/aperturerobotics/jacobsa-crypto@v1.1.0
+    go mod tidy
+    go mod vendor
     export CGO_LDFLAGS="$LDFLAGS"
     export GOFLAGS="-buildmode=pie -trimpath -mod=vendor"
     make build
