--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,7 +37,7 @@ depends=(
 )
 makedepends=(
   cmake
-  cuda
+#  cuda
   gcc-fortran
   gtest
   ninja
@@ -68,10 +68,11 @@ build() {
     -W no-dev
     -DCMAKE_BUILD_TYPE=None
     -DCMAKE_INSTALL_PREFIX=/usr
-    -DCMAKE_CUDA_ARCHITECTURES="$cuda_archs"
+#    -DCMAKE_CUDA_ARCHITECTURES="$cuda_archs"
     # Compile source code for supported GPU archs in parallel
-    -DCMAKE_CUDA_FLAGS="--threads $(nproc)"
-    -DADIOS2_USE_CUDA=ON
+#    -DCMAKE_CUDA_FLAGS="--threads $(nproc)"
+#    -DADIOS2_USE_CUDA=ON
+    -DDILL_NATIVE_ONLY=OFF
     -DADIOS2_USE_EXTERNAL_DEPENDENCIES=ON
     -DADIOS2_USE_EXTERNAL_PERFSTUBS=OFF
     # external nanobind does not work: https://gitlab.archlinux.org/archlinux/packaging/packages/nanobind/-/merge_requests/3
