diff --git a/PKGBUILD b/PKGBUILD
index dcda765..3ad9a9e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -39,6 +39,8 @@ pkgver() {
 prepare() {
   mkdir -p glibc-build lib32-glibc-build
 
+  export CFLAGS=${CFLAGS/-fno-plt/}
+
   [[ -d glibc-$pkgver ]] && ln -s glibc-$pkgver glibc
   cd glibc
 }
@@ -51,7 +53,6 @@ build() {
       --enable-bind-now
       --enable-fortify-source
       --enable-kernel=4.4
-      --enable-multi-arch
       --enable-stack-protector=strong
       --enable-systemtap
       --disable-nscd
@@ -83,6 +84,7 @@ build() {
     make info
   )
 
+  : <<COMMENT_SEPARATOR
   (
     cd lib32-glibc-build
     export CC="gcc -m32 -mstackrealign"
@@ -105,6 +107,7 @@ build() {
 
     make -O
   )
+COMMENT_SEPARATOR
 
   # pregenerate locales here instead of in package
   # functions because localedef does not like fakeroot
@@ -224,3 +227,6 @@ package_glibc-locales() {
   # deduplicate locale data
   hardlink -c "${pkgdir}"/usr/lib/locale
 }
+
+pkgname=(${pkgname[@]/lib32-glibc})
+makedepends=(${makedepends[@]/lib32-gcc-libs})
