diff --git a/PKGBUILD b/PKGBUILD
index 26c440a..82cb7cb 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,14 +7,14 @@
 # NOTE: valgrind requires rebuilt with each major glibc version
 
 pkgbase=glibc
-pkgname=(glibc lib32-glibc glibc-locales)
+pkgname=(glibc glibc-locales)
 pkgver=2.40+r16+gaa533d58ff
 _commit=aa533d58ff12e27771d9c960a727d74992a3f2a3
 pkgrel=2
 arch=(x86_64)
 url='https://www.gnu.org/software/libc'
 license=(GPL-2.0-or-later LGPL-2.1-or-later)
-makedepends=(git gd lib32-gcc-libs python)
+makedepends=(git gd python)
 options=(staticlibs !lto)
 source=("git+https://sourceware.org/git/glibc.git#commit=${_commit}"
         locale.gen.txt
@@ -37,7 +37,7 @@ pkgver() {
 }
 
 prepare() {
-  mkdir -p glibc-build lib32-glibc-build
+  mkdir -p glibc-build
 
   [[ -d glibc-$pkgver ]] && ln -s glibc-$pkgver glibc
   cd glibc
@@ -51,7 +51,6 @@ build() {
       --enable-bind-now
       --enable-fortify-source
       --enable-kernel=4.4
-      --enable-multi-arch
       --enable-stack-protector=strong
       --enable-systemtap
       --disable-nscd
@@ -83,6 +82,7 @@ build() {
     make info
   )
 
+  : <<COMMENT_SEPARATOR
   (
     cd lib32-glibc-build
     export CC="gcc -m32 -mstackrealign"
@@ -105,6 +105,7 @@ build() {
 
     make -O
   )
+COMMENT_SEPARATOR
 
   # pregenerate locales here instead of in package
   # functions because localedef does not like fakeroot
