--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,6 +46,8 @@ pkgver() {
 prepare() {
   mkdir -p glibc-build lib32-glibc-build
 
+  export CFLAGS=${CFLAGS/-fno-plt/}
+
   [[ -d glibc-$pkgver ]] && ln -s glibc-$pkgver glibc
   cd glibc
 }
@@ -58,7 +60,6 @@ build() {
       --enable-bind-now
       --enable-fortify-source
       --enable-kernel=4.4
-      --enable-multi-arch
       --enable-stack-protector=strong
       --enable-systemtap
       --disable-nscd
@@ -155,7 +156,7 @@ check() (
   _skip_test tst-shstk-legacy-1g     sysdeps/x86_64/Makefile
   _skip_test tst-adjtime             time/Makefile
 
-  make -O check
+  make TIMEOUTFACTOR=10 -O check
 )
 
 package_glibc() {
@@ -240,3 +241,6 @@ package_glibc-locales() {
   # deduplicate locale data
   hardlink -c "${pkgdir}"/usr/lib/locale
 }
+
+pkgname=(${pkgname[@]/lib32-glibc})
+makedepends=(${makedepends[@]/lib32-gcc-libs})
