--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,6 +46,8 @@ pkgver() {
 prepare() {
   mkdir -p glibc-build lib32-glibc-build
 
+  export CFLAGS=${CFLAGS/-fno-plt/}
+
   [[ -d glibc-$pkgver ]] && ln -s glibc-$pkgver glibc
   cd glibc
 }
@@ -58,7 +60,6 @@ build() {
       --enable-bind-now
       --enable-fortify-source
       --enable-kernel=4.4
-      --enable-multi-arch
       --enable-stack-protector=strong
       --enable-systemtap
       --disable-nscd
@@ -82,7 +83,7 @@ build() {
         --libdir=/usr/lib \
         --libexecdir=/usr/lib \
         --enable-cet \
-        --enable-sframe \
+        --disable-sframe \
         "${_configure_flags[@]}"
 
     make -O
@@ -91,6 +92,7 @@ build() {
     make info
   )
 
+  : <<COMMENT_SEPARATOR
   (
     cd lib32-glibc-build
     export CC="gcc -m32 -mstackrealign"
@@ -113,6 +115,7 @@ build() {
 
     make -O
   )
+COMMENT_SEPARATOR
 
   # pregenerate locales here instead of in package
   # functions because localedef does not like fakeroot
@@ -148,7 +151,7 @@ check() (
   _skip_test tst-shstk-legacy-1g     sysdeps/x86_64/Makefile
   _skip_test tst-adjtime             time/Makefile
 
-  make -O check
+  make TIMEOUTFACTOR=10 -O check
 )
 
 package_glibc() {
@@ -232,3 +235,6 @@ package_glibc-locales() {
   # deduplicate locale data
   hardlink -c "${pkgdir}"/usr/lib/locale
 }
+
+pkgname=(${pkgname[@]/lib32-glibc})
+makedepends=(${makedepends[@]/lib32-gcc-libs})
