diff --git a/PKGBUILD b/PKGBUILD
index 5318e60..b8033e0 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -9,7 +9,7 @@ arch=(x86_64)
 url='https://github.com/containers/toolbox'
 license=(APACHE)
 depends=(podman bash flatpak)
-makedepends=(go shellcheck go-md2man bash-completion ninja git meson podman fish)
+makedepends=(go go-md2man bash-completion ninja git meson podman fish)
 source=("git+$url#tag=$pkgver")
 sha256sums=('9b17fc5298d1c60aeca593b72fc47b42dd2d24d675a851dfb4538be025a2e069')
 
