diff --git a/PKGBUILD b/PKGBUILD
index bc30e54..acf75ca 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,7 +10,6 @@ url='https://rocm.docs.amd.com/projects/HIP/en/latest/'
 license=('MIT')
 _amd_depends=('rocm-core' 'bash' 'perl' 'glibc' 'gcc-libs' 'numactl'
          'mesa' 'comgr' 'rocminfo' 'rocm-llvm' 'libelf' 'rocprofiler-register')
-_nvidia_depends=('cuda')
 makedepends=('git' 'cmake' 'python' 'python-cppheaderparser'
              "${_amd_depends[@]}" "${_nvidia_depends[@]}")
 _tag="tag=rocm-$pkgver"
@@ -32,6 +31,8 @@ sha256sums=('2044f7736ea777e258c2956c89a65bea62102eb1db94edd9e0414d44a93e3c93'
             '09f332aa964079647a9407406305f6159937c72c0bc2b675f841ba0ab1b63c43')
 
 build() {
+  patch -Np1 -d "$srcdir/$pkgbase-clr" -i "$srcdir/clr-loong64-support.patch"
+  sed -i 's!x86_64-unknown-linux-gnu!loongarch64-unknown-linux-gnu!g' "$srcdir/$pkgbase-clr"/hipamd/src/hip_embed_pch.sh
   local hipcc_common_args=(
     -Wno-dev
     -S "$srcdir/$pkgbase-hipcc/amd/hipcc"
@@ -63,6 +64,8 @@ build() {
   cmake "${hip_amd_args[@]}"
   cmake --build build-amd
 
+  # Use a "multi-line comment" to keep patch from rotting
+  : <<COMMENT_SEPARATOR
   local hipcc_nvidia_args=(
     "${hipcc_common_args[@]}"
     -B build-nvidia-hipcc
@@ -87,6 +90,7 @@ build() {
   )
   cmake "${hip_nvidia_args[@]}"
   cmake --build build-nvidia
+COMMENT_SEPARATOR
 }
 
 package_hip-runtime-amd() {
@@ -106,3 +110,7 @@ package_hip-runtime-nvidia() {
   DESTDIR="$pkgdir" cmake --install build-nvidia
   install -Dm644 "$srcdir/$pkgbase/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '^(hip-runtime-nvidia)$'))
+source+=("clr-loong64-support.patch::https://raw.githubusercontent.com/loongarch-moe/rocm-loongarch/f3e08d630fb7151f22e54b8c8cf12bb99413d876/stage2/5.rocm-clr/clr.patch")
+sha256sums+=('464f01d509d90574959d5e6e06f5da8636aa735402266cc0eb0ef0b3c41e77d3')
