diff --git a/PKGBUILD b/PKGBUILD
index 2ee728a..4a70b24 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -35,8 +35,6 @@ makedepends=(
   libspeechd
   libxaw
   ncurses
-  ocaml-ctypes
-  ocaml-findlib
   pcre2
   polkit
   python-setuptools
@@ -116,7 +114,6 @@ package_brltty() {
     'libx11: for xbrlapi'
     'libxfixes: for xbrlapi'
     'libxtst: for xbrlapi'
-    'ocaml: OCaml support'
     'python: Python support'
     'speech-dispatcher: speech-dispatcher driver'
     'tcl: tcl support'
