--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,11 +14,11 @@ pkgname=(
   uwsgi-plugin-psgi
   uwsgi-plugin-cgi
   uwsgi-plugin-python
-  uwsgi-plugin-pypy
+  #uwsgi-plugin-pypy
   uwsgi-plugin-php
   uwsgi-plugin-php-legacy
   uwsgi-plugin-lua51
-  uwsgi-plugin-mono
+  #uwsgi-plugin-mono
   uwsgi-plugin-webdav
   uwsgi-plugin-zabbix
   uwsgi-plugin-notfound
@@ -35,14 +35,14 @@ makedepends=(
   libxcrypt
   libxml2
   lua51
-  mono
+  #mono
   openssl
   pam
   pcre2
   perl
   php-embed
   php-legacy-embed
-  pypy
+  #pypy
   python
   python-gevent
   python-greenlet
@@ -97,6 +97,7 @@ prepare() {
   # link the php_legacy plugin against libphp-legacy.so
   sed -e 's/lphp/lphp-legacy/g; s/UWSGICONFIG_PHPPATH/UWSGICONFIG_PHP_LEGACY_PATH/g; s/NAME=.*/NAME="php_legacy"/g' -i plugins/php_legacy/uwsgiplugin.py
   sed -e 's/php_plugin/php_legacy_plugin/g' -i plugins/php_legacy/php_plugin.c
+  sed -i 's/mono, //;s/pypy, //' ../archlinux.ini
   # copy our custom build profile into place
   cp -v ../archlinux.ini buildconf/
 }
