diff --git a/PKGBUILD b/PKGBUILD
index 0340c45..4f369d4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -45,8 +45,13 @@ sha512sums=('bdfdab4374174af9a49d4c25ea456e867a2e30a7795b751a77c946df0560c9418ba
 
 prepare() {
   cd ${pkgbase}
+  # fix modesetting driver for loongson and gsgpu
+  patch -Np1 -i ../0001-modesetting-match-against-Multimedia-Video-Controlle.patch
 }
 
+source+=('0001-modesetting-match-against-Multimedia-Video-Controlle.patch')
+sha512sums+=('0e55cc994dd8f1309c6ea40cb1f5b2d763d850a466c9d30bd8619fd68ce369b19f098d28ab684db1bd2758a6402654cf93188651e863f0e1be4eed8cf9b40e14')
+
 build() {
   # Since pacman 5.0.2-2, hardened flags are now enabled in makepkg.conf
   # With them, modules fail to load with undefined symbol.
@@ -64,6 +69,7 @@ build() {
     -D xephyr=true \
     -D glamor=true \
     -D udev=true \
+    -D udev_kms=true \
     -D dtrace=false \
     -D systemd_logind=true \
     -D suid_wrapper=true \
