diff --git a/PKGBUILD b/PKGBUILD
index 17d0822..590b96c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -33,6 +33,7 @@ b2sums=('66cb636f9f7d65f2aa739315db7382ace980b886f9cc38cc3d81c41f432874f6fb33b0d
 
 prepare() {
   cd "${pkgname}-${pkgver}"
+  patch -Np1 -i "${srcdir}/fix-dwarf-addsub-source-info.patch"
   cp "${srcdir}/test-config.toml" .
   cargo fetch --locked --target host-tuple
 }
@@ -59,4 +60,8 @@ package() {
   install -Dm 644 LICENSE-* -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+source+=('fix-dwarf-addsub-source-info.patch')
+sha256sums+=('a7963d6a3c3c8200e9227d216ab44ccfd4dcc1059a9da19d765df836556b165b')
+b2sums+=('751cdca74fe5484fe74be64dc64f0c28c1b1ca27dc2968cd2b0f4ce8ad24637e52a6e06c1c7c9575c2a248dbdc2d4c7983b1723802426a7f963d757f6115c463')
+
 # vim: ts=2 sw=2 et:
