diff --git a/PKGBUILD b/PKGBUILD
index 035ecbe..e3f9060 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,6 +21,9 @@ sha256sums=('76bfac6fba31a9c4ec196d9b9b2d5ec51b8b68840b3fba8686aa42323d76a425'
 _dirname="$(basename "$url")-$(basename "${source[0]}" .tar.gz)"
 
 prepare() {
+    # clang on loong64 doesn't support `-fstack-clash-protection` and will cause Werror in `check()`
+    export CXXFLAGS="${CXXFLAGS//-fstack-clash-protection/}"
+    export CFLAGS="${CFLAGS//-fstack-clash-protection/}"
     cd "$_dirname"
     # Git version tests fail because we're not working in a local git checkout
     rm test/pass/{version-norepo.cmake,version-parent.cmake}
