--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,6 +17,10 @@ b2sums=('df1b8706f3639c52ec5905a542052ee0eb7c5a727c9ee27af7c9f53b2b9f97b0445c6c6
 
 prepare() {
   cd $pkgname
+  git cherry-pick -n 396096223ec58ff7f437ec0de7275240946531c5
+  git cherry-pick -n d2ee6351c08518c866bb48d89f58a67bb36931fc
+  git cherry-pick -n a6fcb73b11fc518dc60495acd408fcaf4c423baa
+  git cherry-pick -n 3bf931e5ca5b86ea738cda9f8fc339a8b1145dca
   sed -i 's/-interim git//g' src/version.h
   setconf scripts/genGitHdr.sh GIT_URL "'""${source:4:34}""'"
   setconf scripts/genGitHdr.sh GIT_REV "${source#*=}"
