--- a/PKGBUILD
+++ b/PKGBUILD
@@ -53,6 +53,7 @@ prepare() {
 
   # HACK from Fedora's spec: Replace git tag with version of our tar archive
   sed -i "s/NOT latest_esmi_tag/NOT esmi_pkg_ver-$_esmi_ver/" CMakeLists.txt
+  patch -Np1 -i "$srcdir"/amdsmi_go_shim_loongarch64.patch
 }
 
 build() {
@@ -63,6 +64,7 @@ build() {
     -D CMAKE_INSTALL_PREFIX=/opt/rocm
     -D CMAKE_BUILD_TYPE=None
     -D BUILD_TESTS=OFF
+    -D ENABLE_ESMI_LIB=OFF
   )
   cmake "${cmake_args[@]}"
   cmake --build build
@@ -81,3 +83,6 @@ package() {
   # cmake also installs python package under share.
   rm -r "${pkgdir}"/opt/rocm/share/amd_smi
 }
+
+source+=("amdsmi_go_shim_loongarch64.patch")
+sha256sums+=('cbdb2aab71903152762ab7367e30cc3a004f25a2ee411d5abef77713adb70bad')
