diff --git a/PKGBUILD b/PKGBUILD
index a26cdab..8359a92 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,9 +20,13 @@ _archive="$pkgname-$pkgver"
 source=("$_url/archive/v$pkgver/$_archive.tar.gz")
 sha256sums=('b1ba054e821073daafd90675c4822bcd8166f33fe2e3acba87ba1451a0d1fc56')
 
+source+=("00-update-lock.patch")
+sha256sums+=('7f190d940242096d34a12da0c8bd381353e01aa7688a01a2f8a872b2522e8926')
+
 prepare() {
        cd "$_archive"
-       cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+       patch Cargo.lock < ../00-update-lock.patch
+       cargo fetch --locked
 }
 
 build() {
