diff --git a/PKGBUILD b/PKGBUILD
index dc90256..0697c44 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,4 +37,15 @@ package() {
   install -Dm644 LICENSE "$pkgdir"/usr/share/licenses/box2d/LICENSE
 }
 
+prepare() {
+  cd $pkgname-$pkgver
+  patch -p1 -i "${srcdir}/remove-unsupported-cpu.patch"
+  patch -d "extern" -p1 -i "${srcdir}/Fix-type-convert-error-for-LSX.patch"
+}
+
+source+=("remove-unsupported-cpu.patch"
+        "Fix-type-convert-error-for-LSX.patch")
+sha512sums+=('cfd1eebb7cf807541be6307481fe2f68dbd506c7f0aa5287b267eceb04a5836ad304987d4625d77cc5c624e073f9319f4007d18d4a48a625234b8edadc388fb0'
+            'd97e4e3c36bed2a710fbc9055337cbd27c4eb6b4bb70545a9abd539b5f8ef29195eadd64a6fb481d40924ae36133b1111e6861c65d4bcab69e73465578761c55')
+
 # vim: sw=2 ts=2 et:
