diff --git a/PKGBUILD b/PKGBUILD
index df3e7ba..ca658d0 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,6 +19,7 @@ prepare() {
   cd $_pkgname-$pkgver
 
   patch -p1 < ../add-dependency.patch
+  patch -p1 < ../add-definition.patch
 }
 
 build() {
@@ -44,3 +45,5 @@ package() {
 
   DESTDIR="$pkgdir" ninja install
 }
+source+=("add-definition.patch")
+sha512sums+=('542de2619d73fe391e46c4f364c392eaf2a3367cc2728fce1f427147b10a2e026a700eea6c1761ec4a577a563ae304c31ec7d66e49d42fa152bfe2a6869d1598')
