diff --git a/PKGBUILD b/PKGBUILD
index 0b3d993..6005ee3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -65,6 +65,7 @@ check() {
 
 prepare() {
   cd $pkgname-$pkgver
+  patch -Np1 -i "${srcdir}/0001-fix-installer-skip-waiting-for-mirror-selection-on-l.patch"
   # Patching away `wayfire` as it's an AUR package.
   # Instead of pushing an entire release on github, we patch
   # here for simplicity until the next version is out.
@@ -82,3 +83,7 @@ package() {
   python -m installer --destdir="$pkgdir" dist/*.whl
   install -vDm 644 docs/_build/man/archinstall.1 -t "$pkgdir/usr/share/man/man1/"
 }
+
+source+=("0001-fix-installer-skip-waiting-for-mirror-selection-on-l.patch")
+sha512sums+=('dc28724dfe4ed25a88de0516296233dff75809737ef979cdbe8be32f068774d3addd8748bcfe49e2416d82d0cd740979ce716d91f0450078ea03b7e0068fdb45')
+b2sums+=('63cd91f10f53b9f7aba8370da33d70793a796d036298cdc5ca9adc6ebb040a6189eee9c30532b373e5890745d6f8a1189e38cb17bd828ac1eff57fdd904a593b')
