diff --git a/PKGBUILD b/PKGBUILD
index eacc0fc..bb3dbce 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -65,6 +65,8 @@ check() {
 
 prepare() {
   cd $pkgname-$pkgver
+  patch -Np1 -i "${srcdir}/fix-installer-skip-waiting-for-mirror-selection-on-l.patch"
+  patch -Np1 -i "${srcdir}/fix-installer-use-platform.machine-for-target-archit.patch"
   # Patching away `wayfire` as it's an AUR package.
   # Instead of pushing an entire release on github, we patch
   # here for simplicity until the next version is out.
@@ -82,3 +84,10 @@ package() {
   python -m installer --destdir="$pkgdir" dist/*.whl
   install -vDm 644 docs/_build/man/archinstall.1 -t "$pkgdir/usr/share/man/man1/"
 }
+
+source+=("fix-installer-use-platform.machine-for-target-archit.patch::https://github.com/wszqkzqk/archinstall/commit/1681304c614efc920d17120656dbda5c36e543cc.diff"
+         "fix-installer-skip-waiting-for-mirror-selection-on-l.patch::https://github.com/wszqkzqk/archinstall/commit/94b75f6cd9e50f43ba7ec83e3d966895cd8c0ce8.diff")
+sha512sums+=('c0c19f879e53604ac0816b6ad08e9141b399082bc0093697edd07a85a19ba8babb83f86a4e397301a7f7ad09b248440db47631175b5c630d9a4189811f06e3cd'
+             '0065e4bed0f8427c48708f9a28e131acc929be65d263e801061d87e2d8de3fff1109845196cedb1dfab7cbf15e9c6b17a7aea354e1a4a7078e1193ae3174d401')
+b2sums+=('f731c8a3db578af66cb639a1410d43575e2f5983a17b60a3e16b9db7d6c95b29729200908ee3a792ba62463d0ae6793dd5d91721df90f0570dba193b2c32cebb'
+         '8fa9698f9e397162e04bcd4aec05e6eaa436ae5a2f6a0117ecd26403ce0b7879ea2aaa8cee08e145bf19e1ba96b9bf9e6c1eff91a782963d0f040e7caaa42502')
