--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,7 +23,7 @@ depends=(
   grep
   libarchive libarchive.so
   openssl libcrypto.so
-  pacman-mirrorlist
+  pacman-mirrorlist-loong64
   systemd
   libmakepkg-dropins
 )
@@ -71,6 +71,15 @@ pkgver() {
 }
 
 prepare() {
+  if [[ $CARCH != "x86_64" ]]; then
+    cp --dereference "$srcdir/pacman.conf" "$srcdir/pacman.$CARCH.conf"
+    cp --dereference "$srcdir/makepkg.conf" "$srcdir/makepkg.$CARCH.conf"
+    patch "$srcdir/pacman.$CARCH.conf" -i "$srcdir/pacman.$CARCH.diff"
+    patch "$srcdir/makepkg.$CARCH.conf" -i "$srcdir/makepkg.$CARCH.diff"
+    mv -f "$srcdir/pacman.$CARCH.conf" "$srcdir/pacman.conf"
+    mv -f "$srcdir/makepkg.$CARCH.conf" "$srcdir/makepkg.conf"
+  fi
+
   cd "$pkgname"
 
   # apply patch level commits on top of annotated tag
@@ -143,4 +152,11 @@ package() {
   done
 }
 
+source+=(makepkg.loong64.diff
+         pacman-use-loong64.patch
+         pacman.loong64.diff)
+sha256sums+=('008ca7e3b57a30b079318e9425d2440d5674aeddf7a9de2d18595411cd5e4ee2'
+             '5ab2a0ae2747184e7a3b0d4f6f45f0a66b6ad85e853b78fb77102e4b32667cdc'
+             'fadd0c5453aadfcb89cebb0bfc64674f68c7d988e512d8a4fca14718afa2cc62')
+
 # vim: set ts=2 sw=2 et:
