diff --git a/PKGBUILD b/PKGBUILD
index 9bcaec8..9d913c4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -24,6 +24,8 @@ build() {
     -D CMAKE_BUILD_TYPE=None
     -D CMAKE_CXX_COMPILER=/opt/rocm/bin/hipcc
     -D CMAKE_CXX_FLAGS="${CXXFLAGS} -fcf-protection=none"
+    -D CMAKE_EXE_LINKER_FLAGS="-fuse-ld=mold"
+    -D CMAKE_SHARED_LINKER_FLAGS="-fuse-ld=mold"
   )
   cmake "${cmake_args[@]}"
   cmake --build build
@@ -34,3 +36,5 @@ package() {
 
   install -Dm644 "$_dirname/LICENSE.md" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+makedepends+=(mold)
