diff --git a/PKGBUILD b/PKGBUILD
index 16ddc0b..26826d2 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -50,6 +50,7 @@ case "${CARCH}" in
   'x86_64'  ) _JARCH=amd64   ; _DOC_ARCH=x86_64  ;;
   'i686'    ) _JARCH=i386    ; _DOC_ARCH=x86     ;;
   'aarch64' ) _JARCH=aarch64 ; _DOC_ARCH=aarch64 ;;
+  'loong64') _JARCH=loongarch64; _DOC_ARCH=loongarch64 ;;
 esac
 
 _jdkname=openjdk8
@@ -64,7 +65,7 @@ _nonheadless=(
 )
 
 prepare() {
-  cd jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}
+  cd jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}-ls-0
 
   # Do not treats warnings as errors
   sed -E -i 's/(^WARNINGS_ARE_ERRORS = -W)(error)/\1no-\2/' \
@@ -94,7 +95,7 @@ END
 }
 
 build() {
-  cd jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}
+  cd jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}-ls-0
 
   unset JAVA_HOME
 
@@ -147,7 +148,7 @@ build() {
 }
 
 check() {
-  cd jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}
+  cd jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}-ls-0
   #make -k test
 }
 
@@ -348,8 +349,11 @@ package_openjdk8-doc() {
   pkgdesc='OpenJDK Java 8 documentation'
 
   install -d -m 755 "${pkgdir}/usr/share/doc/${pkgbase}/"
-  cp -r "${srcdir}"/jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}/build/linux-${_DOC_ARCH}-normal-server-release/docs/* \
+  cp -r "${srcdir}"/jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}-ls-0/build/linux-${_DOC_ARCH}-normal-server-release/docs/* \
     "${pkgdir}/usr/share/doc/${pkgbase}/"
 }
 
 # vim: ts=2 sw=2 et:
+
+source[0]="https://github.com/loongson/jdk${_majorver}u/archive/refs/tags/jdk${_majorver}u${_minorver}-b${_updatever}-ls-0.tar.gz"
+b2sums[0]='b2ba7bb9a9cdfab36798b57967607f84b2a91ceffb977de66bf121f53083dc544f60d207da885b93f42c08448cb6ddbcdbf0359b672ec4010bf539931a366275'
