diff --git a/PKGBUILD b/PKGBUILD
index 6977346..1d0ab4f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,7 +26,7 @@ build() {
 check() {
   cd "${pkgname}"
   local python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
-  PYTHONPATH="$PWD/build/lib.linux-$CARCH-cpython-$python_version" pytest
+  PYTHONPATH="$PWD/build/lib.linux-$(uname -m)-cpython-$python_version" pytest
 }
 
 package() {
