--- a/PKGBUILD
+++ b/PKGBUILD
@@ -55,10 +55,16 @@ package() {
   meson install -C build --destdir "$pkgdir"
 
   local p
-  for p in {x86_64,i686}-pc-linux-gnu; do
-    install -Dt "$pkgdir/usr/share/pkgconfig/personality.d" -m644 $p.personality
-    ln -s pkgconf "$pkgdir/usr/bin/$p-pkg-config"
-  done
+  if [[ $CARCH == loong64 ]];then
+    install -Dt "$pkgdir/usr/share/pkgconfig/personality.d" -m644 \
+      loongarch64-unknown-linux-gnu.personality
+    ln -s pkgconf "$pkgdir/usr/bin/loongarch64-unknown-linux-gnu-pkg-config"
+  else
+    for p in {x86_64,i686}-pc-linux-gnu; do
+      install -Dt "$pkgdir/usr/share/pkgconfig/personality.d" -m644 $p.personality
+      ln -s pkgconf "$pkgdir/usr/bin/$p-pkg-config"
+    done
+  fi
 
   ln -s pkgconf "$pkgdir/usr/bin/pkg-config"
   ln -s pkgconf.1 "$pkgdir/usr/share/man/man1/pkg-config.1"
@@ -66,4 +72,7 @@ package() {
   install -Dt "$pkgdir/usr/share/licenses/$pkgname" -m644 pkgconf/COPYING
 }
 
+source+=(loongarch64-unknown-linux-gnu.personality)
+b2sums+=('335aac01ee674f3d0d4d895145fa0fba29f053338ab6d5f9225a16d6f25de841ec812a77dc9bdd6fa7f0fe7a1f4e47cd6762540700741874a6190b3e41c5222e')
+
 # vim:set sw=2 sts=-1 et:
