diff -urN postgresql-16.3.orig/src/include/storage/s_lock.h postgresql-16.3/src/include/storage/s_lock.h
--- postgresql-16.3.orig/src/include/storage/s_lock.h	2024-09-28 22:17:04.858619319 +0800
+++ postgresql-16.3/src/include/storage/s_lock.h	2024-09-28 22:20:36.064578007 +0800
@@ -256,7 +256,7 @@
  * We use the int-width variant of the builtin because it works on more chips
  * than other widths.
  */
-#if defined(__arm__) || defined(__arm) || defined(__aarch64__)
+#if defined(__arm__) || defined(__arm) || defined(__aarch64__) || defined(__loongarch64)
 #ifdef HAVE_GCC__SYNC_INT32_TAS
 #define HAS_TEST_AND_SET
 
