--- a/PKGBUILD
+++ b/PKGBUILD
@@ -98,6 +98,7 @@ prepare() {
   patch -Np1 < ../0001-Set-DEFAULT_PGSOCKET_DIR-to-run-postgresql.patch
   patch -Np1 < ../0002-Force-RPATH-to-be-used-for-the-PL-Perl-plugin.patch
   patch -Np1 < ../0003-doc-Remove-stray-trailing-whitespace-in-xref-linkend.patch
+  patch -p1 < ../add-loongarch-support.patch
 }
 
 build() {
@@ -117,7 +118,6 @@ build() {
     --with-liburing
     --with-libxml
     --with-libxslt
-    --with-llvm
     --with-lz4
     --with-openssl
     --with-pam
@@ -142,6 +142,7 @@ build() {
 
 check() (
   export LANG=C LC_ALL=C
+  export PGTZ="America/Los_Angeles"
   if ! make -C build check-world; then
     find . -name regression.diffs | while read -r line; do
       echo "make check-world failure: ${line}"
@@ -257,4 +258,11 @@ package_postgresql-docs() {
   install -Dm644 postgresql-${pkgver}/COPYRIGHT -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+makedepends=(${makedepends[@]/clang})
+makedepends=(${makedepends[@]/llvm})
+depends=(${depends[@]/llvm-libs})
+source+=('add-loongarch-support.patch')
+md5sums+=('49336e9f0512a980cbb661be0bb4a509')
+sha256sums+=('da98a27c64c2d41c48ab039fe134076fe70fe6e2316dff487e405463f8007121')
+b2sums+=('0173ab0f556ce569bd5d93e055b07eebb17d784460d14023ebecf4d6da1c0f8df9be10519a097dcabfd9a1a1582360e75f3b9daa7ac9f8bbb0862e682408a2f9')
 # vim:set sw=2 sts=-1 et:
