diff --git a/PKGBUILD b/PKGBUILD
index 34cabad..4a48f7a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -79,6 +79,10 @@ prepare() {
 
   echo "Setting config..."
   cp ../config .config
+  if [ $CARCH == loong64 ]; then
+    make savedefconfig
+    cat defconfig ../loong-addition.config > .config
+  fi
   make olddefconfig
   diff -u ../config .config || :
 
@@ -147,7 +151,11 @@ _package-headers() {
   install -Dt "$builddir" -m644 .config Makefile Module.symvers System.map \
     localversion.* version vmlinux # tools/bpf/bpftool/vmlinux.h
   install -Dt "$builddir/kernel" -m644 kernel/Makefile
-  install -Dt "$builddir/arch/x86" -m644 arch/x86/Makefile
+  if [ $CARCH == x86_64 ]; then
+    install -Dt "$builddir/arch/x86" -m644 arch/x86/Makefile
+  elif [ $CARCH == loong64 ]; then
+    install -Dt "$builddir/arch/loongarch" -m644 arch/loongarch/Makefile
+  fi
   cp -t "$builddir" -a scripts
   # ln -srt "$builddir" "$builddir/scripts/gdb/vmlinux-gdb.py"
 
@@ -159,8 +167,13 @@ _package-headers() {
 
   echo "Installing headers..."
   cp -t "$builddir" -a include
-  cp -t "$builddir/arch/x86" -a arch/x86/include
-  install -Dt "$builddir/arch/x86/kernel" -m644 arch/x86/kernel/asm-offsets.s
+  if [ $CARCH == x86_64 ]; then
+    cp -t "$builddir/arch/x86" -a arch/x86/include
+    install -Dt "$builddir/arch/x86/kernel" -m644 arch/x86/kernel/asm-offsets.s
+  elif [ $CARCH == loong64 ]; then
+    cp -t "$builddir/arch/loongarch" -a arch/loongarch/include
+    install -Dt "$builddir/arch/loongarch/kernel" -m644 arch/loongarch/kernel/asm-offsets.s
+  fi
 
   install -Dt "$builddir/drivers/md" -m644 drivers/md/*.h
   install -Dt "$builddir/net/mac80211" -m644 net/mac80211/*.h
@@ -182,7 +195,11 @@ _package-headers() {
   echo "Removing unneeded architectures..."
   local arch
   for arch in "$builddir"/arch/*/; do
-    [[ $arch = */x86/ ]] && continue
+    if [ $CARCH == x86_64 ]; then
+      [[ $arch = */x86/ ]] && continue
+    elif [ $CARCH == loong64 ]; then
+      [[ $arch = */loongarch/ ]] && continue
+    fi
     echo "Removing $(basename "$arch")"
     rm -r "$arch"
   done
@@ -250,4 +267,14 @@ for _p in "${pkgname[@]}"; do
   }"
 done
 
+source+=('loong-addition.config'
+         '0001-LOONGARCH64-drm-xe-fix-build-on-non-4K-kernels.patch'
+         '0002-LOONGARCH64-drm-radeon-Call-mmiowb-at-the-end-of-rad.patch')
+sha256sums+=('6f83b39a55b57e46149af253bba9670dc2e164f1721d70bd0b401dd6661a35d7'
+             '83fa0ac8b4874311b8c590b1fd8d9036679f8bbd51c10da5509742167456b9ed'
+             'b25bbb9d0afe724aa49cf96ffbd5e88e8b39414309d9b690b64c664d18b54834')
+b2sums+=('c4b1a3dadb8fb5c07e870350c1bb462cd198ef3224f689af4e486b35da8d73e406669a8391f1291f16747ecaeff206dfc1d5220f5806a8990d2a265c3d9b980f'
+         '264d58172c877b1cda3f2016c4f3e4a6fcf37b17e4a2de2713bc99570c597f9ef0f947df6cdff5a93cc96849fe10582fb2decf54baabcc803fde96e889c107d7'
+         '3fca438a40be08dede4badcf973629070cb6c50567d215d6d42638dfcf3452d01aae9c253b783459d9cdf8ee1c08a394c47db2e2e2b1394898c0b03f3df8407f')
+
 # vim:set ts=8 sts=2 sw=2 et:
