--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -16,7 +16,9 @@
 	TARGET_ARCH = $(shell uname -m)
 endif
 
-ifneq (,$(findstring 64,$(TARGET_ARCH)))
+ifneq (,$(findstring loongarch64,$(TARGET_ARCH)))
+   TARGET_ARCH := loongarch64
+else ifneq (,$(findstring 64,$(TARGET_ARCH)))
    TARGET_ARCH := x86_64
 else ifneq (,$(findstring 86,$(TARGET_ARCH)))
    TARGET_ARCH := x86
--- a/libretro/Makefile.common
+++ b/libretro/Makefile.common
@@ -384,6 +384,7 @@
 	$(EXTDIR)/cpu_features/src/impl_mips_linux_or_android.c \
 	$(EXTDIR)/cpu_features/src/impl_ppc_linux.c \
 	$(EXTDIR)/cpu_features/src/impl_riscv_linux.c \
+	$(EXTDIR)/cpu_features/src/impl_loongarch_linux.c \
 	$(EXTDIR)/cpu_features/src/impl_s390x_linux.c \
 	$(EXTDIR)/cpu_features/src/impl_x86_freebsd.c \
 	$(EXTDIR)/cpu_features/src/impl_x86_linux_or_android.c \
@@ -543,6 +544,7 @@
 	$(COMMONDIR)/OSVersion.cpp \
 	$(COMMONDIR)/MemoryUtil.cpp \
 	$(COMMONDIR)/RiscVCPUDetect.cpp \
+	$(COMMONDIR)/LoongArchCPUDetect.cpp \
 	$(COMMONDIR)/LogReporting.cpp \
 	$(COMMONDIR)/SysError.cpp \
 	$(COMMONDIR)/StringUtils.cpp \
@@ -960,6 +962,19 @@
 		     $(COREDIR)/MIPS/ARM64/Arm64IRRegCache.cpp \
 		     $(COREDIR)/Util/DisArm64.cpp \
 		     $(GPUCOMMONDIR)/VertexDecoderArm64.cpp
+   else ifeq ($(TARGET_ARCH),loongarch64)
+      DYNAFLAGS += -D_ARCH_64
+      SOURCES_CXX += $(COMMONDIR)/LoongArch64Emitter.cpp \
+		     $(COREDIR)/MIPS/LoongArch64/LoongArch64Asm.cpp \
+		     $(COREDIR)/MIPS/LoongArch64/LoongArch64CompALU.cpp \
+		     $(COREDIR)/MIPS/LoongArch64/LoongArch64CompBranch.cpp \
+		     $(COREDIR)/MIPS/LoongArch64/LoongArch64CompFPU.cpp \
+		     $(COREDIR)/MIPS/LoongArch64/LoongArch64CompLoadStore.cpp \
+		     $(COREDIR)/MIPS/LoongArch64/LoongArch64CompSystem.cpp \
+		     $(COREDIR)/MIPS/LoongArch64/LoongArch64CompVec.cpp \
+		     $(COREDIR)/MIPS/LoongArch64/LoongArch64Jit.cpp \
+		     $(COREDIR)/MIPS/LoongArch64/LoongArch64RegCache.cpp \
+		     $(GPUCOMMONDIR)/VertexDecoderLoongArch64.cpp
    else
 	ifneq (,$(findstring msvc,$(platform)))
 	ifeq (,$(findstring x64,$(platform)))
@@ -1008,6 +1023,8 @@
 #UDIS86
 # Compiled and linked even on ARM for now
 
+SOURCES_CXX += $(EXTDIR)/loongarch-disasm.cpp
+
 SOURCES_C += $(EXTDIR)/udis86/decode.c \
 			 $(EXTDIR)/udis86/itab.c \
 			 $(EXTDIR)/udis86/syn-intel.c \
