--- a/PKGBUILD
+++ b/PKGBUILD
@@ -45,9 +45,13 @@ prepare() {
   mv data/keymaps/i386/colemak/{en-latin9,colemak}.map
   # fix euro2 #28213
   patch -Np1 -i ../fix-euro2.patch
+  patch -Np1 -i ../kbd-max-font-size.patch
   autoreconf -if
 }
 
+source+=('kbd-max-font-size.patch')
+b2sums+=('d1338900946f34042c4521bd16ec867f7e65d45295abcda795c2c31e5f0a40fe899c7223a52687c9c9553a71f12a455c9494e2659f2775b7d94d56b367f31b26')
+
 build() {
   cd "${pkgname}"
   ./configure --prefix=/usr --sysconfdir=/etc --datadir=/usr/share/kbd --mandir=/usr/share/man --enable-optional-progs
