diff --git a/PKGBUILD b/PKGBUILD
index 3523ba3..265d43e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,6 +13,11 @@ source=("$pkgbase::git+https://github.com/ROCm/llvm-project#tag=rocm-$pkgver")
 sha256sums=('09cb1afda4d2585b35e12142d4a4a4cab42be6ade7f41be59fd13df86d77ae10')
 options=(staticlibs !lto)
 
+prepare() {
+    cd "$pkgbase"
+    patch -p1 -i "${srcdir}/rocm-llvm-fix-fp16.patch"
+}
+
 build() {
     # Build only minimal debug info to reduce size
     CFLAGS+=' -g1'
@@ -27,10 +32,11 @@ build() {
         -DLLVM_HOST_TRIPLE=$CHOST
         -DLLVM_ENABLE_PROJECTS='llvm;clang;lld;compiler-rt;clang-tools-extra'
         -DCLANG_ENABLE_AMDCLANG=ON
-        -DLLVM_ENABLE_RUNTIMES='libcxx;libcxxabi;libunwind'
+        -DLLVM_ENABLE_RUNTIMES='libcxx;libcxxabi'
+        -DLIBCXXABI_USE_LLVM_UNWINDER=OFF
         -DLIBCXX_ENABLE_STATIC=ON
         -DLIBCXXABI_ENABLE_STATIC=ON
-        -DLLVM_TARGETS_TO_BUILD='AMDGPU;NVPTX;X86'
+        -DLLVM_TARGETS_TO_BUILD='AMDGPU;NVPTX;LoongArch'
         -DCLANG_DEFAULT_LINKER=lld
         -DLLVM_INSTALL_UTILS=ON
         -DLLVM_ENABLE_BINDINGS=OFF
@@ -119,3 +125,6 @@ package_comgr() {
     cd "$pkgbase/amd/comgr"
     install -Dm644 LICENSE.txt "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=("rocm-llvm-fix-fp16.patch")
+sha256sums+=('a0ed008fcc206ce252e0c5ca9898b768429e8fa8de13d2eb4709d1f6a98d08f5')
