--- a/PKGBUILD
+++ b/PKGBUILD
@@ -15,10 +15,14 @@ sha512sums=('64f24f935c4aef35418f10a85f827134d1e5eaa3325dd4dfe60f927f8ea3f0f526c
 
 prepare() {
   cd deepin-pw-check-$pkgver
+  patch -Np1 -i ../0001-fix-build-against-new-version-of-cracklib.patch
   sed -i 's|gcc |gcc ${CFLAGS} ${LDFLAGS} |;/export GOPATH/d' Makefile
   sed -i '/stdio.h/a #include <stdlib.h>' tool/pwd_conf_update.c
 }
 
+source+=("0001-fix-build-against-new-version-of-cracklib.patch::https://github.com/linuxdeepin/deepin-pw-check/pull/44.patch")
+sha512sums+=('a97edc50b232d9d93600983ebeddc66c30377941139fe1ccee2fe60d39a11491bb732334f6289737e3d716a71b11429350d1da0aebe7ff1152c4fca420c8866e')
+
 build() {
   export CGO_CPPFLAGS="${CPPFLAGS}"
   export CGO_CFLAGS="${CFLAGS}"
