--- a/PKGBUILD
+++ b/PKGBUILD
@@ -41,6 +41,8 @@ source=(https://download.qt.io/official_releases/qt/${_pkgver%.*}/${_pkgver}/sin
 sha256sums=('81895fb038a9c3d6c6f698d7611339a189eb45c3d91746c7789b0b77a5981aa3')
 
 build() {
+  patch -d "$_pkgfn" -p1 -i "${srcdir}/qt6-webengine-add-loong64-support.patch"
+
   cmake -B build -S $_pkgfn -G Ninja \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DINSTALL_BINDIR=lib/qt6/bin \
@@ -81,3 +83,6 @@ package_qt6-examples() {
     cp -rn $_dir/* $_base
   done
 }
+
+source+=("qt6-webengine-add-loong64-support.patch")
+sha256sums+=('7f1941b42851e568900a261eab8f15c1a743b987965e98c168d6298ef3b97a92')
