--- a/PKGBUILD
+++ b/PKGBUILD
@@ -41,6 +41,8 @@ source=(https://download.qt.io/official_releases/qt/${_pkgver%.*}/${_pkgver}/sin
 sha256sums=('364fde2d7fa42dd7c9b2ea6db3d462dd54f3869e9fd0ca0a0ca62f750cd8329b')
 
 build() {
+  patch -d "$_pkgfn" -p1 -i "${srcdir}/qt6-webengine-add-loong64-support.patch"
+
   cmake -B build -S $_pkgfn -G Ninja \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DINSTALL_BINDIR=lib/qt6/bin \
@@ -81,3 +83,6 @@ package_qt6-examples() {
     cp -rn $_dir/* $_base
   done
 }
+
+source+=("qt6-webengine-add-loong64-support.patch")
+sha256sums+=('093b17adbbd549241bf8b9c25f8df6587e5126512943b56ac1c2f97de815f2bc')
