diff --git a/PKGBUILD b/PKGBUILD
index b5e0eaa..a8bd39b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,6 +36,8 @@ source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/${pkgver/-/}/si
 sha256sums=('a3f1d257cbb14c6536585ffccf7c203ce7017418e1a0c2ed7c316c20c729c801')
 
 build() {
+  patch -d "$_pkgfn" -p1 -i "${srcdir}/0007-LOONGARCH64-qtwebengine-chromium-add-LoongArch-suppo.patch"
+
   cmake -B build -S $_pkgfn -G Ninja \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DQT_HOST_PATH=/usr \
@@ -77,3 +79,6 @@ package_qt6-examples() {
     cp -rn $_dir/* $_base
   done
 }
+
+source+=("0007-LOONGARCH64-qtwebengine-chromium-add-LoongArch-suppo.patch")
+sha256sums+=('d1cb3d7cc635da18a3023d7b71130dcf65c4ec48dc34c8b191a06d50fe071c87')
