diff --git a/PKGBUILD b/PKGBUILD
index eb984b0..f660e9c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -41,6 +41,7 @@ prepare() {
   sed -i 's|/etc/openbao/openbao.env|/etc/default/openbao|g' .release/linux/package/usr/lib/systemd/system/openbao.service
 
   pushd ui
+  node .yarn/releases/yarn-3.5.0.cjs set version 4.9.2
   yarn install
   popd
 
@@ -56,7 +57,12 @@ build() {
   yarn run build
   popd
 
-  export CGO_ENABLED=0
+  # Go requires external linking for PIE on linux/loong64.
+  export CGO_ENABLED=1
+  export CGO_CPPFLAGS="${CPPFLAGS}"
+  export CGO_CFLAGS="${CFLAGS}"
+  export CGO_CXXFLAGS="${CXXFLAGS}"
+  export CGO_LDFLAGS="${LDFLAGS}"
   export GOPATH="${srcdir}"
 
   # https://openbao.org/docs/contributing/packaging/
@@ -71,7 +77,7 @@ build() {
     -buildmode=pie \
     -mod=readonly \
     -modcacherw \
-    -ldflags "-X github.com/openbao/openbao/version.fullVersion=${pkgver} -X github.com/openbao/openbao/version.GitCommit=${_commit} -X github.com/openbao/openbao/version.BuildDate=${_build_date}" \
+    -ldflags "-linkmode=external -X github.com/openbao/openbao/version.fullVersion=${pkgver} -X github.com/openbao/openbao/version.GitCommit=${_commit} -X github.com/openbao/openbao/version.BuildDate=${_build_date}" \
     -tags 'openbao ui' \
     -o dist/bao \
     .
@@ -95,3 +101,5 @@ package() {
     install -Dm644 "${file}" "${pkgdir}/usr/share/doc/${pkgname}/${file}"
   done
 }
+
+depends+=(glibc)
