--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,20 +27,6 @@ build() {
       -DNO_REGRESSION=ON \
       -DNO_TUTORIALS=ON \
       -DCMAKE_BUILD_TYPE=Release \
-      -DOSD_CUDA_NVCC_FLAGS="-gencode=arch=compute_75,code=sm_75 \
-                             -gencode=arch=compute_80,code=sm_80 \
-                             -gencode=arch=compute_86,code=sm_86 \
-                             -gencode=arch=compute_87,code=sm_87 \
-                             -gencode=arch=compute_88,code=sm_88 \
-                             -gencode=arch=compute_89,code=sm_89 \
-                             -gencode=arch=compute_90,code=sm_90 \
-                             -gencode=arch=compute_100,code=sm_100 \
-                             -gencode=arch=compute_103,code=sm_103 \
-                             -gencode=arch=compute_110,code=sm_110 \
-                             -gencode=arch=compute_120,code=sm_120 \
-                             -gencode=arch=compute_121,code=sm_121 \
-                             -gencode=arch=compute_121,code=compute_121" \
-      -DCUDA_HOST_COMPILER="$NVCC_CCBIN" \
       -DCMAKE_INSTALL_PREFIX=/usr
 
   make -C build
@@ -51,3 +37,5 @@ package() {
 
   DESTDIR="$pkgdir/" make -C build install
 }
+
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda)$'))
