--- a/PKGBUILD
+++ b/PKGBUILD
@@ -18,7 +18,8 @@ depends=(
   systemd
 )
 makedepends=(
-  pandoc-cli
+  #pandoc-cli
+  asciidoctor
   python-build
   python-installer
   python-setuptools
@@ -65,7 +66,8 @@ build() {
   cd $pkgname-$pkgver
   python -m build --wheel --no-isolation
   for manpage in mkosi-addon.1.md  mkosi-initrd.1.md  mkosi-sandbox.1.md  mkosi.1.md  mkosi.news.7.md; do
-    pandoc -t man -s -o "$pkgname/resources/man/${manpage%.md}" "$pkgname/resources/man/${manpage}"
+   # pandoc -t man -s -o "$pkgname/resources/man/${manpage%.md}" "$pkgname/resources/man/${manpage}"
+   asciidoctor "$pkgname/resources/man/${manpage}" -o "$pkgname/resources/man/${manpage%.md}"
   done
 }
 
