diff --git a/PKGBUILD b/PKGBUILD
index 88538e1..76cc6b0 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -44,13 +44,13 @@ prepare() {
 
   # Do not hijack SIGHUP https://gitlab.archlinux.org/archlinux/packaging/packages/openucx/-/issues/3
   patch -Np1 -i ../ucx-conf.patch
+  patch -Np1 -i ../LoongArch64-support.patch
 }
 
 build() {
   local configure_options=(
     --prefix=/usr
     --sysconfdir=/etc
-    --with-cuda=/opt/cuda
     --with-rocm=/opt/rocm
     --with-verbs
     --with-rc
@@ -85,3 +85,11 @@ package() {
   # install the license
   install -vDm 644 $_name-$pkgver/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda)$'))
+source+=(
+  'LoongArch64-support.patch::https://github.com/enkerewpo/ucx/commit/71a2fef6448a1dad37d5ca397cc18efe697853f5.patch'
+)
+b2sums+=(
+  'a46f4241ea82f87802c1f8cef08ad0fb1261127769c6f0811f30dffdc6c1532923d78cdc7f8637dfe7a9dfd8876effed384af4af67b31d994ddf097743f69e10'
+)
