diff --git a/PKGBUILD b/PKGBUILD
index 5d4d1fd..981583a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -55,6 +55,10 @@ case "$CARCH" in
     _vscode_arch=riscv64
     _electron_arch=riscv64
     ;;
+  loong64)
+    _vscode_arch=loong64
+    _electron_arch=loong64
+    ;;
   *)
     # Needed for mksrcinfo
     _vscode_arch=DUMMY
@@ -83,7 +87,11 @@ prepare() {
   patch -p0 -i ../product_json.diff
 
   # OpenVSX signature verification
-  patch -p1 -i ../0009-openvsx-extension-signature.patch
+  # Not supported on loong64
+  #patch -p1 -i ../0009-openvsx-extension-signature.patch
+
+  # Add missing loong64 definition
+  sed -i "/BUILD_TARGETS = \[/a { platform: 'linux', arch: 'loong64' }," build/gulpfile.vscode.js
 
   # Set the commit and build date
   sed -e "s/@COMMIT@/$(git rev-parse HEAD)/" -e "s/@DATE@/$(date -u -Is | sed 's/\+00:00/Z/')/" -i product.json
