--- a/PKGBUILD
+++ b/PKGBUILD
@@ -343,7 +343,7 @@ package_vlc() {
     'vlc-plugin-nfs: for NFS access support'
     'vlc-plugin-notify: for notify support'
     'vlc-plugin-ogg: for OGG mux and demux support'
-    'vlc-plugin-quicksync: for Intel QuickSync H264/H262 codec support'
+#    'vlc-plugin-quicksync: for Intel QuickSync H264/H262 codec support'
     'vlc-plugin-samplerate: for samplerate audio filter support'
     'vlc-plugin-sdl: for SDL codec support'
     'vlc-plugin-sftp: for SFTP access support'
@@ -511,7 +511,7 @@ package_vlc() {
     _pick $pkgname-plugin-pulse usr/lib/vlc/plugins/audio_output/libpulse_plugin.so
     _pick $pkgname-plugin-pulse usr/lib/vlc/plugins/services_discovery/libpulselist_plugin.so
 
-    _pick $pkgname-plugin-quicksync usr/lib/vlc/plugins/codec/libqsv_plugin.so
+#    _pick $pkgname-plugin-quicksync usr/lib/vlc/plugins/codec/libqsv_plugin.so
 
     _pick $pkgname-plugin-samplerate usr/lib/vlc/plugins/audio_filter/libsamplerate_plugin.so
 
@@ -792,16 +792,16 @@ package_vlc() {
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libgrey_yuv_plugin.so
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_10_p010_plugin.so
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_nv12_plugin.so
-    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_rgb_mmx_plugin.so
+#    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_rgb_mmx_plugin.so
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_rgb_plugin.so
-    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_rgb_sse2_plugin.so
-    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_yuy2_mmx_plugin.so
+#    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_rgb_sse2_plugin.so
+#    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_yuy2_mmx_plugin.so
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_yuy2_plugin.so
-    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_yuy2_sse2_plugin.so
+#    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi420_yuy2_sse2_plugin.so
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi422_i420_plugin.so
-    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi422_yuy2_mmx_plugin.so
+#    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi422_yuy2_mmx_plugin.so
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi422_yuy2_plugin.so
-    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi422_yuy2_sse2_plugin.so
+#    _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libi422_yuy2_sse2_plugin.so
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/librv32_plugin.so
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libyuvp_plugin.so
     _pick vlc-plugins-base usr/lib/vlc/plugins/video_chroma/libyuy2_i420_plugin.so
@@ -1916,7 +1916,7 @@ package_vlc-plugins-extra() {
     vlc-plugin-mtp
     vlc-plugin-musepack
     vlc-plugin-nfs
-    vlc-plugin-quicksync
+#    vlc-plugin-quicksync
     vlc-plugin-samplerate
     vlc-plugin-sdl
     vlc-plugin-sftp
@@ -1969,5 +1969,6 @@ package_vlc-plugins-visualization() {
 
   mv -v $pkgname/* "$pkgdir"
 }
+pkgname=(${pkgname[@]/vlc-plugin-quicksync})
 
 # vim: ts=2 sw=2 et:
