diff --git a/PKGBUILD b/PKGBUILD
index 10e3433..e3197b6 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,7 +22,7 @@ validpgpkeys=('462225C3B46F34879FC8496CD605848ED7E69871'  # Daiki Ueno
 
 prepare() {
   cd $pkgname-$pkgver
-  automake
+
   # Do not build libtextstyle, as it depends on libcroco
   # which is now unmaintained and has known security bugs.
   # patch from Fedora	
@@ -31,6 +31,8 @@ prepare() {
 
 build() {
   cd $pkgname-$pkgver
+  LTOFLAGS=""
+  autoreconf --force
   ./configure \
     --prefix=/usr \
     --enable-csharp \
