--- a/PKGBUILD
+++ b/PKGBUILD
@@ -42,7 +42,7 @@ build() {
     --mt=pthread \
     --with-gmp
   make all
-  make -C Olinux-${CARCH} gp-sta
+  make -C Olinux-$(uname -m) gp-sta
 }
 
 check() {
@@ -53,7 +53,7 @@ check() {
 package() {
   cd $pkgname-$pkgver
   make DESTDIR="$pkgdir" install
-  make DESTDIR="$pkgdir" -C Olinux-${CARCH} install-bin-sta
+  make DESTDIR="$pkgdir" -C Olinux-$(uname -m) install-bin-sta
   ln -sf gp.1.gz "$pkgdir"/usr/share/man/man1/pari.1
   chrpath -d "$pkgdir"/usr/bin/gp-*
 }
