--- a/PKGBUILD
+++ b/PKGBUILD
@@ -67,3 +67,12 @@ package() {
   make DESTDIR="$pkgdir" install
   install -Dm644 -t "$pkgdir/usr/share/licenses/$pkgname/" COPYING
 }
+
+source+=("$pkgname-fix-instruction-detection.patch::https://github.com/cacalabs/libcaca/pull/76.patch")
+b2sums+=('1cfd28a3777cec6cb8f2bab8ae2f6d9496452e40dd7f0f7a7cf67ebef1a4ac48390e63ce464b1283207545fe8fe050f9789cf39202b513d3d22291dc84521f10')
+
+prepare() {
+  cd $pkgname
+  patch -Np1 -i ../$pkgname-fix-instruction-detection.patch
+  autoreconf -fi
+}
