diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93793d22790b2..afecd75138e0a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -931,12 +931,13 @@

 include(cmake/Dependencies.cmake)

+# aotriton requires CUDA, only available on x86_64 and aarch64
 cmake_dependent_option(
   USE_FLASH_ATTENTION
   "Whether to build the flash_attention kernel for scaled dot product attention.\
   Will be disabled if not supported by the platform"
   ON
-  "(USE_CUDA AND NOT MSVC) OR USE_ROCM"
+  "CPU_INTEL OR CPU_AARCH64;(USE_CUDA AND NOT MSVC) OR USE_ROCM"
   OFF)


@@ -883,7 +884,7 @@ cmake_dependent_option(
   USE_MEM_EFF_ATTENTION
   "Enable memory-efficient attention for scaled dot product attention.\
   Will be disabled if not supported by the platform" ON
-  "USE_CUDA OR USE_ROCM" OFF)
+  "CPU_INTEL OR CPU_AARCH64;USE_CUDA OR USE_ROCM" OFF)
 
 #
 # Cannot be put into Dependencies.cmake due circular dependency:
