--- a/PKGBUILD
+++ b/PKGBUILD
@@ -217,6 +217,11 @@ prepare() {
 
   git -c protocol.file.allow=always submodule update --init --recursive
 
+  # Fix for loong64
+  patch -Np1 -d third_party/sleef -i "${srcdir}"/sleef_loong64.patch
+  patch -Np1 -d third_party/cpuinfo -i "${srcdir}"/cpuinfo_loong64.patch
+  patch -Np1 -i "${srcdir}"/disable-aotriton.patch
+
   # Fix cmake prefix path (FS#78665)
   patch -Np1 -i "${srcdir}"/fix_cmake_prefix_path.patch
 
@@ -269,7 +274,7 @@ _prepare() {
   # Check tools/setup_helpers/cmake.py, setup.py and CMakeLists.txt for a list of flags that can be set via env vars.
   export BUILD_TEST=OFF  # do not build tests
   export ATEN_NO_TEST=ON  # do not build ATen tests
-  export USE_MKLDNN=ON
+  # export USE_MKLDNN=ON
   export BUILD_CUSTOM_PROTOBUF=OFF
   export USE_GFLAGS=ON
   export USE_GLOG=ON
@@ -278,10 +283,11 @@ _prepare() {
   export USE_MAGMA=ON
   # export USE_SYSTEM_LIBS=ON  # experimental, not all libs present in repos
   # USE_SYSTEM_ONNX=ON does not work and onnx itself should be removed from pytorch: https://github.com/pytorch/pytorch/issues/166546#issuecomment-3463370459
-  export USE_NCCL=ON
-  export USE_SYSTEM_NCCL=ON
+  # export USE_NCCL=ON
+  # export USE_SYSTEM_NCCL=ON
   export USE_SYSTEM_PYBIND11=ON
   export USE_SYSTEM_EIGEN_INSTALL=ON
+  : <<COMMENT_SEPARATOR
   export USE_GOLD_LINKER=ON
   export NCCL_VERSION=$(pkg-config nccl --modversion)
   export NCCL_VER_CODE=$(sed -n 's/^#define NCCL_VERSION_CODE\s*\(.*\).*/\1/p' /usr/include/nccl.h)
@@ -302,7 +308,7 @@ _prepare() {
   # https://github.com/pytorch/pytorch/blob/f9074c7332c3dfd43fe39e8733ec98f7f29b3e61/torch/utils/cpp_extension.py#L2417-L2420
   # (note that 8.8 is not supported)
   export TORCH_CUDA_ARCH_LIST="7.5 8.0 8.6 8.7 8.9 9.0 10.0 10.3 11.0 12.0 12.1"
-
+COMMENT_SEPARATOR
   export ROCM_PATH=/opt/rocm
   export HIP_ROOT_DIR=/opt/rocm
   # gfx950 lacks support for 128 bit atomics
@@ -314,7 +320,7 @@ _prepare() {
   export USE_FBGEMM_GENAI=OFF
 
   # Compile source code for supported GPU archs in parallel (but using too many jobs is not helpful)
-  export HIPCC_COMPILE_FLAGS_APPEND="-parallel-jobs=4 --gcc-install-dir=$(dirname $($CC -print-libgcc-file-name))"
+  export HIPCC_COMPILE_FLAGS_APPEND="-parallel-jobs=4"
   export HIPCC_LINK_FLAGS_APPEND="-parallel-jobs=4"
 
   export AOTRITON_INSTALLED_PREFIX=/usr
@@ -329,23 +335,24 @@ _prepare() {
 
 build() {
   cd "${srcdir}/${_pkgname}"
-  echo "Building without cuda or rocm and without non-x86-64 optimizations"
+  echo "Building without cuda or rocm and without loong64 optimizations"
   _prepare
   export USE_CUDA=0
   export USE_CUDNN=0
   export USE_ROCM=0
-  echo "add_definitions(-march=x86-64)" >> cmake/MiscCheck.cmake
+  echo "add_definitions(-march=loongarch64)" >> cmake/MiscCheck.cmake
   python -m build --wheel --no-isolation
 
   cd "${srcdir}/${_pkgname}-opt"
-  echo "Building without cuda or rocm and with non-x86-64 optimizations"
+  echo "Building without cuda or rocm and with loong64 optimizations"
   _prepare
   export USE_CUDA=0
   export USE_CUDNN=0
   export USE_ROCM=0
-  echo "add_definitions(-march=x86-64-v3)" >> cmake/MiscCheck.cmake
+  echo "add_definitions(-march=la464)" >> cmake/MiscCheck.cmake
   python -m build --wheel --no-isolation
 
+  : <<COMMENT_SEPARATOR
   cd "${srcdir}/${_pkgname}-cuda"
   echo "Building with cuda and without non-x86-64 optimizations"
   _prepare
@@ -366,9 +373,10 @@ build() {
   _prepare
   echo "add_definitions(-march=x86-64-v3)" >> cmake/MiscCheck.cmake
   python -m build --wheel --no-isolation
+COMMENT_SEPARATOR
 
   cd "${srcdir}/${_pkgname}-rocm"
-  echo "Building with rocm and without non-x86-64 optimizations"
+  echo "Building with rocm and without loong64 optimizations"
   # -fcf-protection is not supported by HIP, see
   # https://rocm.docs.amd.com/projects/llvm-project/en/latest/reference/rocmcc.html#support-status-of-other-clang-options
   CXXFLAGS+=" -fcf-protection=none"
@@ -377,19 +385,21 @@ build() {
   export USE_CUDNN=0
   export USE_ROCM=1
   export MAGMA_HOME=/opt/rocm
-  echo "add_definitions(-march=x86-64)" >> cmake/MiscCheck.cmake
+  export MAX_JOBS=16
+  echo "add_definitions(-march=loongarch64)" >> cmake/MiscCheck.cmake
   # Conversion of CUDA to ROCm source files
   python tools/amd_build/build_amd.py
   python -m build --wheel --no-isolation
 
   cd "${srcdir}/${_pkgname}-opt-rocm"
-  echo "Building with rocm and with non-x86-64 optimizations"
+  echo "Building with rocm and with loong64 optimizations"
   _prepare
   export USE_CUDA=0
   export USE_CUDNN=0
   export USE_ROCM=1
   export MAGMA_HOME=/opt/rocm
-  echo "add_definitions(-march=x86-64-v3)" >> cmake/MiscCheck.cmake
+  export MAX_JOBS=16
+  echo "add_definitions(-march=la464)" >> cmake/MiscCheck.cmake
   # Conversion of CUDA to ROCm source files
   python tools/amd_build/build_amd.py
   python -m build --wheel --no-isolation
@@ -430,7 +440,7 @@ package_python-pytorch() {
 }
 
 package_python-pytorch-opt() {
-  pkgdesc="${_pkgdesc} (with AVX2 CPU optimizations)"
+  pkgdesc="${_pkgdesc} (with LASX CPU optimizations)"
   conflicts=(python-pytorch)
   provides=(python-pytorch=${pkgver})
 
@@ -460,7 +470,7 @@ package_python-pytorch-opt-cuda() {
 
 package_python-pytorch-rocm() {
   pkgdesc="${_pkgdesc} (with ROCm)"
-  depends+=(rocm-hip-sdk hipblaslt roctracer miopen-hip magma-hip onednn python-triton python-aotriton)
+  depends+=(rocm-hip-sdk hipblaslt roctracer miopen-hip magma-hip onednn)
   conflicts=(python-pytorch)
   provides=(python-pytorch=${pkgver})
 
@@ -469,8 +479,8 @@ package_python-pytorch-rocm() {
 }
 
 package_python-pytorch-opt-rocm() {
-  pkgdesc="${_pkgdesc} (with ROCm and AVX2 CPU optimizations)"
-  depends+=(rocm-hip-sdk hipblaslt roctracer miopen-hip magma-hip onednn python-triton python-aotriton)
+  pkgdesc="${_pkgdesc} (with ROCm and LASX CPU optimizations)"
+  depends+=(rocm-hip-sdk hipblaslt roctracer miopen-hip magma-hip onednn)
   conflicts=(python-pytorch)
   provides=(python-pytorch=${pkgver} python-pytorch-rocm=${pkgver})
 
@@ -478,4 +488,20 @@ package_python-pytorch-opt-rocm() {
   _package
 }
 
+depends=(${depends[@]/intel-oneapi-mkl})
+makedepends=(${makedepends[@]/magma-cuda})
+makedepends=(${makedepends[@]/cuda})
+makedepends=(${makedepends[@]/nccl})
+makedepends=(${makedepends[@]/cudnn})
+makedepends=(${makedepends[@]/python-triton})
+makedepends=(${makedepends[@]/python-aotriton})
+pkgname=(${pkgname[@]/"${pkgbase}-cuda"})
+pkgname=(${pkgname[@]/"${pkgbase}-opt-cuda"})
+source+=("disable-aotriton.patch"
+         "sleef_loong64.patch"
+         "cpuinfo_loong64.patch::https://raw.githubusercontent.com/loong64/pytorch/refs/heads/main/cpuinfo/cpuinfo_loong64.patch")
+b2sums+=('007a8ffef259856b57d6714e2535c579c2c00932e324eb51c505a33ae1cbe46c6223aacf89823f0e4faf0ea6a4dd595cd21f6a57ab6eccb219f6e690a1500c32'
+         '5a06663c88476341c169dadf8a2adda13fd264fda9468bcd1e3c3d120c8ba3949e9eb6396faf824afae4b2557665f5dd4f09ccd162f08dc022cf2076fe0620fe'
+         'b56ac09123c229431d82ef47415ef97600ceedd34576058efe3feb511fecd8ee65944f3556995d7b21c20df9d41cc1bd8768f5208831a2817a0569f0d9346824')
+
 # vim:set ts=2 sw=2 et:
