diff --git a/PKGBUILD b/PKGBUILD
index 0f051ed..6d02c4e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -475,6 +475,9 @@ prepare() {
   echo "Putting together electron sources"
   # Generate gclient gn args file and prepare-electron-source-tree.sh
   python makepkg-source-roller.py generate electron/DEPS $pkgname
+  # Do not use the script to fetch esbuild since google doesn't distribute it
+  # The esbuild is added manually
+  sed -i '/esbuild/d' prepare-electron-source-tree.sh
   rbash prepare-electron-source-tree.sh "$CARCH"
   mv electron src/electron
 
@@ -488,12 +491,50 @@ prepare() {
     -s src/third_party/skia --header src/skia/ext/skia_commit_hash.h
   src/build/util/lastchange.py \
     -s src/third_party/dawn --revision src/gpu/webgpu/DAWN_VERSION
-  src/tools/update_pgo_profiles.py --target=linux update \
-    --gs-url-base=chromium-optimization-profiles/pgo_profiles
+  #src/tools/update_pgo_profiles.py --target=linux update \
+  #  --gs-url-base=chromium-optimization-profiles/pgo_profiles
+
+  # Link to system tools required by the build in advance
+  pushd src
+  mkdir -p third_party/node/linux/node-linux-x64/bin
+  ln -sfn /usr/bin/node third_party/node/linux/node-linux-x64/bin/
+  mkdir -p third_party/jdk/current/bin
+  ln -sfn /usr/bin/java third_party/jdk/current/bin/
+  ln -sfn /usr/bin/clang-format buildtools/linux64
+  popd
+
+  pushd src/third_party/node/
+  sed -i -e 's/@rollup/rollup/' -e "s/'wasm-node',//" node_modules.py
+  #local _rollup_ver="$(jq -r .dependencies.\"@rollup/wasm-node\" package.json)"
+  local _rollup_ver="4.32.0" # Known version that supports loong64
+  jq ".dependencies.rollup=\"$_rollup_ver\" | .dependencies.\"@rollup/rollup-linux-loongarch64-gnu\"=\"$_rollup_ver\"" package.json > package.json.new
+  mv package.json{.new,}
+  popd
+
+  export SENTRYCLI_SKIP_DOWNLOAD=1
 
   # https://gitlab.archlinux.org/archlinux/packaging/packages/electron32/-/issues/1
   src/third_party/node/update_npm_deps
 
+  pushd src/third_party/devtools-frontend/src
+  sed -i -e 's/@rollup/rollup/' -e "s/'wasm-node',//" scripts/devtools_paths.py
+  #local _rollup_ver="$(jq -r .devDependencies.\"@rollup/wasm-node\" package.json)"
+  jq ".devDependencies.rollup=\"$_rollup_ver\" | .devDependencies.\"@rollup/rollup-linux-loongarch64-gnu\"=\"$_rollup_ver\""  package.json > package.json.new
+  mv package.json{.new,}
+  # Chromium hosts a custom registry at https://npm.skia.org/chrome-devtools/
+  # and rejects some packages:
+  # Package fs-extra with version 11.3.0 was created 108h0m0s time ago. This is less than 1 week and so failed the audit.
+  sed -i /registry/d .npmrc
+  # Replace direct invocation of wasm rollup
+  sed -i 's\@rollup/wasm-node\rollup\' \
+    inspector_overlay/BUILD.gn \
+    front_end/models/live-metrics/web-vitals-injected/BUILD.gn \
+    front_end/Images/BUILD.gn \
+    front_end/panels/recorder/injected/BUILD.gn \
+    scripts/build/ninja/bundle.gni
+  popd
+  python src/third_party/devtools-frontend/src/scripts/deps/manage_node_deps.py
+
   src/electron/script/apply_all_patches.py \
       src/electron/patches/config.json
 
@@ -502,7 +543,8 @@ prepare() {
 
   pushd src
   pushd electron
-  yarn install --frozen-lockfile
+  # Skip build scripts for native test addons only needed for the Electron test suite and fail on loong64.
+  yarn install --frozen-lockfile --mode=skip-build
   popd
 
   echo "Applying local patches..."
@@ -569,10 +611,10 @@ prepare() {
            third_party/jdk/current/bin \
            third_party/gperf/cipd/bin
 
-  ln -s /usr/bin/node third_party/node/linux/node-linux-x64/bin/
-  ln -s /usr/bin/rustc third_party/rust-toolchain/bin/
-  ln -s /usr/bin/java third_party/jdk/current/bin/
-  ln -s /usr/bin/gperf third_party/gperf/cipd/bin/
+  ln -sf /usr/bin/node third_party/node/linux/node-linux-x64/bin/
+  ln -sf /usr/bin/rustc third_party/rust-toolchain/bin/
+  ln -sf /usr/bin/java third_party/jdk/current/bin/
+  ln -sf /usr/bin/gperf third_party/gperf/cipd/bin/
 
   # Electron specific fixes
   #  fix: use GTK UI theme for Linux message boxes #52238
@@ -581,6 +623,20 @@ prepare() {
   patch -Np1 -i "${srcdir}/jinja-python-3.10.patch" -d "third_party/electron_node/tools/inspector_protocol/jinja2"
   patch -Np1 -i "${srcdir}/use-system-libraries-in-node.patch"
 
+  # Add loong64 support for chromium
+  patch -Np1 -i "${srcdir}/chromium-loong64-support.patch"
+  patch -Np1 -i "${srcdir}/allow-sched_getaffinity-in-seccomp-for-loong64.patch"
+  patch -Np1 -i "${srcdir}/libyuv-loongarch-fix-row_lsx.cc-and-row_lasx.cc.patch"
+  # Add loong64 support for electron
+  patch -Np1 -d electron -i "${srcdir}/electron_runtime_api_dalagate-add-loongarch64-support.patch"
+
+  pushd ${srcdir}
+  # Add esbuild binary manually
+  npm install @esbuild/linux-loong64@0.25.1
+  install -Dm755 node_modules/@esbuild/linux-loong64/bin/esbuild src/third_party/devtools-frontend/src/third_party/esbuild/esbuild
+  rm -r node_modules # Cleaning
+  popd
+
   # Allow building against system libraries in official builds
   echo "Patching Chromium for using system libraries..."
   sed -i 's/OFFICIAL_BUILD/GOOGLE_CHROME_BUILD/' \
@@ -704,6 +760,14 @@ build() {
     CXXFLAGS="${CXXFLAGS/-march=*([^ ]) }"
   fi
 
+  # Disable xnnpack on loong64
+  # Make sure PGO is disabled on loong64
+  _flags+=(
+    build_litert_with_xnnpack=false
+    build_tflite_with_xnnpack=false
+    chrome_pgo_phase=0
+  )
+
   export CHROMIUM_BUILDTOOLS_PATH="${PWD}/buildtools"
   gn gen out/Release \
       --args="import(\"//electron/build/args/release.gn\") ${_flags[*]}"
@@ -728,3 +792,13 @@ package() {
   install -Dm644 src/electron/default_app/icon.png \
           "${pkgdir}/usr/share/pixmaps/${pkgname}.png"  # hicolor has no 1024x1024
 }
+
+makedepends+=(jq)
+source+=("chromium-loong64-support.patch"
+         "electron_runtime_api_dalagate-add-loongarch64-support.patch"
+         "allow-sched_getaffinity-in-seccomp-for-loong64.patch"
+         "libyuv-loongarch-fix-row_lsx.cc-and-row_lasx.cc.patch")
+sha256sums+=('941b1b63ae4e4432c9b9b6a3fea9a6f5352235c51bf7f980e5e249023fe84866'
+             'de2cad8f1c8dced39a2f69e2908b173e7a27321d30d35b25acdc8f9ba3e57b50'
+             'b48d40e93f020b5e8861f1f320437984d8f7d62c568ad1995af78e49e88de7a9'
+             '63f6dff26b20c178ef86f9220c9f72c4da39059b9bad8c4f551c605515626300')
