diff --git a/PKGBUILD b/PKGBUILD
index c2e4ca4..ccdbad2 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,12 +14,15 @@ _commit=464c51a0353c71f08fe45f683d6a97a638d47833
 source=(git+${url}#commit=${_commit})
 sha512sums=(SKIP)
 
+: <<'COMMENT_SEPARATOR'
 pkgver() {
   cd ${pkgname}
   printf "r%s+%s" "$(git rev-list --count HEAD)" "$(git rev-parse --short HEAD)"
 }
+COMMENT_SEPARATOR
 
 prepare() {
+  patch -p3 -i "${srcdir}/0102-Fix-libyuv-build-with-LSX-LASX.patch"
   sed -i 's|yuvconvert ${JPEG_LIBRARY}|${ly_lib_shared} ${JPEG_LIBRARY}|' ${pkgname}/CMakeLists.txt
 }
 
@@ -33,3 +36,6 @@ package() {
   make -C build DESTDIR="${pkgdir}" install
   install -Dm644 ${pkgname}/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+source+=("0102-Fix-libyuv-build-with-LSX-LASX.patch")
+sha512sums+=('1a1917fcbe2af02b48e2a839f6fdb8e9fe27eea2301fde6e6dd9506f79078d5e78b6c17eafbd8cac147e4b894680c6838306a14f92904a09d34866aa88307b2d')
