--- a/qtwebengine/src/3rdparty/chromium/third_party/blink/renderer/platform/BUILD.gn
+++ b/qtwebengine/src/3rdparty/chromium/third_party/blink/renderer/platform/BUILD.gn
@@ -1593,10 +1593,17 @@
   }
 
   if (current_cpu == "loong64") {
-    cflags = [
-      "-mlsx",
-      "-flax-vector-conversions=all",
-    ]
+    if (is_clang) {
+      cflags = [
+        "-mlsx",
+        "-flax-vector-conversions=all",
+      ]
+    } else {
+      cflags = [
+        "-mlsx",
+        "-flax-vector-conversions",
+      ]
+    }
   }
 
   if (is_apple) {
