--- a/PKGBUILD
+++ b/PKGBUILD
@@ -95,6 +95,13 @@ prepare() {
   # Bump chromium to head of stable branch
   cd src/3rdparty
   [[ -n $_chromium ]] && git checkout $_chromium || true
+  pushd "${srcdir}/${_pkgfn}"
+  # Fix build for loong64
+  patch -Np2 -i "$srcdir/add-loong64-support.patch"
+  patch -Np1 -i "$srcdir/fix-para-type-in-delay.cc.patch" -d "src/3rdparty/chromium"
+  # Temporarily switch to mold to workaround with binutils's bfd's bug
+  export LDFLAGS="${LDFLAGS} -fuse-ld=mold"
+  popd
 }
 
 build() {
@@ -116,3 +123,10 @@ package() {
 
   install -Dm644 "$srcdir"/${_pkgfn}/src/3rdparty/chromium/LICENSE "$pkgdir"/usr/share/licenses/${pkgname}/LICENSE.chromium
 }
+
+source+=("add-loong64-support.patch"
+         "fix-para-type-in-delay.cc.patch")
+sha256sums+=('8592cef263fee7a8d806b2b16fb07e7296593b6a0b3c35174fd3a5f53f29bc94'
+             '9dddd42413ce4a7b877f5778402e360cab3310a316f5209b0bcefa0d1d2d27fe')
+# Temporarily switch to mold to workaround with binutils's bfd's bug
+makedepends+=(mold)
