--- a/PKGBUILD
+++ b/PKGBUILD
@@ -91,6 +91,22 @@ prepare() {
 
 # Disable FFmpeg allow lists https://bugreports.qt.io/browse/QTBUG-132762
   patch -d src/3rdparty -p1 < ../disable-ffmpeg-allow-lists.patch
+  cd "${srcdir}/${_pkgfn}"
+  # Fix libyuv's lsx and lsax
+  patch -d src/3rdparty -p1 -i "$srcdir/0001-fix-libyuv-unified-source-and-LoongArch-SIMD-build.patch"
+  # Fix build for loong64 (Chen Jiajie's patch)
+  patch -p2 -i "$srcdir/0002-add-loong64-support.patch"
+  # Fix build with gcc on loong64
+  patch -p2 -i "$srcdir/0003-blink-renderer-platform-fix-gcc-build.patch"
+
+  # Add `-mcmodel=medium` to CFLAGS etc.
+  # to avoid `relocation R_LARCH_B26 overflow`
+  export CFLAGS="${CFLAGS} -mcmodel=medium"
+  export CXXFLAGS="${CXXFLAGS} -mcmodel=medium"
+  export RUSTFLAGS="${RUSTFLAGS} -C code-model=medium"
+  # Temporarily switch to mold to workaround with binutils's bfd's bug
+  # that will still fail with `R_LARCH_B26 overflow`
+  export LDFLAGS="${LDFLAGS} -fuse-ld=mold"
 }
 
 build() {
@@ -112,3 +128,12 @@ package() {
 
   install -Dm644 "$srcdir"/${_pkgfn}/src/3rdparty/chromium/LICENSE "$pkgdir"/usr/share/licenses/${pkgname}/LICENSE.chromium
 }
+
+source+=("0001-fix-libyuv-unified-source-and-LoongArch-SIMD-build.patch"
+         "0002-add-loong64-support.patch"
+         "0003-blink-renderer-platform-fix-gcc-build.patch")
+sha256sums+=("d971477d4920a51de1ef9ffb96d1c755bf48d7ba6f3fbf869da2131f083d9f40"
+             "77e8dcf1519b30f801e5783b9604b2147749034c3f10a65e40aa9038cd2d3bcd"
+             "649443ce93aacaa10371b9632a8eedc8e9f8c5f15dbd2ecc3079d06d7cd89478")
+# Temporarily switch to mold to workaround with binutils's bfd's bug
+makedepends+=(mold)
