--- a/PKGBUILD
+++ b/PKGBUILD
@@ -191,6 +191,7 @@ build() {
     [[ "${i}" == "i386-pc" ]] && _configure_options+=(--enable-efiemu --with-platform="pc" --target="i386")
     [[ "${i}" == "i386-efi" ]] && _configure_options+=(--disable-efiemu --with-platform="efi" --target="i386")
     [[ "${i}" == "x86_64-efi" ]] && _configure_options+=(--with-platform="efi" --target="x86_64")
+    [[ "${i}" == "loong64-efi" ]] && _configure_options+=(--with-platform="efi" --target="loongarch64")
     ./configure PACKAGE_VERSION="${epoch}:${pkgver}-${pkgrel}" \
                 ${_configure_options[@]}
     echo "Build language and doc files only for most common variant..."
@@ -223,3 +224,5 @@ package() {
   install -D -m0644 "${srcdir}/grub.default" "${pkgdir}/etc/default/grub"
   sed -e "s/%PKGVER%/${epoch}:${pkgver}-${pkgrel}/" < "${srcdir}/sbat.csv" > "${pkgdir}/usr/share/grub/sbat.csv"
 }
+
+_platform=(loong64-efi)
