diff --git a/PKGBUILD b/PKGBUILD
index df136d1..c4f5817 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,7 +13,6 @@ pkgname=(
   gst-libav
   gst-plugin-gtk
   gst-plugin-msdk
-  gst-plugin-onnx
   gst-plugin-opencv
   gst-plugin-qml6
   gst-plugin-qmlgl
@@ -128,7 +127,6 @@ makedepends=(
   nasm
   neon
   nettle
-  onnxruntime
   openal
   opencore-amr
   opencv
@@ -160,7 +158,6 @@ makedepends=(
   spandsp
   speex
   svt-av1
-  svt-hevc
   systemd-libs
   taglib
   twolame
@@ -226,10 +223,12 @@ build() {
     -D gst-plugins-bad:isac=disabled
     -D gst-plugins-bad:magicleap=disabled
     -D gst-plugins-bad:mfx_api=oneVPL
+    -D gst-plugins-bad:onnx=disabled
     -D gst-plugins-bad:openh264=disabled
     -D gst-plugins-bad:openni2=disabled
     -D gst-plugins-bad:opensles=disabled
     -D gst-plugins-bad:qt6d3d11=disabled
+    -D gst-plugins-bad:svthevcenc=disabled
     -D gst-plugins-bad:tinyalsa=disabled
     -D gst-plugins-bad:voaacenc=disabled
     -D gst-plugins-bad:voamrwbenc=disabled
@@ -738,7 +737,6 @@ package_gst-plugins-bad() {
     spandsp
     srt
     svt-av1
-    svt-hevc
     webrtc-audio-processing-1
     wildmidi
     x265
@@ -801,7 +799,6 @@ package_gst-plugins-bad() {
     usr/lib/gstreamer-1.0/libgstsrt.so
     usr/lib/gstreamer-1.0/libgstsrtp.so
     usr/lib/gstreamer-1.0/libgstsvtav1.so
-    usr/lib/gstreamer-1.0/libgstsvthevcenc.so
     usr/lib/gstreamer-1.0/libgstteletext.so
     usr/lib/gstreamer-1.0/libgsttimecode.so
     usr/lib/gstreamer-1.0/libgstttmlsubs.so
@@ -901,6 +898,7 @@ package_gst-plugin-msdk() {
   ); _install
 }
 
+: <<COMMENT_SEPARATOR
 package_gst-plugin-onnx() {
   pkgdesc+=" - onnx plugin"
   depends=(
@@ -917,6 +915,7 @@ package_gst-plugin-onnx() {
     usr/lib/gstreamer-1.0/libgstonnx.so
   ); _install
 }
+COMMENT_SEPARATOR
 
 package_gst-plugin-opencv() {
   pkgdesc+=" - opencv plugin"
