diff --git a/PKGBUILD b/PKGBUILD
index df136d1..f785da2 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -160,7 +160,6 @@ makedepends=(
   spandsp
   speex
   svt-av1
-  svt-hevc
   systemd-libs
   taglib
   twolame
@@ -230,6 +229,7 @@ build() {
     -D gst-plugins-bad:openni2=disabled
     -D gst-plugins-bad:opensles=disabled
     -D gst-plugins-bad:qt6d3d11=disabled
+    -D gst-plugins-bad:svthevcenc=disabled
     -D gst-plugins-bad:tinyalsa=disabled
     -D gst-plugins-bad:voaacenc=disabled
     -D gst-plugins-bad:voamrwbenc=disabled
@@ -738,7 +738,6 @@ package_gst-plugins-bad() {
     spandsp
     srt
     svt-av1
-    svt-hevc
     webrtc-audio-processing-1
     wildmidi
     x265
@@ -801,7 +800,6 @@ package_gst-plugins-bad() {
     usr/lib/gstreamer-1.0/libgstsrt.so
     usr/lib/gstreamer-1.0/libgstsrtp.so
     usr/lib/gstreamer-1.0/libgstsvtav1.so
-    usr/lib/gstreamer-1.0/libgstsvthevcenc.so
     usr/lib/gstreamer-1.0/libgstteletext.so
     usr/lib/gstreamer-1.0/libgsttimecode.so
     usr/lib/gstreamer-1.0/libgstttmlsubs.so
