--- a/PKGBUILD
+++ b/PKGBUILD
@@ -43,7 +43,7 @@ build() {
 check() {
   cd ${pkgbase}
   local python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
-  PYTHONPATH="$PWD/bin/lib.linux-$CARCH-cpython-${python_version}" python -m unittest discover python "*_test.py"
+  PYTHONPATH="$PWD/bin/lib.linux-`uname -m`-cpython-${python_version}" python -m unittest discover python "*_test.py"
   cd build
   ctest --output-on-failure --stop-on-failure -j$(nproc)
 }
