diff --git a/PKGBUILD b/PKGBUILD
index 505359b..ba6f4bb 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -52,6 +52,8 @@ b2sums=('547e757cb3ecfaa3a43d403f0bbed8d890dc6a8d9be59be1e4105d450a4ffc5892117ac
 
 prepare() {
   cd $pkgname
+  # generate bindgen at build time
+  cargo add mysqlclient-sys@0.4.1 --features buildtime_bindgen
   # download dependencies
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 
@@ -96,3 +98,10 @@ package() {
   # binary
   install -vDm755 -t "$pkgdir/usr/bin" target/release*/$pkgname
 }
+
+depends_loong64=(
+  mariadb-libs
+  clang
+  mariadb-libs
+  postgresql-libs
+)
