diff --git a/PKGBUILD b/PKGBUILD
index 2569c7f..f644145 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -30,7 +30,7 @@ makedepends=(
   git
   libdrm
   ninja
-  pandoc
+  #pandoc
   python
   python-docutils
   systemd
@@ -88,6 +88,7 @@ build() {
         -DCMAKE_INSTALL_LIBEXECDIR=/usr/lib/rdma
         -DCMAKE_INSTALL_SYSCONFDIR=/etc
         -DCMAKE_INSTALL_PERLDIR=/usr/share/perl5/vendor_perl
+        -DNO_MAN_PAGES=1 # need pandoc
     )
     cmake "${cmake_options[@]}"
     cmake --build build
