--- a/PKGBUILD
+++ b/PKGBUILD
@@ -56,6 +56,14 @@ prepare() {
   sed -i '/m64=/s/lib64/lib/' gcc/config/i386/t-linux64
   sed -i '/lp64=/s/lib64/lib/' gcc/config/aarch64/t-aarch64-linux
 
+  # Patches for loong64
+  # Fix libdir
+  patch -Np1 -i "$srcdir/gcc-lib64-lib.patch"
+  # Apply upstream fix for LoongArch ICE in highway-1.3.0 testsuite
+  git cherry-pick -n 1276b391b4b068655b1d4511d51421302c1a7849
+  # Apply upstream fix to extend the narrower shift amount before broadcasting it
+  git cherry-pick -n 1139fdadc85a0b602a2833e6b7f2cfe2a8a90cdb
+
   mkdir -p "$srcdir/gcc-build"
 }
 
@@ -83,6 +91,7 @@ build() {
       --enable-linker-build-id
       --enable-lto
       --disable-multilib
+      --disable-multiarch
       --enable-plugin
       --enable-shared
       --enable-threads=posix
@@ -132,8 +141,8 @@ package_gcc15-libs() {
 
   cd gcc-build
   make -C $CHOST/libgcc DESTDIR="$pkgdir" install-shared
-  mv "${pkgdir}/${_libdir}"/../lib/* "${pkgdir}/${_libdir}"
-  rmdir "${pkgdir}/${_libdir}"/../lib
+#  mv "${pkgdir}/${_libdir}"/../lib/* "${pkgdir}/${_libdir}"
+#  rmdir "${pkgdir}/${_libdir}"/../lib
   rm -f "$pkgdir/$_libdir/libgcc_eh.a"
 
   for lib in libasan.so \
@@ -175,8 +184,8 @@ package_gcc15() {
   install -m755 -t "$pkgdir/${_libdir}/" gcc/{cc1,cc1plus,collect2,lto1,gcov{,-tool}}
 
   make -C $CHOST/libgcc DESTDIR="$pkgdir" install
-  rm -f "${pkgdir}/${_libdir}"/../lib/libgcc_s.so*
-  rmdir "${pkgdir}/${_libdir}"/../lib
+  rm -f "${pkgdir}/${_libdir}"/libgcc_s.so*
+#  rmdir "${pkgdir}/${_libdir}"/../lib
 
   make -C $CHOST/libstdc++-v3/src DESTDIR="$pkgdir" install
   make -C $CHOST/libstdc++-v3/include DESTDIR="$pkgdir" install
@@ -210,7 +219,7 @@ package_gcc15() {
   # create cc-rs compatible symlinks
   # https://github.com/rust-lang/cc-rs/blob/1.0.73/src/lib.rs#L2578-L2581
   for binary in {c++,g++,gcc,gcc-ar,gcc-nm,gcc-ranlib}; do
-    ln -s /usr/bin/${binary} "${pkgdir}"/usr/bin/x86_64-linux-gnu-${binary}-15
+    ln -s /usr/bin/${binary} "${pkgdir}"/usr/bin/$(uname -m)-linux-gnu-${binary}-15
   done
 
   # POSIX conformance launcher scripts for c89 and c99
@@ -248,3 +257,6 @@ package_gcc15-fortran() {
   ln -s /usr/share/licenses/gcc-libs/RUNTIME.LIBRARY.EXCEPTION \
     "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+source+=("gcc-lib64-lib.patch")
+sha256sums+=('d09cbb949364442a78e886b8f55593f07ad17f1e5369ecc83d6c6826015ba22e')
