diff --git a/PKGBUILD b/PKGBUILD
index 35dbec1..d25761d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -135,6 +135,13 @@ prepare() {
     done
     popd
 
+    # Enable loong64 support in external/firebird
+    # Copy `0001-firebird-la64-support.patch` to its patch dir
+    cp "${srcdir}/0001-firebird-la64-support.patch" external/firebird
+    # And then patch `UnpackedTarball_firebird.mk` to enable to
+    # use `0001-firebird-la64-support.patch`
+    patch external/firebird/UnpackedTarball_firebird.mk < "${srcdir}/0002-UnpackedTarball_firebird.mk-apply-la64-support.diff"
+
 # should be obsolete - Debian has dropped this long time ago
     # fix not upstreamable pyuno paths - FS#54250
 #    patch -Np1 -i "${srcdir}"/make-pyuno-work-with-system-wide-module-install.diff
@@ -424,3 +431,8 @@ package_libreoffice-still() {
     'sid','sk','sl','sq','sv','ta','tg','tr','ug','uk','vi','zh-CN','zh-TW'])" \
         > "${pkgdir}"/usr/lib/libreoffice/help/languages.js
 }
+
+source+=("0001-firebird-la64-support.patch"
+         "0002-UnpackedTarball_firebird.mk-apply-la64-support.diff")
+sha256sums+=('7e160a4e30f66aefbb60c0ed877d196d04840221724e6473c8571fda70ab7922'
+             'fb2f789f20e447f4bb68f931843a27637e8b5d2e496b65be256ab7ee3a62fc7a')
