--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,7 +19,7 @@ _securityver=17
 _updatever=10
 pkgver=${_majorver}.${_minorver}.${_securityver}.u${_updatever}
 pkgrel=1
- _git_tag=jdk-${_majorver}.${_minorver}.${_securityver}+${_updatever}
+ _git_tag=jdk-${_majorver}.${_minorver}.${_securityver}+${_updatever}-ls-0
 arch=('x86_64')
 url='https://openjdk.java.net/'
 license=('LicenseRef-Java')
@@ -58,12 +58,12 @@ options=(
   !lto
 )
 source=(
-  https://github.com/openjdk/jdk${_majorver}u/archive/${_git_tag}.tar.gz
+  https://github.com/loongson/jdk${_majorver}u/archive/${_git_tag}.tar.gz
   freedesktop-java.desktop
   freedesktop-jconsole.desktop
   freedesktop-jshell.desktop
 )
-sha256sums=('6c8015923a618a3568d8ca92ec8bd263116461b1a7320f9c4176b813e3ea95a6'
+sha256sums=('006373aa38f95f7502680cdfd1c0833974bf443369891317cc9a2e3b81909a87'
             '85c4742764590783160af74587a47269e6797fbdf17ec485c7644bd239adf61d'
             'abac1ab09a33a3654378bc69be5a7cf075263ab3ae9efec1eb25cf388e711bb7'
             'e7cce0ecf868f656d8dc2eb25ab82ab665526a0a28aba20f02632dd29962dac3')
@@ -73,6 +73,7 @@ case "${CARCH}" in
   i686)   _JARCH='x86';;
   riscv64)_JARCH='riscv64';;
   aarch64)_JARCH='aarch64';;
+  loong64) _JARCH='loongarch64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
