--- a/PKGBUILD
+++ b/PKGBUILD
@@ -166,6 +166,21 @@ check() {
     --deselect=tests/test_sentinel.py::test_get_sentinels
     --deselect=tests/test_sentinel.py::test_get_master_addr_by_name
     --deselect=tests/test_sentinel.py::test_redis_master_usage
+
+    # To build on loong64 (floating point precision issues)
+    --deselect=tests/test_asyncio/test_commands.py::TestRedisCommands::test_geopos
+    --deselect=tests/test_asyncio/test_commands.py::TestRedisCommands::test_georadius_with
+    --deselect=tests/test_asyncio/test_commands.py::TestRedisCommands::test_georadius_store_dist
+    --deselect=tests/test_asyncio/test_commands.py::TestRedisCommands::test_georadiusmember
+    --deselect=tests/test_commands.py::TestRedisCommands::test_geopos
+    --deselect=tests/test_commands.py::TestRedisCommands::test_georadius_with
+    --deselect=tests/test_commands.py::TestRedisCommands::test_georadius_store_dist
+    --deselect=tests/test_commands.py::TestRedisCommands::test_georadiusmember
+    --deselect=tests/test_commands.py::TestRedisCommands::test_geosearch_member
+    --deselect=tests/test_commands.py::TestRedisCommands::test_geosearch_with
+    --deselect=tests/test_commands.py::TestRedisCommands::test_geosearchstore_dist
+    --deselect=tests/test_cluster.py::TestClusterRedisCommands::test_geosearchstore_dist
+    --deselect=tests/test_cluster.py::TestClusterRedisCommands::test_cluster_georadius_store_dist
   )
 
   # Run standalone test suite - targets the Redis server running :6379 and the
