--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,6 +22,7 @@ source=("https://github.com/intel/gmmlib/archive/intel-gmmlib-${pkgver}/${pkgnam
 sha256sums=('6ae262b95540681ad5478c63affb880135f772c95410b092009a6f7dfc04a5ba')
 
 build() {
+    patch -Np1 -d gmmlib-${pkgname}-${pkgver} -i "${srcdir}/add-loongarch64-build-support.patch"
     cmake -B build -S "gmmlib-${pkgname}-${pkgver}" \
         -G 'Unix Makefiles' \
         -DCMAKE_BUILD_TYPE='Release' \
@@ -35,3 +36,7 @@ package() {
     DESTDIR="$pkgdir" cmake --install build
     install -D -m644 "gmmlib-${pkgname}-${pkgver}/LICENSE.md" -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
+
+source+=("add-loongarch64-build-support.patch::https://patch-diff.githubusercontent.com/raw/intel/gmmlib/pull/133.diff")
+sha256sums+=('3205f921fa1bedff310e99df69afb75862969593c3e26e0c71c9f071600ad6e2')
+makedepends+=(simde)
