--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,7 +20,9 @@ sha256sums=('d053bf6ce1d588a75cfe8c9316269486e9d8fb89dcdf6fd92836fa2e3df61305'
 prepare() {
   cd ghc-$_ghcver
   # apply patch to enable ghci for riscv64
-  patch -p1 < "$srcdir"/ghc-enable-ghci-for-riscv64.patch
+  #patch -p1 < "$srcdir"/ghc-enable-ghci-for-riscv64.patch
+  # Always enable ghci like Debian
+  patch -p1 < "$srcdir"/hadrian-enable-interpreter.patch
   # Allow GHC 9.6's bundled directory on Linux; the newer minimum fixes a Windows-only race.
   # https://gitlab.haskell.org/ghc/ghc/-/issues/24382
   uusi -u QuickCheck -u 'directory:>=1.3.8.5 && <1.4' hadrian/hadrian.cabal
@@ -46,3 +48,6 @@ package() {
   install -D -m644 LICENSE -t "$pkgdir"/usr/share/licenses/$pkgname/
   rm -f "$pkgdir"/usr/share/doc/$pkgname/LICENSE
 }
+
+source+=("hadrian-enable-interpreter.patch")
+sha256sums+=('2e07b656f46f00c2d60a4c447bc47c9a6012ba98d98d6d830a7c51afbda2695a')
