diff --git a/PKGBUILD b/PKGBUILD
index 90e1e9e..9b3bc6b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,11 +20,10 @@ arch=('x86_64')
 license=('GPL-2.0-or-later')
 url='https://valgrind.org/'
 depends=('glibc' 'perl' 'debuginfod')
-makedepends=('gdb' 'lib32-glibc' 'lib32-gcc-libs' 'docbook-xml'
+makedepends=('gdb' 'docbook-xml'
              'docbook-xsl' 'docbook-sgml')
 checkdepends=('procps-ng')
 optdepends=(
-  'lib32-glibc: 32-bit ABI support'
   'python: cg_* scripts'
 )
 provides=('valgrind-multilib')
@@ -45,7 +44,7 @@ b2sums=('5337096c846b62899017589fbb242ca601741ebb353834cd788efb60f951f2380c5904b
 options=(!lto) # https://bugs.kde.org/show_bug.cgi?id=338252
 
 prepare() {
-  cd valgrind-${pkgver}
+  cd valgrind-loongarch64
   patch -Np1 < ../valgrind-3.7.0-respect-flags.patch
   sed -i 's|sgml/docbook/xsl-stylesheets|xml/docbook/xsl-stylesheets-1.79.2-nons|' docs/Makefile.am
 
@@ -53,7 +52,7 @@ prepare() {
 }
 
 build() {
-  cd valgrind-${pkgver}
+  cd valgrind-loongarch64
   ./configure \
     --prefix=/usr \
     --sysconfdir=/etc \
@@ -69,7 +68,7 @@ check() {
   # only run if glibc-debug is supplied manually
   if ! pacman -Q glibc-debug; then echo -e "\033[1;31mcheck() not run, supply glibc-debug if unintended!\033[0m"; return 0; fi
 
-  cd valgrind-${pkgver}
+  cd valgrind-loongarch64
 
   # Make sure a basic binary runs. There should be no errors.
   ./vg-in-place --error-exitcode=1 /bin/true
@@ -107,7 +106,7 @@ check() {
 }
 
 package() {
-  cd valgrind-${pkgver}
+  cd valgrind-loongarch64
   make DESTDIR="${pkgdir}" install
 
   install -Dm644 docs/*.1 -t "$pkgdir/usr/share/man/man1"
@@ -117,4 +116,11 @@ package() {
   fi
 }
 
+# Unable to use the version control used by Arch Linux upstream
+_commit_hash='1e729f9bf417a8af9e2aa5f3933d6efb12f8d46c'
+source+=("git+https://github.com/FreeFlyingSheep/valgrind-loongarch64.git#commit=${_commit_hash}")
+makedepends+=(git)
+sha512sums+=('SKIP')
+b2sums+=('SKIP')
+
 # vim: ts=2 sw=2 et:
