--- a/PKGBUILD
+++ b/PKGBUILD
@@ -76,6 +76,17 @@ prepare() {
   sed 's|${DEP_SHA1_cxxopts}|&\n\ \ \ \ \FIND_PACKAGE_ARGS NAMES cxxopts|g' \
     -i cmake/external/onnxruntime_external_deps.cmake
 
+  # Find system nsync for loong64. The above sed command doesn't work for loong64.
+  cat >> cmake/external/onnxruntime_external_deps.cmake << EOF
+
+  # Find and use system nsync library
+  find_package(nsync_cpp QUIET)
+  if(nsync_cpp_FOUND)
+    message(STATUS "Found system nsync_cpp")
+    add_library(nsync::nsync_cpp ALIAS nsync_cpp)
+  endif()
+EOF
+
   patch -Np1 -i "${srcdir}/${pkgbase}-install-orttraining-files.patch"
   patch -Np1 -i "${srcdir}/${pkgbase}-system-flatbuffers.patch"
 
@@ -87,8 +98,8 @@ prepare() {
 
   cd "${srcdir}"
   cp -r "${pkgbase}" "${pkgbase}-cpu"
-  cp -r "${pkgbase}" "${pkgbase}-cuda"
-  cp -r "${pkgbase}" "${pkgbase}-opt-cuda"
+  #cp -r "${pkgbase}" "${pkgbase}-cuda"
+  #cp -r "${pkgbase}" "${pkgbase}-opt-cuda"
   cp -r "${pkgbase}" "${pkgbase}-rocm"
   cp -r "${pkgbase}" "${pkgbase}-opt-rocm"
 }
@@ -168,6 +179,7 @@ build() {
   cp -r build/onnxruntime/* onnxruntime
   python -m build --wheel --no-isolation
 
+  : <<COMMENT_SEPARATOR
   ##### onnxruntime-cuda
 
   echo "Build onnxruntime with CUDA without optimization"
@@ -198,7 +210,7 @@ build() {
   python setup.py --help # We have to call it like this once to generate the file we need.
   cp -r build/onnxruntime/* onnxruntime
   python -m build --wheel --no-isolation
-
+COMMENT_SEPARATOR
 
   ##### onnxruntime-rocm
 
@@ -216,11 +228,11 @@ build() {
 
   ##### onnxruntime-opt-rocm
 
-  echo "Build onnxruntime with ROCm with AVX optimizations"
+  echo "Build onnxruntime with ROCm with LASX optimizations"
   cd "${srcdir}/${pkgbase}-opt-rocm"
-  echo 'string(APPEND CMAKE_C_FLAGS " -march=haswell")' \
+  echo 'string(APPEND CMAKE_C_FLAGS " -march=la464")' \
     >> cmake/adjust_global_compile_flags.cmake
-  echo 'string(APPEND CMAKE_CXX_FLAGS " -march=haswell")' \
+  echo 'string(APPEND CMAKE_CXX_FLAGS " -march=la464")' \
     >> cmake/adjust_global_compile_flags.cmake
   cmake "${_cmake_rocm_args[@]}"
   cmake --build build
@@ -285,7 +297,7 @@ package_onnxruntime-rocm() {
 }
 
 package_onnxruntime-opt-rocm() {
-  pkgdesc="$_pkgdesc (with ROCm and AVX2 CPU optimizations)"
+  pkgdesc="$_pkgdesc (with ROCm and LASX CPU optimizations)"
   depends+=('rocm-hip-sdk' 'roctracer' 'rccl' 'migraphx')
   provides=("${pkgbase}=${pkgver}")
   conflicts=("${pkgbase}")
@@ -347,7 +359,7 @@ package_python-onnxruntime-rocm() {
 }
 
 package_python-onnxruntime-opt-rocm() {
-  pkgdesc="$_pkgdesc (with ROCm and AVX2 CPU optimizations)"
+  pkgdesc="$_pkgdesc (with ROCm and LASX CPU optimizations)"
   depends+=("${pkgbase}-opt-rocm" "${_pydepends[@]}")
   provides=("python-${pkgbase}=${pkgver}")
   conflicts=("python-${pkgbase}")
@@ -357,3 +369,6 @@ package_python-onnxruntime-opt-rocm() {
   install -Dm644 LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
   install -Dm644 ThirdPartyNotices.txt "${pkgdir}/usr/share/licenses/${pkgname}/ThirdPartyNotices.txt"
 }
+
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|cudnn|nccl)$'))
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '(cuda)$'))
