--- a/PKGBUILD
+++ b/PKGBUILD
@@ -78,8 +78,22 @@ prepare() {
       -e '368aadd_library(nsync::nsync_cpp ALIAS nsync_cpp)' \
       -i cmake/external/onnxruntime_external_deps.cmake
 
+  # Find system nsync for loong64. The above sed command doesn't work for loong64.
+  cat >> cmake/external/onnxruntime_external_deps.cmake << EOF
+
+  # Find and use system nsync library
+  find_package(nsync_cpp QUIET)
+  if(nsync_cpp_FOUND)
+    message(STATUS "Found system nsync_cpp")
+    add_library(nsync::nsync_cpp ALIAS nsync_cpp)
+  endif()
+EOF
+
   patch -Np1 -i "${srcdir}/${pkgbase}-install-orttraining-files.patch"
   patch -Np1 -i "${srcdir}/${pkgbase}-system-flatbuffers.patch"
+  patch -Np1 -i "${srcdir}/qgemm_kernel_lsx-add-S8U8-S8S8.patch"
+  patch -Np1 -i "${srcdir}/fixgitlab.patch"
+  patch -Np1 -i "${srcdir}/fixgcc-15.patch"
 
   cd "${srcdir}"
   cp -r "${pkgbase}" "${pkgbase}-cuda"
@@ -106,7 +120,7 @@ build() {
     -DBUILD_TESTING=OFF
     -Donnxruntime_ENABLE_TRAINING=ON
     -Donnxruntime_ENABLE_LAZY_TENSOR=OFF
-    -Donnxruntime_USE_MPI=ON
+    -Donnxruntime_USE_MPI=OFF
     -Donnxruntime_USE_DNNL=ON
     # Stable release of eigen is too old for onnxruntime.
     -Donnxruntime_USE_PREINSTALLED_EIGEN=OFF
@@ -150,9 +164,9 @@ build() {
 
   ##### onnxruntime-cuda
 
-  echo "Build onnxruntime with CUDA without optimization"
+  echo "Build onnxruntime without optimization" # We don't have cuda
   cd "${srcdir}/${pkgbase}-cuda"
-  cmake "${_cmake_cuda_args[@]}"
+  cmake "${_cmake_args[@]}"
   cmake --build build
 
   # Manually fix https://github.com/microsoft/onnxruntime/issues/24570
@@ -164,13 +178,13 @@ build() {
 
   ##### onnxruntime-opt-cuda
 
-  echo "Build onnxruntime with CUDA with AVX optimizations"
+  echo "Build onnxruntime with LASX optimizations"
   cd "${srcdir}/${pkgbase}-opt-cuda"
-  echo 'string(APPEND CMAKE_C_FLAGS " -march=haswell")' \
+  echo 'string(APPEND CMAKE_C_FLAGS " -march=la464")' \
     >> cmake/adjust_global_compile_flags.cmake
-  echo 'string(APPEND CMAKE_CXX_FLAGS " -march=haswell")' \
+  echo 'string(APPEND CMAKE_CXX_FLAGS " -march=la464")' \
     >> cmake/adjust_global_compile_flags.cmake
-  cmake "${_cmake_cuda_args[@]}"
+  cmake "${_cmake_args[@]}"
   cmake --build build
 
   # Manually fix https://github.com/microsoft/onnxruntime/issues/24570
@@ -198,9 +212,9 @@ build() {
 
   echo "Build onnxruntime with ROCm with AVX optimizations"
   cd "${srcdir}/${pkgbase}-opt-rocm"
-  echo 'string(APPEND CMAKE_C_FLAGS " -march=haswell")' \
+  echo 'string(APPEND CMAKE_C_FLAGS " -march=la464")' \
     >> cmake/adjust_global_compile_flags.cmake
-  echo 'string(APPEND CMAKE_CXX_FLAGS " -march=haswell")' \
+  echo 'string(APPEND CMAKE_CXX_FLAGS " -march=la464")' \
     >> cmake/adjust_global_compile_flags.cmake
   cmake "${_cmake_rocm_args[@]}"
   cmake --build build
@@ -214,9 +228,6 @@ build() {
 
 package_onnxruntime() {
   pkgdesc="$_pkgdesc"
-  optdepends+=('cuda: nVidia GPU acceleration'
-               'cudnn: nVidia GPU acceleration'
-               'nccl: nVidia GPU acceleration')
 
   cd "${pkgbase}-cuda"
   DESTDIR="${pkgdir}" cmake --install build
@@ -226,10 +237,8 @@ package_onnxruntime() {
 }
 
 package_onnxruntime-opt() {
-  pkgdesc="$_pkgdesc (with AVX2 CPU optimizations)"
-  optdepends+=('cuda: nVidia GPU acceleration'
-               'cudnn: nVidia GPU acceleration'
-               'nccl: nVidia GPU acceleration')
+  pkgdesc="$_pkgdesc (with LASX CPU optimizations)"
+
   provides=("${pkgbase}=${pkgver}")
   conflicts=("${pkgbase}")
 
@@ -254,7 +263,7 @@ package_onnxruntime-rocm() {
 
 package_onnxruntime-opt-rocm() {
   echo "foo"
-  pkgdesc="$_pkgdesc (with ROCm and AVX2 CPU optimizations)"
+  pkgdesc="$_pkgdesc (with ROCm and LASX CPU optimizations)"
   depends+=('rocm-hip-sdk' 'roctracer' 'rccl')
   provides=("${pkgbase}=${pkgver}")
   conflicts=("${pkgbase}")
@@ -268,9 +277,6 @@ package_onnxruntime-opt-rocm() {
 package_python-onnxruntime() {
   pkgdesc="$_pkgdesc"
   depends+=("${pkgbase}" "${_pydepends[@]}")
-  optdepends+=('cuda: nVidia GPU acceleration'
-               'cudnn: nVidia GPU acceleration'
-               'nccl: nVidia GPU acceleration')
 
   cd "${pkgbase}-cuda"
   python -m installer --destdir="${pkgdir}" dist/*.whl
@@ -279,13 +285,10 @@ package_python-onnxruntime() {
 }
 
 package_python-onnxruntime-opt() {
-  pkgdesc="$_pkgdesc (with AVX2 CPU optimizations)"
+  pkgdesc="$_pkgdesc (with LASX CPU optimizations)"
   depends+=("${pkgbase}-opt" "${_pydepends[@]}")
   provides=("python-${pkgbase}=${pkgver}")
   conflicts=("python-${pkgbase}")
-  optdepends+=('cuda: nVidia GPU acceleration'
-               'cudnn: nVidia GPU acceleration'
-               'nccl: nVidia GPU acceleration')
 
   cd "${pkgbase}-opt-cuda"
   python -m installer --destdir="${pkgdir}" dist/*.whl
@@ -306,7 +309,7 @@ package_python-onnxruntime-rocm() {
 }
 
 package_python-onnxruntime-opt-rocm() {
-  pkgdesc="$_pkgdesc (with ROCm and AVX2 CPU optimizations)"
+  pkgdesc="$_pkgdesc (with ROCm and LASX CPU optimizations)"
   depends+=("${pkgbase}-opt-rocm" "${_pydepends[@]}")
   provides=("python-${pkgbase}=${pkgver}")
   conflicts=("python-${pkgbase}")
@@ -316,3 +319,12 @@ package_python-onnxruntime-opt-rocm() {
   install -Dm644 LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
   install -Dm644 ThirdPartyNotices.txt "${pkgdir}/usr/share/licenses/${pkgname}/ThirdPartyNotices.txt"
 }
+
+depends=($(printf "%s\n" "${depends[@]}" | grep -Ev '^(intel-oneapi-mkl)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|cudnn|nccl|gcc14)$'))
+source+=("qgemm_kernel_lsx-add-S8U8-S8S8.patch::https://github.com/microsoft/onnxruntime/commit/641449fa23e89d6577582c903fe31d44f4d82a15.diff"
+         "fixgitlab.patch::https://github.com/microsoft/onnxruntime/commit/f57db79743c4d1a3553aa05cf95bcd10966030e6.patch"
+         "fixgcc-15.patch::https://github.com/intel/onnxruntime/commit/911094fe0f1712bea686134431f0b5e55430265d.patch")
+b2sums+=('ff3bda75bd08218a409fd1dbc0197e0a36fb6ecf8d75e673f8e86ea785bc7bed310a27049958049d4378e78fc91c2009b3aadc1d1e0989030a5cab989803d448'
+         '2b4938212a7f776deffc1b19dc200a74c2ee2096f3c316d1e5bc36eac89bfbcb25748390eb4feee9d469a400df4c9fa779e7fc55038c3df502ba05bee8f69601'
+         '4aa8595eafa70a57e4e3af8b330e7f24b56c2c42b334a47ae42ac3f2b25e1ffe9319e3c85e7253b6be610c2dc59bd3cd4297612ddd9d4e980d3d10531bcea61e')
