--- a/PKGBUILD
+++ b/PKGBUILD
@@ -102,6 +102,17 @@ prepare() {
 
   patch -Np1 -i "${srcdir}/${pkgbase}-system-dnnl.patch"
 
+  # Find system nsync for loong64. The above sed command doesn't work for loong64.
+  cat >> cmake/external/onnxruntime_external_deps.cmake << EOF
+
+  # Find and use system nsync library
+  find_package(nsync_cpp QUIET)
+  if(nsync_cpp_FOUND)
+    message(STATUS "Found system nsync_cpp")
+    add_library(nsync::nsync_cpp ALIAS nsync_cpp)
+  endif()
+EOF
+
   patch -Np1 -i "${srcdir}/${pkgbase}-install-orttraining-files.patch"
   patch -Np1 -i "${srcdir}/${pkgbase}-system-flatbuffers.patch"
 
@@ -157,8 +168,8 @@ prepare() {
 
   cd "${srcdir}"
   cp -r "${pkgbase}" "${pkgbase}-cpu"
-  cp -r "${pkgbase}" "${pkgbase}-cuda"
-  cp -r "${pkgbase}" "${pkgbase}-opt-cuda"
+  #cp -r "${pkgbase}" "${pkgbase}-cuda"
+  #cp -r "${pkgbase}" "${pkgbase}-opt-cuda"
   cp -r "${pkgbase}" "${pkgbase}-rocm"
   cp -r "${pkgbase}" "${pkgbase}-opt-rocm"
 }
@@ -245,6 +256,7 @@ build() {
   cp -r build/onnxruntime/* onnxruntime
   python -m build --wheel --no-isolation
 
+  : <<COMMENT_SEPARATOR
   ##### onnxruntime-cuda
 
   echo "Build onnxruntime with CUDA without optimization"
@@ -269,7 +281,7 @@ build() {
 
   cp -r build/onnxruntime/* onnxruntime
   python -m build --wheel --no-isolation
-
+COMMENT_SEPARATOR
 
   ##### onnxruntime-rocm
 
@@ -285,11 +297,11 @@ build() {
 
   ##### onnxruntime-opt-rocm
 
-  echo "Build onnxruntime with ROCm with AVX optimizations"
+  echo "Build onnxruntime with ROCm with LASX optimizations"
   cd "${srcdir}/${pkgbase}-opt-rocm"
-  echo 'string(APPEND CMAKE_C_FLAGS " -march=haswell")' \
+  echo 'string(APPEND CMAKE_C_FLAGS " -march=la464")' \
     >> cmake/adjust_global_compile_flags.cmake
-  echo 'string(APPEND CMAKE_CXX_FLAGS " -march=haswell")' \
+  echo 'string(APPEND CMAKE_CXX_FLAGS " -march=la464")' \
     >> cmake/adjust_global_compile_flags.cmake
   cmake "${_cmake_rocm_args[@]}"
   cmake --build build
@@ -352,7 +364,7 @@ package_onnxruntime-rocm() {
 }
 
 package_onnxruntime-opt-rocm() {
-  pkgdesc="$_pkgdesc (with ROCm and AVX2 CPU optimizations)"
+  pkgdesc="$_pkgdesc (with ROCm and LASX CPU optimizations)"
   depends+=('rocm-hip-sdk' 'roctracer' 'rccl' 'migraphx')
   provides=("${pkgbase}=${pkgver}")
   conflicts=("${pkgbase}")
@@ -417,7 +429,7 @@ package_python-onnxruntime-rocm() {
 }
 
 package_python-onnxruntime-opt-rocm() {
-  pkgdesc="$_pkgdesc (with ROCm and AVX2 CPU optimizations)"
+  pkgdesc="$_pkgdesc (with ROCm and LASX CPU optimizations)"
   depends+=('rocm-hip-sdk' 'roctracer' 'rccl' 'migraphx')
   depends+=("${_pydepends[@]}")
   provides=("python-${pkgbase}=${pkgver}")
@@ -428,3 +440,6 @@ package_python-onnxruntime-opt-rocm() {
   install -Dm644 LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
   install -Dm644 ThirdPartyNotices.txt "${pkgdir}/usr/share/licenses/${pkgname}/ThirdPartyNotices.txt"
 }
+
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|cudnn|nccl|cudnn-frontend)$'))
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '(cuda)$'))
