diff --git a/PKGBUILD b/PKGBUILD
index 8a34c0f..8aced32 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -41,6 +41,7 @@ source=("cpu-x::git+https://github.com/TheTumultuousUnicornOfDarkness/CPU-X.git#
 sha512sums=('9153b2dfb1ac8d15ea4ad1f953d345077a13dbc698179c1831047c150f3e13477ba1ff4ef18d160ad925d98564b400909438e7224a4e3bb37f15169c3e220594')
 
 build() {
+	patch -Np1 -d "$pkgname" -i "${srcdir}/Fix-build-for-unsupport-processer.patch"
 	cmake -S "$pkgname" -B build -GNinja -DCMAKE_BUILD_TYPE=None -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBEXECDIR="lib/cpu-x"
 	cmake --build build
 }
@@ -52,3 +53,6 @@ check() {
 package() {
 	DESTDIR="$pkgdir" cmake --install build
 }
+
+source+=("Fix-build-for-unsupport-processer.patch")
+sha512sums+=('29b5adf648d5eb1203177c6a85f6a24bd1982889b1f5aba2985989feb3ddf6683e554e49204aad6247d5f76d827ad81a7669590d80d63860239ae7ac6b7803a3')
