diff --git a/PKGBUILD b/PKGBUILD
index 6eb2f7f..94207ae 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -34,6 +34,7 @@ prepare() {
   cd $pkgname-$pkgver
 
   patch -Np1 -i "$srcdir"/pin-java-version.patch
+  patch -Np1 -i "$srcdir"/use-system-node.patch
 }
 
 build() {
@@ -83,4 +84,8 @@ package() {
   install -Dm 644 LICENSE.txt "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE.txt"
 }
 
+makedepends+=('nodejs' 'npm' 'pnpm')
+source+=(use-system-node.patch)
+sha512sums+=('dc8df0c48091dd9e5cc57b6e0d2b4ad4d608274390734d4ba125ce3a5d5307d64da5da9b942a514231a86a530c148a348780313f864f1ae92a6a0fe75f16fe8b')
+
 # vim: ts=2 sw=2 et:
