--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,6 +36,9 @@ validpgpkeys=('ECCAC84C1BA08A6CC8E63FBBF22FB1D78A77AEAB'    # Giancarlo Razzolin
 prepare() {
   cd "$pkgname-$pkgver"
   patch -Np1 < ../0001-trigger.patch
+  patch -Np1 < ../mkinitcpio-loong64.patch
+  patch -Np1 < ../mkinitcpio-fix-error.patch
+  patch -Np1 < ../0001-test-cases-zsh-completion-Sync-Linux-banner-fix-from.patch
 }
 
 check() {
@@ -45,3 +48,12 @@ check() {
 package() {
   make -C "$pkgname-$pkgver" DESTDIR="$pkgdir" install
 }
+
+source+=('mkinitcpio-loong64.patch' 'mkinitcpio-fix-error.patch'
+         '0001-test-cases-zsh-completion-Sync-Linux-banner-fix-from.patch')
+sha512sums+=('9909059d59a2b7bd42320ac7d7b4a76bd215c6d9fbb95d5f3c77d338196b248daaa9d0a1514f6e14da16a4977e08426100e225d00e4320d8820d41eaccfd94fa'
+             '33e903a41168f156318a9b03809717a07d56911837bee8f2718a21b0e08b6f8ff361a13a26ff202006a00dfa90f00f2e0efa6cf83db1038833f7d600d3438772'
+             '2c898449e92ba3370cad7a2a8a43a09e08642693123b2abbb3a4ac6223bcd5b84adb81e2b226fe01ebbe672294604eb41f99cad522d8cbf7698636d59d42d3ab')
+b2sums+=('2c55cc8a863f171063b40b6fde60a2846f8a885d473426e61bd8f7da09661f0f0c0d97ae04d312a5801f271b6715cb40a8b5fbb2611741b6c19a25b9050b49e4'
+         'b9ccc1c97cadf88aec03d549d8bdf89536b95a64b05d316481815faca06c94aa867523c3cbfda352c44956c4c5959b42f70c05e71cb51278446f2389566ef62b'
+         '52198e620e91ba6114bc4551653a368497d707c6fe75029a53f6d4e9217634244b7a0ed510ce147ff2b8a58062100f0d5f8a5506b568e575cbcfc0577040178e')
