diff --git a/PKGBUILD b/PKGBUILD
index 1aaabd6..7402f7c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,6 +13,7 @@ b2sums=('3bffa4f9beef917c1ac731507e61ac716164829b1ce038e20708ca8d0511dd9a4382716
 
 prepare() {
   cd ${pkgname}-${pkgver}
+  patch -p1 -i "$srcdir/update-cargo.patch"
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
@@ -28,3 +29,6 @@ package() {
   install -Dm755 target/release/${pkgname} "${pkgdir}"/usr/bin/${pkgname}
 }
 
+
+source+=("update-cargo.patch")
+b2sums+=('e5ac1c6ce3dccc4ce2cbd216292beac69c33b4ae76a50bd53ecdf65431f866c0ff005d1d78b456a9a1e9c7293207f22dce0393b79fc13d7ee49d6b280e2541ff')
