--- a/PKGBUILD
+++ b/PKGBUILD
@@ -9,7 +9,7 @@ arch=('x86_64')
 url='https://nuspell.github.io/'
 license=('LGPL-3.0-or-later')
 depends=('icu' 'glibc' 'gcc-libs')
-makedepends=('cmake' 'pandoc')
+makedepends=('cmake')
 checkdepends=('catch2' 'expat')
 source=(${pkgname}-${pkgver}.tar.gz::"https://github.com/nuspell/nuspell/archive/v${pkgver}.tar.gz")
 sha512sums=('ab6d9394a55d9a2a347ccae47aeef6a96af70f421ad6ea8f7ac7fde2052790f37fb1c7ec3112daac7600d193430a560cb1915ab6557c9353717f65cb32f13ab8')
@@ -22,7 +22,8 @@ build() {
     -S ${pkgbase}-${pkgver} \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DCMAKE_INSTALL_LIBDIR=/usr/lib \
-    -DBUILD_SHARED_LIBS=ON
+    -DBUILD_SHARED_LIBS=ON \
+    -DBUILD_DOCS=OFF
   cmake --build "${pkgname}-${pkgver}"/build
 }
 
