--- a/PKGBUILD
+++ b/PKGBUILD
@@ -12,7 +12,7 @@ url="https://docs.xfce.org/panel-plugins/xfce4-xkb-plugin/start"
 license=('GPL-2.0-or-later')
 groups=('xfce4-goodies')
 depends=('xfce4-panel' 'libxklavier' 'librsvg' 'libwnck3' 'libnotify')
-makedepends=('git' 'xfce4-dev-tools')
+makedepends=('git' 'xfce4-dev-tools' 'intltool')
 optdepends=('xfce4-notifyd: show notifications on layout change')
 install=$pkgname.install
 source=("git+https://gitlab.xfce.org/panel-plugins/xfce4-xkb-plugin.git#tag=$pkgname-$pkgver")
