diff --git a/PKGBUILD b/PKGBUILD
index 684f377..8f18ecb 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -80,6 +80,9 @@ check() {
 
   rm test/parallel/test-http-client-close-with-default-agent.js
 
+  rm -f test/parallel/test-dgram-send-cb-quelches-error.js
+  rm -f test/parallel/test-net-autoselectfamily.js
+
   make test-only
 
 }
@@ -90,3 +93,4 @@ package() {
   make DESTDIR="${pkgdir}" install
   install -Dm644 LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
