diff --git a/PKGBUILD b/PKGBUILD
index 15b7a67..fdcfa34 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,7 +36,7 @@ depends=(
 )
 makedepends=(
   asciidoctor
-  shellcheck
+  # shellcheck # FIXME: shellcheck depends on ghc, which is missing
 )
 checkdepends=(
   bats
@@ -72,4 +72,6 @@ package() {
   make PREFIX=/usr DESTDIR="${pkgdir}" install
 }
 
+optdepends_loong64=("devtools-loong64: Support to build loong64 pkgs")
+
 # vim: ts=2 sw=2 et:
