--- a/PKGBUILD
+++ b/PKGBUILD
@@ -79,8 +79,8 @@ makedepends=(
 )
 checkdepends=(
   'python-pytest' 'python-jsbeautifier' 'python-h5py' 'acl' 'binutils' 'bzip2' 'cdrtools' 'cpio' 'e2fsprogs' 'enjarify'
-  'hdf5' 'imagemagick' 'java-environment=11' 'fontforge' 'gettext' 'ghc' 'gnupg' 'mono' 'pgpdump' 'poppler' 'sqlite' 'squashfs-tools'
-  'libxmlb' 'lz4' 'unzip' 'gzip' 'tar' 'tcpdump' 'vim' 'xz' 'llvm' 'colord' 'fpc' 'openssh' 'openssl' 'odt2txt' 'docx2txt' 'r' 'dtc'
+  'hdf5' 'imagemagick' 'java-environment=11' 'fontforge' 'gettext' 'gnupg' 'pgpdump' 'poppler' 'sqlite' 'squashfs-tools'
+  'libxmlb' 'lz4' 'unzip' 'gzip' 'tar' 'tcpdump' 'vim' 'xz' 'llvm' 'colord' 'openssh' 'openssl' 'odt2txt' 'docx2txt' 'r' 'dtc'
   'giflib' 'gnumeric' 'python-progressbar' 'binwalk' 'python-argcomplete' 'zstd' 'uboot-tools' 'asar')
 source=(https://diffoscope.org/archive/diffoscope-${pkgver}.tar.bz2{,.asc})
 sha512sums=('ee18f7a55e824b49b065c92f9a07a71a9e6f2b4e8c2b627195a453fface693247a49aa0f644f06f5ec192df589fd3533447c333397423a23a7b7402ffb54a4c4'
