diff -urN rtorrent.orig/scripts/common.m4 rtorrent/scripts/common.m4
--- rtorrent.orig/scripts/common.m4	2026-07-06 18:19:34.441336043 +0800
+++ rtorrent/scripts/common.m4	2026-07-06 18:29:54.406194442 +0800
@@ -118,6 +118,10 @@
             AC_MSG_RESULT([linux fallback enterprise 128 bytes])
             AC_DEFINE([LT_SMP_CACHE_BYTES], 128, [Fallback 128-byte alignment for Linux enterprise hardware.])
             ;;
+          loongarch32*|loongarch64*)
+            AC_MSG_RESULT([linux fallback LoongArch 64 bytes])
+            AC_DEFINE([LT_SMP_CACHE_BYTES], 64, [Fallback 64-byte alignment for Linux LoongArch hardware.])
+            ;;
           *)
             AC_MSG_RESULT([unrecognized CPU arch on Linux header fallback])
             AC_MSG_FAILURE([Unrecognized CPU architecture ($host_cpu) on Linux fallback path. Aborting build.])
