diff --git a/PKGBUILD b/PKGBUILD
index 04e614b..5de40d7 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -81,6 +81,7 @@ validpgpkeys=(D1CF2444A7858C5F2FB095B74A77747BC2386E50) # digiKam.org (digiKam p
 
 prepare() {
   patch -d $pkgname-$pkgver -p1 < kf6-dependencies.patch
+  patch -d $pkgname-$pkgver -p1 < fix-build-with-qt6.8.patch
 }
 
 build() {
@@ -99,3 +100,6 @@ build() {
 package() {
   DESTDIR="$pkgdir" cmake --install build
 }
+
+source+=("fix-build-with-qt6.8.patch::https://invent.kde.org/graphics/digikam/-/commit/a8b49ed8df676cae0f48b3369831edde2b74903e.patch")
+sha256sums+=('6de622280539fe01bf6d1f4825fb35cfa009014b6c4d9d9202718d6592712984')
