diff --git a/lib/fst/fst.h b/lib/fst/fst.h
index 7571e6d8..0b09daf2 100644
--- a/lib/fst/fst.h
+++ b/lib/fst/fst.h
@@ -111,7 +111,7 @@
 #define VSTCALLBACK
 
  /* t_fstPtrInt: pointer sized int */
-#if defined(_WIN32) && (defined(__x86_64__) || defined (_M_X64))
+#if defined(_WIN32) && (defined(__x86_64__) || defined (_M_X64) || defined(__loongarch_lp64))
 typedef long long t_fstPtrInt;
 #else
 typedef long t_fstPtrInt;
diff --git a/lib/vst3sdk/base/source/fdebug.cpp b/lib/vst3sdk/base/source/fdebug.cpp
index 18fa1c9b..5d2c7922 100644
--- a/lib/vst3sdk/base/source/fdebug.cpp
+++ b/lib/vst3sdk/base/source/fdebug.cpp
@@ -230,7 +230,7 @@ void FDebugBreak (const char* format, ...)
 #elif SMTG_OS_MACOS && __arm64__
 			raise (SIGSTOP);
 
-#elif __ppc64__ || __ppc__ || __arm__
+#elif __ppc64__ || __ppc__ || __arm__ || __loongarch_lp64
 			kill (getpid (), SIGINT);
 #elif __i386__ || __x86_64__
 			{