diff --git a/scripts/build_and_deploy.sh b/scripts/build_and_deploy.sh
index dd22bc89..27ed3a97 100644
--- a/scripts/build_and_deploy.sh
+++ b/scripts/build_and_deploy.sh
@@ -48,7 +48,7 @@ main()
 
     if [[ "$plugin_type$target_os$arch" = "" ]]
     then
-        echo "Usage: $0 fst|vst3 linux|windows x86|x86_64|riscv64 [avx|sse2|none]" >&2
+        echo "Usage: $0 fst|vst3 linux|windows x86|x86_64|riscv64|loongarch64 [avx|sse2|lsx|none]" >&2
         return 1
     fi
 
@@ -62,8 +62,9 @@ main()
     case "$arch" in
         "x86") target_platform="i686" ;;
         "x86_64"|"riscv64") target_platform="$arch" ;;
+        "loongarch64") target_platform="loong64"
         *)
-            echo "Unknown architecture: \"$arch\" - should be either \"x86\" or \"x86_64\" or \"riscv64\"" >&2
+            echo "Unknown architecture: \"$arch\" - should be either \"x86\" or \"x86_64\" or \"riscv64\" or \"loongarch64\"" >&2
             return 1
             ;;
     esac
@@ -94,16 +95,18 @@ main()
     TARGET_PLATFORM="$target_platform" INSTRUCTION_SET="$instruction_set" make "$plugin_type"
 
     case "$target_os-$plugin_type-$arch" in
-        "linux-fst-x86_64")     replace_in_dir "$built_plugin" $FST_DIRS_LINUX_64 ;;
-        "linux-vst3-x86_64")    replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_64 ;;
-        "linux-fst-riscv64")    replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_64 ;;
-        "linux-vst3-riscv64")   replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_64 ;;
-        "windows-fst-x86_64")   replace_in_dir "$built_plugin" $FST_DIRS_WINE_64 ;;
-        "windows-vst3-x86_64")  replace_in_dir "$built_plugin" $VST3_DIRS_WINE_64 ;;
-        "linux-fst-x86")        replace_in_dir "$built_plugin" $FST_DIRS_LINUX_32 ;;
-        "linux-vst3-x86")       replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_32 ;;
-        "windows-fst-x86")      replace_in_dir "$built_plugin" $FST_DIRS_WINE_32 ;;
-        "windows-vst3-x86")     replace_in_dir "$built_plugin" $VST3_DIRS_WINE_32 ;;
+        "linux-fst-x86_64")         replace_in_dir "$built_plugin" $FST_DIRS_LINUX_64 ;;
+        "linux-vst3-x86_64")        replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_64 ;;
+        "linux-fst-riscv64")        replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_64 ;;
+        "linux-vst3-riscv64")       replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_64 ;;
+        "linux-fst-loongarch64")    replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_64 ;;
+        "linux-vst3-loongarch64")   replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_64 ;;
+        "windows-fst-x86_64")       replace_in_dir "$built_plugin" $FST_DIRS_WINE_64 ;;
+        "windows-vst3-x86_64")      replace_in_dir "$built_plugin" $VST3_DIRS_WINE_64 ;;
+        "linux-fst-x86")            replace_in_dir "$built_plugin" $FST_DIRS_LINUX_32 ;;
+        "linux-vst3-x86")           replace_in_dir "$built_plugin" $VST3_DIRS_LINUX_32 ;;
+        "windows-fst-x86")          replace_in_dir "$built_plugin" $FST_DIRS_WINE_32 ;;
+        "windows-vst3-x86")         replace_in_dir "$built_plugin" $VST3_DIRS_WINE_32 ;;
     esac
 
     echo "SUCCESS" >&2