--- a/PKGBUILD
+++ b/PKGBUILD
@@ -45,8 +45,8 @@ build() {
   make \
     CXX_WARNINGS='-Wno-error' \
     SYS_LIB_PATH=/usr/lib \
-    TARGET_PLATFORM="$CARCH-gpp" \
-    INSTRUCTION_SET=sse2 \
+    TARGET_PLATFORM="$(uname -m)-gpp" \
+    INSTRUCTION_SET=lsx \
     VERSION_STR=$(git describe --tags) \
     VERSION_INT=$(git describe --tags | sed "s/[^0-9]//g") \
     vst3
@@ -57,8 +57,8 @@ package() {
 
   # vst3
   install -vDm644 \
-    "dist/js80p-dev-linux-$CARCH-sse2-vst3_single/js80p.vst3" \
-    "$pkgdir/usr/lib/vst3/$pkgname.vst3/Contents/$CARCH-linux/$pkgname.so"
+    "dist/js80p-dev-linux-$(uname -m)-lsx-vst3_single/js80p.vst3" \
+    "$pkgdir/usr/lib/vst3/$pkgname.vst3/Contents/$(uname -m)-linux/$pkgname.so"
 
   # presets
   install -vDm644 -t "$pkgdir/usr/share/$pkgname/presets" presets/*.js80p
