--- PKGBUILD
+++ PKGBUILD
@@ -36,6 +36,14 @@
 
   # remove cppcheck from makedepends
   patch -p1 -i "$srcdir/do-not-run-cppcheck.patch"
+
+  # add loong64 support
+  git cherry-pick -n afa480b48ce398250f42df5fe918254ddee1faa4
+  git cherry-pick -n 31d4bd535aa18efa44b05df0d08f9cb832c6402b
+  git cherry-pick -n 5090c2b53c5c4097cd52b158a2e3851a77b3c6ad
+  git cherry-pick -n 2bcb821bd028a30ad5c49c5f00f7697e159bedf6
+  git cherry-pick -n 7a9f8c0ce0076e51d589c09eeae70dc7e3e7cf50
+  git cherry-pick -n bef8ba950388f585442fb96dc57aae52597144b0
 }
 
 build() {
