diff --git a/PKGBUILD b/PKGBUILD
index f5f1db5..162643f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -67,7 +67,7 @@ build() {
 
   cd lemonade/src/app
   export RUSTUP_TOOLCHAIN=stable
-  npm run tauri build -- --no-bundle
+  cargo tauri build --no-bundle
 }
 
 package_lemonade-server() {
@@ -109,3 +109,5 @@ package_lemonade-desktop() {
   install -Dm 644 data/lemonade-app.desktop -t "${pkgdir}/usr/share/applications/"
   install -Dm 644 src/app/assets/logo.svg "${pkgdir}/usr/share/pixmaps/lemonade-app.svg"
 }
+
+makedepends+=(cargo-tauri)
