--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,9 +23,12 @@ sha512sums=('e95b2d30cb0273e5b723cf532f63b7b18f44c7f909a9d03331e0b7b7200bee8b635
 #  python scripts/regenerate-fixtures.py
 #}
 
+source+=("lexicon_update.diff")
+sha512sums+=("06c88015820aade51c2c96415c015a93c4c98546c3a1f23758cede87127240677ce08a3643075a6fb5274826999277c4a8605863e9c50b0912799b9fb14fb435")
+
 build() {
   cd $_pkgname-$pkgver
-
+  patch -Np1 < ../lexicon_update.diff
   SETUPTOOLS_SCM_PRETEND_VERSION=$pkgver python -m build --wheel --no-isolation
 }
 
