--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,11 +25,14 @@ prepare() {
   magick src/mime/codeblocks.png +set date:create +set date:modify -background none -extent 64x64 src/mime/codeblocks.png
   sed -i 's|$(datadir)/pixmaps|$(datadir)/icons/hicolor/64x64/apps|' src/mime/Makefile.{am,in}
   sed -i 's|$(datarootdir)/appdata|$(datarootdir)/metainfo|' Makefile.{am,in} src/plugins/contrib/appdata/Makefile.{am,in}
+  patch -p1 < ../add-loong64-support.patch
 }
 
 build() {
   cd ${pkgname}_$pkgver
-  ./configure --prefix=/usr --with-contrib-plugins=all,-FileManager
+  cp /usr/share/automake-1.1?/config.sub .
+  cp /usr/share/automake-1.1?/config.guess .
+  ./configure --prefix=/usr --with-contrib-plugins=all,-FileManager --with-boost-libdir=/usr/lib
   sed -i 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
 }
@@ -38,3 +41,6 @@ package() {
   cd ${pkgname}_$pkgver
   make DESTDIR="$pkgdir" install
 }
+
+source+=("add-loong64-support.patch")
+sha256sums+=('2f7e268226a6cb19b615b14a3b0dbea0d3711013106a570f9fface3f631cae01')
