--- a/PKGBUILD
+++ b/PKGBUILD
@@ -50,7 +50,7 @@ prepare() {
   # Install icon into hicolor directory
   magick src/mime/codeblocks.png +set date:create +set date:modify -background none -extent 64x64 src/mime/codeblocks.png
   sed -i 's|$(datadir)/pixmaps|$(datadir)/icons/hicolor/64x64/apps|' src/mime/Makefile.am
-
+  patch -p1 < ../add-loong64-support.patch
   autoreconf -fiv
 }
 
@@ -68,3 +68,6 @@ package() {
   cd ${pkgname}_$pkgver
   make DESTDIR="$pkgdir" install
 }
+
+source+=("add-loong64-support.patch")
+b2sums+=('eb8c99810770520375e4c48e8aa64fdb26965e9ffac988207dd1b399a4ec3d50ce259f19fe30622d6c8ebbc9dfe9fd24ab11e5eae6fb1346189a0b43354b6b90')
