diff --git a/vendor/github.com/moby/profiles/seccomp/default.json b/vendor/github.com/moby/profiles/seccomp/default.json
index dafe5cf6a6a3e..45a9a276dc2d3 100644
--- a/vendor/github.com/moby/profiles/seccomp/default.json
+++ b/vendor/github.com/moby/profiles/seccomp/default.json
@@ -52,6 +52,10 @@
 		{
 			"architecture": "SCMP_ARCH_RISCV64",
 			"subArchitectures": null
+		},
+		{
+			"architecture": "SCMP_ARCH_LOONGARCH64",
+			"subArchitectures": null
 		}
 	],
 	"syscalls": [
@@ -842,4 +846,4 @@
 			}
 		}
 	]
-}
\ No newline at end of file
+}
diff --git a/vendor/github.com/moby/profiles/seccomp/default_linux.go b/vendor/github.com/moby/profiles/seccomp/default_linux.go
index 26c50c342e7d3..93b57962fc8b6 100644
--- a/vendor/github.com/moby/profiles/seccomp/default_linux.go
+++ b/vendor/github.com/moby/profiles/seccomp/default_linux.go
@@ -39,6 +39,10 @@ func arches() []Architecture {
 			Arch:      specs.ArchRISCV64,
 			SubArches: nil,
 		},
+		{
+			Arch:      specs.ArchLOONGARCH64,
+			SubArches: nil,
+		},
 	}
 }
 
diff --git a/vendor/github.com/moby/profiles/seccomp/seccomp_linux.go b/vendor/github.com/moby/profiles/seccomp/seccomp_linux.go
index 92d4c188ba784..04da61278f6e9 100644
--- a/vendor/github.com/moby/profiles/seccomp/seccomp_linux.go
+++ b/vendor/github.com/moby/profiles/seccomp/seccomp_linux.go
@@ -42,6 +42,7 @@ var nativeToSeccomp = map[string]specs.Arch{
 	"riscv64":     specs.ArchRISCV64,
 	"s390":        specs.ArchS390,
 	"s390x":       specs.ArchS390X,
+	"loong64":     specs.ArchLOONGARCH64,
 }
 
 // GOARCH => libseccomp string
@@ -61,6 +62,7 @@ var goToNative = map[string]string{
 	"riscv64":     "riscv64",
 	"s390":        "s390",
 	"s390x":       "s390x",
+	"loong64":     "loong64",
 }
 
 // inSlice tests whether a string is contained in a slice of strings or not.
