diff --git a/PKGBUILD b/PKGBUILD
index 2226438..5b00b26 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -29,6 +29,7 @@ prepare() {
   git config submodule."ext-src/mscclpp".url "${srcdir}/${pkgname}"-mscclpp
 
   git -c protocol.file.allow=always submodule update --init --recursive
+  patch -p1 -d "${srcdir}/${pkgname}" -i "${srcdir}/rccl-loongarch64-support.patch"
 }
 
 build() {
@@ -59,3 +60,6 @@ package() {
 
   install -Dm644 "$srcdir/$pkgname/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=("rccl-loongarch64-support.patch")
+sha256sums+=('10e3306db067a75a5ade7681212ca2bcf44c20c715a2609a09f2d21cf5978073')
