--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,7 +19,7 @@ _securityver=9
 _updatever=10
 pkgver=${_majorver}.${_minorver}.${_securityver}.u${_updatever}
 pkgrel=1
- _git_tag=jdk-${_majorver}.${_minorver}.${_securityver}+${_updatever}
+ _git_tag=jdk-${_majorver}.${_minorver}.${_securityver}+${_updatever}-ls-0
 arch=('x86_64')
 url='https://openjdk.java.net/'
 license=('LicenseRef-Java')
@@ -55,12 +55,12 @@ options=(
   !lto
 )
 source=(
-  https://github.com/openjdk/jdk${_majorver}u/archive/${_git_tag}.tar.gz
+  https://github.com/loongson/jdk${_majorver}u/archive/${_git_tag}.tar.gz
   freedesktop-java.desktop
   freedesktop-jconsole.desktop
   freedesktop-jshell.desktop
 )
-sha256sums=('f2f9a21fd61ab3f63dac7e3d8747e849c9e0a1f14ed617c239ef9911755c756b'
+sha256sums=('747d5b6d8e965269ed09797a873aff894b77ceca08305596b93eee39ad64f977'
             '72111743ab6ab36854b0c85a504172983715d0798fce10bc4e35689b7d15fd93'
             '8ecdf5c1605bafa58b3f7da615e6d8d3d943e3a2d3831930d6efa7815aacce07'
             '50fc0d677489b73d549df2f08d759d5f057f200adbbab83ea5e87456152ee03e')
@@ -70,6 +70,7 @@ case "${CARCH}" in
   i686)   _JARCH='x86';;
   riscv64)_JARCH='riscv64';;
   aarch64)_JARCH='aarch64';;
+  loong64)_JARCH='loongarch64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
