--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,7 +23,6 @@ depends=(
   'abseil-cpp'
 )
 makedepends=(
-  'bazel'
   'cmake'
   'gtest'
   'python-build'
@@ -71,10 +70,10 @@ build() {
   cmake "${cmake_options[@]}"
   cmake --build build --verbose
 
-  cd "$pkgbase-$pkgver"
-  bazel build //python/dist:binary_wheel --enable_workspace=true
+  cd "$pkgbase-5.$pkgver"
+  python -m build --wheel --no-isolation
 
-  cd ruby
+  cd ../"$pkgbase-$pkgver"/ruby
   local _gemdir="$(gem env gemdir)"
   local _gemver=4.$pkgver
 
@@ -145,7 +144,7 @@ package_python-protobuf() {
     'python'
   )
 
-  python -m installer --destdir="$pkgdir" "$pkgbase-$pkgver"/bazel-bin/python/dist/*.whl
+  python -m installer --destdir="$pkgdir" "$pkgbase-5.$pkgver"/dist/*.whl
   install -vDm 644 $pkgbase-$pkgver/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
 
@@ -160,3 +159,6 @@ package_ruby-google-protobuf() {
   install -vDm 644 $pkgbase-$pkgver/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
   install -Dm 0644 $pkgbase-$pkgver/ruby/*.md -t "$pkgdir/usr/share/doc/${pkgname}"
 }
+
+source+=("$pkgbase-$pkgver-python.tar.gz::https://files.pythonhosted.org/packages/source/${pkgbase::1}/${pkgbase//-/_}/${pkgbase//-/_}-5.$pkgver.tar.gz")
+sha512sums+=('SKIP')
