diff --git a/PKGBUILD b/PKGBUILD
index 6a08705..57f6577 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -75,6 +75,7 @@ build() {
     --with-ruby-libdir=`ruby -e 'puts RbConfig::CONFIG["vendorlibdir"]'` \
     --with-perl-archdir=`perl -V:installvendorarch | cut -f2 -d\'` \
     --with-optionals=launcher,xmmsclient++,xmmsclient++-glib,perl,ruby,nycli,pixmaps,et,mdns,medialib-updater,sqlite2s4 \
+    --libdir=/usr/lib \
     --without-optionals=python
   ./waf build
 }
