--- a/PKGBUILD
+++ b/PKGBUILD
@@ -72,6 +72,7 @@ case "${CARCH}" in
   i686)   _JARCH='x86';;
   riscv64)_JARCH='riscv64';;
   aarch64) _JARCH='aarch64';;
+  loong64) _JARCH='loongarch64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -456,4 +457,14 @@ package_openjdk25-doc() {
   ln -s ${pkgbase} "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+# We cannot use upstream's version control
+# Reset some versions
+_loongfixver=7
+pkgver=${_majorver}.0.4.u${_loongfixver}ls
+_jdkdir=jdk25u-jdk-${_majorver}.0.4-${_loongfixver}-ls-ga
+_imgdir=${_jdkdir}/build/linux-${_JARCH}-server-release/images
+
+source+=("https://github.com/loongson/jdk25u/archive/refs/tags/jdk-${_majorver}.0.4+${_loongfixver}-ls-ga.tar.gz")
+sha256sums+=('b985b6f0777110528cf5c7992180baea9e333621ad258a43ccce3621a40992f8')
+
 # vim: ts=2 sw=2 et:
