--- a/PKGBUILD
+++ b/PKGBUILD
@@ -43,3 +43,11 @@ package() {
   DESTDIR="${pkgdir}" meson install -C build
   install -Dm 644 cpp-httplib/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+prepare() {
+  cd "${srcdir}/cpp-httplib"
+  patch -Np1 -i "${srcdir}/meson-fix-libanl.patch"
+}
+
+source+=(meson-fix-libanl.patch)
+b2sums+=('a656bbb6ef06bb201f23fd15be693dee1b46564e1078749a63687d67458aec18feeb882030ff900dbc76aac3ede0b26b5311d72946983fb682678b8f47a4e9e6')
