--- a/PKGBUILD
+++ b/PKGBUILD
@@ -31,3 +31,10 @@ build() {
 package() {
     meson install -C build --destdir "$pkgdir"
 }
+
+prepare() {
+    patch -d "$pkgname-$pkgver" -p1 -i "$srcdir/fix-FTBFS-with-incompatible-pointer-types.patch"
+}
+
+source+=("fix-FTBFS-with-incompatible-pointer-types.patch::https://patch-diff.githubusercontent.com/raw/BuddiesOfBudgie/budgie-control-center/pull/80.patch")
+b2sums+=('23d908433c135479fe6e3ed1af3fb8012990da7cb66b64a27dd63ab6cf42e39e30dc1786998019ef3938fd5becf42d8932888d2425f13e740cb463cfc1467e1f')
