--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,8 +36,13 @@ prepare() {
   sed -i 's|"libraries"|"/usr/share/materialx/libraries"|g' source/MaterialXGenShader/GenOptions.h
 
   dos2unix python/Scripts/*
+  dos2unix source/MaterialXView/NanoGUI/CMakeLists.txt
+  patch -Np1 -i "$srcdir"/nanogui-do-not-fallback-to-nehalem.patch -d source/MaterialXView/NanoGUI
 }
 
+source+=("nanogui-do-not-fallback-to-nehalem.patch::https://github.com/mitsuba-renderer/nanogui/pull/156.diff")
+sha256sums+=('fe60c485ddc71df9be0bd79ddda8bdd127d1ff8beec7b6fc2ffc4e1dcc0bbb70')
+
 build() {
   cmake -S MaterialX-${pkgver} -B build -G Ninja \
   	-DCMAKE_INSTALL_PREFIX=/usr \
