diff --git a/PKGBUILD b/PKGBUILD
index 59382e9..1899f89 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -111,6 +111,10 @@ _reverts=(
 )
 
 prepare() {
+  # dont build systemd with -mlsx, else linuxloongach64.stub does not work in edk2
+  export CFLAGS=${CFLAGS/-mlsx /}
+  export CXXFLAGS=${CXXFLAGS/-mlsx /}
+
   cd "${_systemd_src_dir}"
 
   local _c _l
@@ -383,4 +387,6 @@ package_systemd-ukify() {
   mv systemd-ukify/install.d "$pkgdir"/usr/lib/kernel/install.d
 }
 
+makedepends=(${makedepends[@]/lib32-gcc-libs})
+
 # vim:ft=sh syn=sh et sw=2:
