--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,8 +27,12 @@ pkgver() {
 
 prepare() {
   cd webrtc-audio-processing
+  patch -Np1 -i ../webrtc-audio-processing-loong64.patch
 }
 
+source+=('webrtc-audio-processing-loong64.patch')
+b2sums+=('82645a6da3e482209975fd04eec41bf4b94781e2fd0ed9df433c959738cba26c41e6d7833fadc5427a37feaa8a12a71ee5600b9531de6a6651f1b60ccd4983ed')
+
 build() {
   local meson_options=(
     # must match abseil
