--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,6 +27,7 @@ prepare() {
   # replace pkg_resources with importlib.metadata for version lookup
   # https://github.com/wiseman/py-webrtcvad/pull/96
   git cherry-pick -n ca2377f2b6ac4c6e8d473e733dd8249302aef482
+  patch -Np1 -i ../add-loong64-build.patch
 }
 
 build() {
@@ -38,3 +39,6 @@ package() {
   python -m installer --destdir="${pkgdir}" py-webrtcvad/dist/*.whl
   install -Dm 644 py-webrtcvad/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+source+=('add-loong64-build.patch')
+b2sums+=('6ef76a3848eaec8c645d2aca92a99cb086445364d407bc52f2f1af7dffe9a1d7bbdb912559909bfc90f5012d9ea57a488ba112c67c268c01ad64156967cebb74')
