--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,8 +13,17 @@ optdepends=('gtk2: GTK+ 2 theme'
 source=(https://download.sugarlabs.org/sources/sucrose/glucose/$pkgname/$pkgname-$pkgver.tar.xz)
 sha256sums=('6052b4d3488daf78901d2914015e3c67017cb1bb0cf669625b1f1d62921e9afa')
 
+source+=('port-to-empy4.2-2.patch::https://github.com/sugarlabs/sugar-artwork/commit/14de4382ab2bc9ce4aa5f4c2fdef595a19c0a452.patch'
+         'port-to-empy4.2-1.patch::https://github.com/sugarlabs/sugar-artwork/commit/84a0f2c115ddabc0d8ec16cc09080d62c067a90f.patch'
+)
+sha256sums+=('7b4641e380323b0fdc5380003dfe940d2d480c8ce200e8352a91921a2cff20e0'
+             'a1c7b8be87a5012bffc1dd8f205d08f9c66c977a55f000343ed974ac0971fc46')
+
 build() {
   cd $pkgname-$pkgver
+  patch -Np1 -i ../port-to-empy4.2-1.patch
+  patch -Np1 -i ../port-to-empy4.2-2.patch
+  autoreconf -ifv
   ./configure --prefix=/usr
   sed -i 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
