--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,6 +7,7 @@ pkgrel=1
 _yajsw=13.18
 _jsarif=2.1
 _protobuf=6.31.0
+_protobuf_cpp=31.0
 _psutil=5.9.8
 _setuptools=80.9.0
 _wheel=0.45.1
@@ -39,6 +40,8 @@ makedepends=(
   python-setuptools
   z3
   z3-java
+  abseil-cpp
+  cmake
 )
 optdepends=(
   'pam: GhidraServer support'
@@ -105,8 +108,9 @@ prepare() {
   # Ignore lack of licensing for YAJSW zip, packed FID datasets, and the native binaries
   sed -i '/FileTree tree/a\\t\texclude "yajsw-stable-**.zip"\n\t\texclude "src/main/fidb/**.fidb"\n\t\texclude "os/linux64/**"' gradle/support/ip.gradle
 
+  patch -Np1 -i ../add-loongarch64-build.patch
   # z3 dependencies
-  install -Dm 644 /usr/lib/libz3* -t dependencies/SymbolicSummaryZ3/os/linux_x86_64
+  install -Dm 644 /usr/lib/libz3* -t dependencies/SymbolicSummaryZ3/os/linux_loongarch_64
   install -Dm 644 /usr/share/java/com.microsoft.z3.jar -t dependencies/flatRepo
 }
 
@@ -119,6 +123,17 @@ _check_dep() {
 }
 
 build() {
+  # Build protoc from source (no prebuilt binary for loongarch64)
+  cd protobuf-${_protobuf_cpp}
+  cmake -B build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -Dprotobuf_BUILD_TESTS=OFF \
+    -Dprotobuf_BUILD_EXAMPLES=OFF \
+    -Dprotobuf_BUILD_PROTOBUF_BINARIES=ON \
+    -DCMAKE_INSTALL_PREFIX="${srcdir}/protobuf-dst"
+  cmake --build build -j$(nproc)
+  cmake --install build
+  cd ..
   cd ghidra-Ghidra_${pkgver}_build
 
   export GRADLE_HOME="$(pwd)/usr/share/java/gradle/"
@@ -133,8 +148,9 @@ build() {
   _build_python_dependency jpype1-${_jpype1} PyGhidra
 
   # Ghidra build
+  export PROTOBUF_DESTDIR="${srcdir}/protobuf-dst"
   gradle yajswDevUnpack
-  gradle buildNatives_linux_x86_64
+  gradle buildNatives_linux_loongarch_64
   gradle sleighCompile
   gradle buildGhidra
 }
@@ -161,7 +177,7 @@ package() {
   install -d "${pkgdir}"/{opt,usr/bin}
   _appver=$(grep -oP '(?<=^application.version=).*$' Ghidra/application.properties)
   _relname=$(grep -oP '(?<=^application.release.name=).*$' Ghidra/application.properties)
-  bsdtar xf "build/dist/ghidra_${_appver}_${_relname}"_*_linux_x86_64.zip -C "${pkgdir}"/opt
+  bsdtar xf "build/dist/ghidra_${_appver}_${_relname}"_*_linux_loongarch_64.zip -C "${pkgdir}"/opt
 
   # Simplify some directory and binary names
   mv "${pkgdir}"/opt/ghidra{_*,}
@@ -170,4 +186,8 @@ package() {
   ln -s /opt/ghidra/support/analyzeHeadless "${pkgdir}"/usr/bin/ghidra-analyzeHeadless
 }
 
+source+=(add-loongarch64-build.patch
+         https://github.com/protocolbuffers/protobuf/releases/download/v${_protobuf_cpp}/protobuf-${_protobuf_cpp}.tar.gz)
+sha512sums+=('a588aaa3570e801d2118ddc285e58ffe0db24dfd134c8ad9c8d4e0c00466689fd9a3ef7b20defdd0d759df7b8c08c34058f9f9d80ee3c4771831006eabf706fa'
+            '818807596b19424b56baa208594c45066d42faa66c1603b420c35e3d01d2e7d9e59ce9300d3d8bcb56d1e6d3580f6e7b3cc8e646385914b4a6552457912c98c5')
 # vim: ts=2 sw=2 et:
