diff --git a/PKGBUILD b/PKGBUILD
index 61a88b7..c0b79a8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -34,6 +34,8 @@ build() {
     -D CMAKE_HIP_ARCHITECTURES="gfx900;gfx906:xnack-;gfx908:xnack-;gfx90a:xnack+;gfx90a:xnack-;gfx940;gfx941;gfx942;gfx1010;gfx1012;gfx1030;gfx1100;gfx1101;gfx1102;gfx1151"
     -D BUILD_DEV=OFF
     -D INSTANCES_ONLY=ON
+    -D CMAKE_C_FLAGS="-mcmodel=medium"
+    -D CMAKE_CXX_FLAGS="-mcmodel=medium"
   )
   cmake "${_cmake_args[@]}"
   cmake --build build
