diff --git a/PKGBUILD b/PKGBUILD
index 5bfdefe..1375ab2 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,12 +19,16 @@ sha256sums=('859b723666eeac7df018209d66045c9853b50b4218cecadb794e2359619ebce7'
 prepare() {
   cd motif-$pkgver
   patch -p1 -i ../no-demos.patch # Don't build demos
+  patch -Np1 -i ../fix_not_literal_format.patch
+  patch -Np1 -i ../add_option_main.patch
+  rm -f tools/wml/wmluiltok.c # force regenerating this file
   touch NEWS AUTHORS
   autoreconf -if
 }
 
 build() {
   cd motif-$pkgver
+  CFLAGS+=" -Wno-implicit-function-declaration"
   ./configure --prefix=/usr
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
@@ -34,3 +38,8 @@ package() {
   cd motif-$pkgver
   make DESTDIR="$pkgdir" install
 }
+
+source+=("fix_not_literal_format.patch"
+         "add_option_main.patch")
+sha256sums+=('23b6750d6c8c4ed3fb9bb448eaaf1f5b008065992b999895b77c12309ded3724'
+             '8a82c4e55d0a7855db50c91fe0daab49a01800d85462e92ffad1e0e4f0ce8d80')
