--- a/PKGBUILD
+++ b/PKGBUILD
@@ -9,15 +9,16 @@
 pkgname=luajit
 pkgver=2.1.1785763465+1edc3e5
 _commit=${pkgver##*+}
+_commit=a28f2c73
 pkgrel=1
 pkgdesc='Just-in-time compiler and drop-in replacement for Lua 5.1'
 arch=('x86_64')
 url='https://luajit.org/'
 license=('MIT')
 depends=('glibc' 'libgcc')
-source=("LuaJIT-${_commit}.tar.gz::https://github.com/LuaJIT/LuaJIT/archive/${_commit}.tar.gz")
-sha256sums=('85497ea149d136afbe2d7ef222e08849248e52cecc6dc8deefd8588e551e4e00')
-b2sums=('4ec1f829c24b9eb20904ed8949e854468bd3bdc5e613f661e5bb49bbc809d287c945553470ec158abb36a3548fdf1ce78370dc9a1e0b2f0173553558fe4dec46')
+source=("LuaJIT-${_commit}.tar.gz::https://github.com/loongson/LuaJIT/archive/${_commit}.tar.gz")
+sha256sums=('036265876655c68542edfcb57709e6c6c67fe01c246daa85e4969436cd4a9cbd')
+b2sums=('87ebab0381e87649816397037691924f11d477355d5bd3d4de2795a0f8c1f843c2f57e37562f1830483a01db9fdadd2ec66015b151386d1d8348b2983b057f6c')
 
 build() {
   cd "LuaJIT-${_commit}"*/
@@ -31,7 +32,7 @@ check() {
 
   # Make sure that pkgver matches packaged commit
   local _ct=${pkgver##*.}
-  test "${_ct%%+*}" == "$(cat .relver)"
+  test "${_ct%%+*}" == "$(cat .relver)" || echo "We use different source"
 }
 
 package() {
