--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,19 +8,20 @@
 
 pkgname=luajit
 # LuaJIT has a "rolling release" where you should follow git HEAD
-_commit=87ae18af97fd4de790bb6c476b212e047689cc93
+_commit=b6deafa04ec9d11cff7f94b8191eaa94053b013d
 # The patch version is the timestamp of the above git commit, obtain via `git show -s --format=%ct`
-_ct=1725453128
-pkgver="2.1.${_ct}"
+_ct=1728455485
+_archct=1725453128
+pkgver="2.1.${_archct}"
 pkgrel=1
 pkgdesc='Just-in-time compiler and drop-in replacement for Lua 5.1'
 arch=('x86_64')
 url='https://luajit.org/'
 license=('MIT')
 depends=('gcc-libs')
-source=("LuaJIT-${_commit}.tar.gz::https://github.com/LuaJIT/LuaJIT/archive/${_commit}.tar.gz")
-sha256sums=('7e34f3aac8cbfacfe8dada50140d4b89d708e0fde60f27ec0643226c2f38ab5f')
-b2sums=('a62203213c5038b75af310bd330c9654840ff180f4fe1f1906a4a65a452489cd2996977c14d4dcd46fafa24c57b7bb69b4e4a026c003916655888d805191e915')
+source=("LuaJIT-${_commit}.tar.gz::https://github.com/setarcos/LuaJIT/archive/${_commit}.tar.gz")
+sha256sums=('aef1b90ddd6a099f7a2e39104a6b9b9012108df2496dc6e6683f0015eab92c3d')
+b2sums=('d83c6f5b290d87492e379ac8becc630dd732724d8d50dc18c4826f6f847a4426e9bb964b54acaa6bcca1b55aaa96a918768edf629fea3afac22c58ea5c3dd491')
 
 build() {
   cd "LuaJIT-${_commit}"
