--- /usr/share/devtools/pacman.conf.d/extra.conf	2024-06-18 20:18:52.000000000 +0800
+++ extra-loong64.conf	2024-08-15 16:44:37.867216368 +0800
@@ -19,7 +19,7 @@
 #XferCommand = /usr/bin/curl -L -C - -f -o %o %u
 #XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
 #CleanMethod = KeepInstalled
-Architecture = auto
+Architecture = loong64
 
 # Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
 #IgnorePkg   =
@@ -72,15 +72,19 @@
 
 #[core-testing]
 #Include = /etc/pacman.d/mirrorlist
+#Server = https://mirrors.pku.edu.cn/loongarch/archlinux/$repo/os/$arch
 
 [core]
-Include = /etc/pacman.d/mirrorlist
+#Include = /etc/pacman.d/mirrorlist
+Server = https://mirrors.pku.edu.cn/loongarch/archlinux/$repo/os/$arch
 
 #[extra-testing]
 #Include = /etc/pacman.d/mirrorlist
+#Server = https://mirrors.pku.edu.cn/loongarch/archlinux/$repo/os/$arch
 
 [extra]
-Include = /etc/pacman.d/mirrorlist
+#Include = /etc/pacman.d/mirrorlist
+Server = https://mirrors.pku.edu.cn/loongarch/archlinux/$repo/os/$arch
 
 # An example of a custom package repository.  See the pacman manpage for
 # tips on creating your own repositories.
