diff --git a/PKGBUILD b/PKGBUILD
index d857e83..b28c19a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,6 +17,8 @@ b2sums=('ae10fc9c4097879fe05ab931c541cdab8a61755210e2f6b084e31a08490a762234ea32d
 
 build() {
   cd $pkgname-$pkgver
+  patch -p1 -i "${srcdir}/add-loongarch64-support.patch"
+  autoreconf -fiv
   CFLAGS+=' -ffat-lto-objects' \
   ./configure \
     --enable-shared \
@@ -33,3 +35,7 @@ check() {
 package() {
   make -C $pkgname-$pkgver DESTDIR="$pkgdir" install
 }
+
+source+=("add-loongarch64-support.patch::https://github.com/polyml/polyml/commit/326b50a52421358c8b7ee4b37f4fed5cd329f1ca.patch")
+sha256sums+=('586cf165984038fa7aced113a901ae5e3b3acce1785d8598a3aad6425904f64a')
+b2sums+=('07900c3c5897b0f21f2a7ce939acdc399affeda953d43e2c71a0d9b845b97a62713cfd4eb928682eca00e5b2781cfb2085cb63599b2f53b75db45747a10b6b6f')
