--- a/PKGBUILD
+++ b/PKGBUILD
@@ -61,6 +61,14 @@ prepare() {
   sed -i '/m64=/s/lib64/lib/' gcc/config/i386/t-linux64
 
   mkdir -p "$srcdir/gcc-build"
+  # Patches for loong64
+  # Fix libdir
+  patch -Np1 -i "$srcdir/gcc-lib64-lib.patch"
+  # Back port fix for __builtin_lsx_vldx
+  patch -Np1 -i "$srcdir/0001-LoongArch-Fix-incorrect-reorder-of-__lsx_vldx-and-__.patch"
+  # Back port Q suffix for __float128
+  patch -Np1 -i "$srcdir/LoongArch-Support-Q-suffix-for-__float128.patch"
+
 }
 
 build() {
@@ -87,6 +95,7 @@ build() {
       --enable-linker-build-id
       --enable-lto
       --disable-multilib
+      --disable-multiarch
       --enable-plugin
       --enable-shared
       --enable-threads=posix
@@ -136,8 +145,8 @@ package_gcc14-libs() {
 
   cd gcc-build
   make -C $CHOST/libgcc DESTDIR="$pkgdir" install-shared
-  mv "${pkgdir}/${_libdir}"/../lib/* "${pkgdir}/${_libdir}"
-  rmdir "${pkgdir}/${_libdir}"/../lib
+  # mv "${pkgdir}/${_libdir}"/../lib/* "${pkgdir}/${_libdir}"
+  # rmdir "${pkgdir}/${_libdir}"/../lib
   rm -f "$pkgdir/$_libdir/libgcc_eh.a"
 
   for lib in libasan.so \
@@ -179,8 +188,8 @@ package_gcc14() {
   install -m755 -t "$pkgdir/${_libdir}/" gcc/{cc1,cc1plus,collect2,lto1,gcov{,-tool}}
 
   make -C $CHOST/libgcc DESTDIR="$pkgdir" install
-  rm -f "${pkgdir}/${_libdir}"/../lib/libgcc_s.so*
-  rmdir "${pkgdir}/${_libdir}"/../lib
+  rm -f "${pkgdir}/${_libdir}"/libgcc_s.so*
+  # rmdir "${pkgdir}/${_libdir}"/../lib
 
   make -C $CHOST/libstdc++-v3/src DESTDIR="$pkgdir" install
   make -C $CHOST/libstdc++-v3/include DESTDIR="$pkgdir" install
@@ -251,3 +260,12 @@ package_gcc14-fortran() {
   ln -s /usr/share/licenses/gcc-libs/RUNTIME.LIBRARY.EXCEPTION \
     "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+source+=("gcc-lib64-lib.patch"
+         # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=119084
+         # Fix https://github.com/cisco/openh264/issues/3857
+         "0001-LoongArch-Fix-incorrect-reorder-of-__lsx_vldx-and-__.patch"
+         "LoongArch-Support-Q-suffix-for-__float128.patch")
+sha256sums+=('d09cbb949364442a78e886b8f55593f07ad17f1e5369ecc83d6c6826015ba22e'
+             '8a4fec2937e22fda73ce549b3f11a70c2bcc343c2dfca2de29217b2708148552'
+             'fabf067e4e5a17480dde6e05af3bdfc5e38e46837538e172d9251b50ec9cc015')
