diff --git a/PKGBUILD b/PKGBUILD
index 6164e19..3b18ae8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -101,6 +101,16 @@ prepare() {
   # Reproducible gcc-ada
   patch -Np0 < "$srcdir/gcc-ada-repro.patch"
 
+  # Patches for loong64
+  # Fix libdir
+  patch -Np1 -i "$srcdir/gcc-lib64-lib.patch"
+  # fix ICE with associating DOT_PROD_EXPR
+  git cherry-pick -n 27b8544e08922d4625a4e2787bd00f36af13d12f
+  # fix missing cpu-features.h
+  git cherry-pick -n 7df57b3b0995562d953b0c231fa28ca8bd5df7c0
+  # fix fixed-point builtins in C++ mode
+  patch -Np1 -i "$srcdir/c-cppbuiltin-avoid-fixed-point-defines-in-cxx.patch"
+
   mkdir -p "$srcdir/gcc-build"
   mkdir -p "$srcdir/libgccjit-build"
 }
@@ -127,7 +137,8 @@ build() {
       --enable-link-serialization=1
       --enable-linker-build-id
       --enable-lto
-      --enable-multilib
+      --disable-multilib
+      --disable-multiarch
       --enable-plugin
       --enable-shared
       --enable-threads=posix
@@ -146,7 +157,7 @@ build() {
   CXXFLAGS=${CXXFLAGS/-Werror=format-security/}
 
   "$srcdir/gcc/configure" \
-    --enable-languages=ada,c,c++,d,fortran,go,lto,m2,objc,obj-c++,rust,cobol \
+    --enable-languages=ada,c,c++,d,fortran,lto,m2,objc,obj-c++,rust,cobol \
     --enable-bootstrap \
     "${_confflags[@]:?_confflags unset}"
 
@@ -238,24 +249,19 @@ package_gcc() {
   install -m755 -t "$pkgdir/$_libdir/" gcc/{cc1,cc1plus,collect2,lto1}
 
   make -C $CHOST/libgcc DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libgcc DESTDIR="$pkgdir" install
 
   make -C $CHOST/libstdc++-v3/src DESTDIR="$pkgdir" install
   make -C $CHOST/libstdc++-v3/include DESTDIR="$pkgdir" install
   make -C $CHOST/libstdc++-v3/libsupc++ DESTDIR="$pkgdir" install
   make -C $CHOST/libstdc++-v3/python DESTDIR="$pkgdir" install
   make -C $CHOST/libatomic DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libstdc++-v3/src DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libstdc++-v3/include DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libstdc++-v3/libsupc++ DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libatomic DESTDIR="$pkgdir" install
 
   make DESTDIR="$pkgdir" install-libcc1
   install -d "$pkgdir/usr/share/gdb/auto-load/usr/lib"
   mv "$pkgdir"/usr/lib/libstdc++.so.6.*-gdb.py \
     "$pkgdir/usr/share/gdb/auto-load/usr/lib/"
-  rm "$pkgdir"/usr/lib{,32}/libstdc++.so*
-  rm "$pkgdir"/usr/lib{,32}/libatomic.so*
+  rm "$pkgdir"/usr/lib/libstdc++.so*
+  rm "$pkgdir"/usr/lib/libatomic.so*
 
   make DESTDIR="$pkgdir" install-fixincludes
   make -C gcc DESTDIR="$pkgdir" install-mkheaders
@@ -272,15 +278,11 @@ package_gcc() {
   make -C $CHOST/libsanitizer/asan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
   make -C $CHOST/libsanitizer/tsan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
   make -C $CHOST/libsanitizer/lsan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
-  make -C $CHOST/32/libgomp DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
-  make -C $CHOST/32/libitm DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
-  make -C $CHOST/32/libsanitizer DESTDIR="$pkgdir" install-nodist_{saninclude,toolexeclib}HEADERS
-  make -C $CHOST/32/libsanitizer/asan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
 
   make -C gcc DESTDIR="$pkgdir" install-man install-info
-  rm "$pkgdir"/usr/share/man/man1/{gccgo,gfortran,lto-dump,gdc,gm2,gcobol}.1
-  rm "$pkgdir"/usr/share/man/man3/gcobol-io.3
-  rm "$pkgdir"/usr/share/info/{gccgo,gfortran,gnat-style,gnat_rm,gnat_ugn,gdc,m2}.info
+  rm -f "$pkgdir"/usr/share/man/man1/{gccgo,gfortran,lto-dump,gdc,gm2,gcobol}.1
+  rm -f "$pkgdir"/usr/share/man/man3/gcobol-io.3
+  rm -f "$pkgdir"/usr/share/info/{gccgo,gfortran,gnat-style,gnat_rm,gnat_ugn,gdc,m2}.info
 
   make -C libcpp DESTDIR="$pkgdir" install
   make -C gcc DESTDIR="$pkgdir" install-po
@@ -291,7 +293,7 @@ package_gcc() {
   # create cc-rs compatible symlinks
   # https://github.com/rust-lang/cc-rs/blob/1.0.73/src/lib.rs#L2578-L2581
   for binary in {c++,g++,gcc,gcc-ar,gcc-nm,gcc-ranlib}; do
-    ln -s /usr/bin/$binary "$pkgdir"/usr/bin/x86_64-linux-gnu-$binary
+    ln -s /usr/bin/$binary "$pkgdir"/usr/bin/$(uname -m)-linux-gnu-$binary
   done
 
   # POSIX conformance launcher scripts for c89 and c99
@@ -302,8 +304,7 @@ package_gcc() {
   make -C $CHOST/libstdc++-v3/doc DESTDIR="$pkgdir" doc-install-man
 
   # remove files provided by lib32-gcc-libs and libgcc
-  rm -f "$pkgdir"/usr/lib{,32}/libgcc_s{,_asneeded}.so*
-  rm -f "$pkgdir"/usr/lib32/libstdc++.so
+  rm -f "$pkgdir"/usr/lib/libgcc_s{,_asneeded}.so*
 
   # byte-compile python libraries
   python -m compileall "$pkgdir/usr/share/gcc-${pkgver%%+*}/"
@@ -344,10 +345,6 @@ package_gcc-ada() {
   make DESTDIR="$pkgdir" INSTALL="install" \
     INSTALL_DATA="install -m644" install-libada
 
-  cd "$srcdir"/gcc-build/$CHOST/32/libada
-  make DESTDIR="$pkgdir" INSTALL="install" \
-    INSTALL_DATA="install -m644" install-libada
-
   ln -s gcc "$pkgdir/usr/bin/gnatgcc"
 
   # insist on dynamic linking, but keep static libraries because gnatmake complains
@@ -356,12 +353,6 @@ package_gcc-ada() {
   ln -s libgnat-${pkgver%%.*}.so "$pkgdir/usr/lib/libgnat.so"
   rm -f "$pkgdir"/$_libdir/adalib/libgna{rl,t}.so
 
-  install -d "$pkgdir/usr/lib32/"
-  mv "$pkgdir"/$_libdir/32/adalib/libgna{rl,t}-${pkgver%%.*}.so "$pkgdir/usr/lib32"
-  ln -s libgnarl-${pkgver%%.*}.so "$pkgdir/usr/lib32/libgnarl.so"
-  ln -s libgnat-${pkgver%%.*}.so "$pkgdir/usr/lib32/libgnat.so"
-  rm -f "$pkgdir"/$_libdir/32/adalib/libgna{rl,t}.so
-
   _install_runtime_library_exception
 }
 
@@ -397,7 +388,6 @@ package_gcc-d() {
 
   make -C $CHOST/libphobos DESTDIR="$pkgdir" install
   rm -f "$pkgdir/usr/lib/"lib{gphobos,gdruntime}.so*
-  rm -f "$pkgdir/usr/lib32/"lib{gphobos,gdruntime}.so*
 
   _install_runtime_library_exception
 }
@@ -425,8 +415,6 @@ package_gcc-fortran() {
   cd gcc-build
   make -C $CHOST/libgfortran DESTDIR="$pkgdir" install-cafexeclibLTLIBRARIES \
     install-{toolexeclibDATA,nodist_fincludeHEADERS,gfor_cHEADERS}
-  make -C $CHOST/32/libgfortran DESTDIR="$pkgdir" install-cafexeclibLTLIBRARIES \
-    install-{toolexeclibDATA,nodist_fincludeHEADERS,gfor_cHEADERS}
   make -C $CHOST/libgomp DESTDIR="$pkgdir" install-nodist_fincludeHEADERS
   make -C gcc DESTDIR="$pkgdir" fortran.install-{common,man,info}
   install -Dm755 gcc/f951 "$pkgdir/$_libdir/f951"
@@ -455,7 +443,7 @@ package_gcc-gcobol() {
 
   install -Dm755 gcc/gcobol "$pkgdir"/usr/bin/gcobol
 
-  install -Dm644 $CHOST/libgcobol/libgcobol.spec "$pkgdir"/$_libdir/
+#  install -Dm644 $CHOST/libgcobol/libgcobol.spec "$pkgdir"/$_libdir/
 
   _install_runtime_library_exception
 }
@@ -486,11 +474,10 @@ package_gcc-go() {
 
   cd gcc-build
   make -C $CHOST/libgo DESTDIR="$pkgdir" install-exec-am
-  make -C $CHOST/32/libgo DESTDIR="$pkgdir" install-exec-am
   make DESTDIR="$pkgdir" install-gotools
   make -C gcc DESTDIR="$pkgdir" go.install-{common,man,info}
 
-  rm -f "$pkgdir"/usr/lib{,32}/libgo.so*
+  rm -f "$pkgdir"/usr/lib/libgo.so*
   install -Dm755 gcc/go1 "$pkgdir/$_libdir/go1"
 
   _install_runtime_library_exception
@@ -619,9 +606,7 @@ package_lib32-gcc-libs() {
   rm -f "$pkgdir/$_libdir/32/libgcc_eh.a"
 
   for lib in libatomic \
-             libgcobol \
              libgfortran \
-             libgo \
              libgomp \
              libitm \
              libquadmath \
@@ -724,8 +709,8 @@ package_libgcobol() {
     libgcobol.so
   )
 
-  cd gcc-build
-  make -C $CHOST/libgcobol DESTDIR="$pkgdir" install-toolexeclibLTLIBRARIES
+#  cd gcc-build
+#  make -C $CHOST/libgcobol DESTDIR="$pkgdir" install-toolexeclibLTLIBRARIES
 
   _install_runtime_library_exception
 }
@@ -797,9 +782,6 @@ package_libgm2() {
 
   make -C $CHOST/libgm2 DESTDIR="$pkgdir" install
 
-  # remove files provided by lib32-gcc-libs
-  rm -rf "$pkgdir"/usr/lib32/
-
   _install_runtime_library_exception
 }
 
@@ -820,9 +802,6 @@ package_libgphobos() {
   rm -rf "$pkgdir"/$_libdir/include/d/
   rm -f "$pkgdir"/usr/lib/libgphobos.spec
 
-  # remove files provided by lib32-gcc-libs
-  rm -rf "$pkgdir"/usr/lib32/
-
   _install_runtime_library_exception
 }
 
@@ -875,9 +854,6 @@ package_libobjc() {
   cd gcc-build
   make -C $CHOST/libobjc DESTDIR="$pkgdir" install-libs
 
-  # remove files provided by lib32-gcc-libs
-  rm -rf "$pkgdir"/usr/lib32/
-
   _install_runtime_library_exception
 }
 
@@ -970,3 +946,10 @@ package_lto-dump() {
 
   _install_runtime_library_exception
 }
+
+source+=("gcc-lib64-lib.patch"
+         "c-cppbuiltin-avoid-fixed-point-defines-in-cxx.patch")
+sha256sums+=('5e8717aae7b83ed219e44fdba8379de7fb02c7c776ae6379ab26a6b22a7878d1'
+             '00590887140f27396828b50f0bda0e0db5df8e3c86341d130f6b12788985d7f2')
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(lib32-glibc|lib32-gcc-libs)$'))
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '^(lib32-gcc-libs|gcc-go|libgo)$'))
