--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,7 +26,7 @@ depends=(boost-libs coin fmt glew libspnav med
          python-markdown python-matplotlib python-pip python-pivy python-ply
          python-yaml qt6-svg qt6-tools verdict xerces-c)
 makedepends=(boost cgns cmake eigen git libharu liblas ninja nlohmann-json
-             openvdb openvr ospray pdal postgresql-libs python-mpi4py shiboken6
+             openvdb openvr pdal postgresql-libs python-mpi4py shiboken6
              swig utf8cpp dos2unix microsoft-gsl fast_float)
 optdepends=('graphviz: dependency graph support'
             'opencamlib: CAM workbench support'
