--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,8 +25,11 @@ source=(git+https://github.com/zapping-vbi/zvbi.git#tag=${_tag})
 b2sums=(SKIP)
 validpgpkeys=(FA26CA784BE188927F22B99F6570EA01146F7354) # Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
 
+source+=(sincos-function-check.patch)
+b2sums+=(46d39ae99561608e15bbca2e1bd3cf43b92ba2e575c718077f7cb763db0c6d3a457d6d3782e25adf6a813fad5cea19d410b06f5b8d4ba205f4652275ee364e65)
 prepare() {
   cd zvbi
+  patch -Np1 -i ../sincos-function-check.patch
   NOCONFIGURE=1 ./autogen.sh
 }
 
