diff --git a/PKGBUILD b/PKGBUILD
index be8780c..54ea350 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -77,6 +77,7 @@ check() {
     --ignore tests/test_proxy_connect.py
     # TODO: raise upstream
     --deselect tests/test_crawl.py::CrawlTestCase::test_start_requests_laziness
+    --deselect tests/test_utils_python.py::UtilsPythonTestCase::test_get_func_args
   )
   local site_packages=$(python -c "import site; print(site.getsitepackages()[0])")
 
