--- a/PKGBUILD
+++ b/PKGBUILD
@@ -93,6 +93,7 @@ check() {
     --deselect tests/test_crawler.py::TestCrawlerProcessSubprocess::test_asyncio_enabled_reactor_different_loop
     --deselect tests/test_pipeline_files.py::TestFilesPipeline::test_file_path
     --deselect tests/test_utils_asyncio.py::TestAsyncio::test_install_asyncio_reactor
+    --deselect tests/test_utils_python.py::UtilsPythonTestCase::test_get_func_args
   )
   local site_packages=$(python -c "import site; print(site.getsitepackages()[0])")
 
