diff --git a/PKGBUILD b/PKGBUILD
index ab5dd31..115f83e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -71,6 +71,9 @@ prepare() {
 
   cd binutils-gdb
 
+  # Backport fix for loong64
+  git cherry-pick -n 3eede6b04a30243fdc4acece156be95b17c16f83
+
   # Turn off development mode (-Werror, gas run-time checks, date in sonames)
   sed -i '/^development=/s/true/false/' bfd/development.sh
 
@@ -96,7 +99,7 @@ build() {
     --enable-colored-disassembly \
     --enable-default-execstack=no \
     --enable-deterministic-archives \
-    --enable-gold \
+    --disable-gold \
     --enable-install-libiberty \
     --enable-jansson \
     --enable-ld=default \
@@ -105,7 +108,6 @@ build() {
     --enable-plugins \
     --enable-relro \
     --enable-shared \
-    --enable-targets=x86_64-pep,bpf-unknown-none \
     --enable-threads \
     --disable-gdb \
     --disable-gdbserver \
