--- a/PKGBUILD
+++ b/PKGBUILD
@@ -74,6 +74,9 @@ prepare() {
 
   cd binutils-gdb
 
+  # Backport fix for loong64
+  git cherry-pick -n 3eede6b04a30243fdc4acece156be95b17c16f83
+
   # Turn off development mode (-Werror, gas run-time checks, date in sonames)
   sed -i '/^development=/s/true/false/' bfd/development.sh
 
@@ -82,6 +85,7 @@ prepare() {
   # unsupported targets.  This allows the binutils to be built with
   # BPF support enabled.
   patch -Np1 < ../gold-warn-unsupported.patch
+  patch -Np1 < ../0001-LoongArch-Restore-pointer_equality_needed-for-GOT-da.patch
 }
 
 build() {
@@ -96,7 +100,7 @@ build() {
     --enable-colored-disassembly \
     --enable-default-execstack=no \
     --enable-deterministic-archives \
-    --enable-gold \
+    --disable-gold \
     --enable-install-libiberty \
     --enable-jansson \
     --enable-ld=default \
@@ -105,7 +109,6 @@ build() {
     --enable-plugins \
     --enable-relro \
     --enable-shared \
-    --enable-targets=x86_64-pep,bpf-unknown-none \
     --enable-threads \
     --disable-gdb \
     --disable-gdbserver \
@@ -165,3 +168,6 @@ EOS
   # install FSF All Permissive License
   install -Dm644 -t "${pkgdir}"/usr/share/licenses/${pkgname}/ FSFAP
 }
+
+source+=(0001-LoongArch-Restore-pointer_equality_needed-for-GOT-da.patch)
+sha256sums+=('ea3002ab39a59c4e48c1f91b0205fb5f226aa96f5039ecc440ca5b5e684613a3')
