--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,3 +46,14 @@ package() {
 
   make DESTDIR="${pkgdir}" install
 }
+
+prepare() {
+  cd "${srcdir}"/${pkgname}-${pkgver}
+  patch -p1 -i "${srcdir}/add-loong64-support.patch"
+  cp /usr/share/automake-1.1?/config.sub .
+  cp /usr/share/automake-1.1?/config.guess .
+}
+
+source+=("add-loong64-support.patch")
+sha512sums+=('fdeed54b83cb97d12fc5e00300f64c87485c5aecde1e2e14ffdfe53bbf934d8ab2ca9537c4678a269cd5c949803d0e0ee163bb74c89838f8e9e376361eaa76f9')
+b2sums+=('7de6918ba07147ab47ba9b2dce90868612099000311a40c9b079a6dd5fd9a4303e77f0225bf8d218aaa01838fad4745445f0c50524d0b779adb0b0a2620534eb')
