diff --git a/PKGBUILD b/PKGBUILD
index 7ad6923..d69c6f6 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,7 +16,7 @@ b2sums=('SKIP')
 prepare() {
   sed -i 's/-shared -Wl/-shared '"$LDFLAGS"'/g' $pkgname/Makefile
   setconf $pkgname/Makefile \
-    CFLAGS="$CFLAGS -shared -fPIC -fvisibility=hidden -fomit-frame-pointer -w"
+    CFLAGS="$CFLAGS -shared -fPIC -fvisibility=hidden -fomit-frame-pointer -w  -Wno-implicit-function-declaration"
 }
 
 build() {
