--- a/PKGBUILD
+++ b/PKGBUILD
@@ -72,6 +72,7 @@ case "${CARCH}" in
   i686)   _JARCH='x86';;
   riscv64)_JARCH='riscv64';;
   aarch64) _JARCH='aarch64';;
+  loong64) _JARCH='loongarch64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -456,4 +457,14 @@ package_openjdk-doc() {
   ln -s ${pkgbase} "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+# We cannot use upstream's version control
+# Reset some versions
+_loongfixver=21
+pkgver=${_majorver}.u${_loongfixver}ls1
+_jdkdir=jdk-jdk-${_majorver}-${_loongfixver}-ls-1
+_imgdir=${_jdkdir}/build/linux-${_JARCH}-server-release/images
+
+source+=("https://github.com/loongson/jdk/archive//refs/tags/jdk-${_majorver}+${_loongfixver}-ls-1.tar.gz")
+sha256sums+=('1394ffabc979335b2d97c195b571714089aa5a582ed68831dd8caf140ac75528')
+
 # vim: ts=2 sw=2 et:
