diff --git a/PKGBUILD b/PKGBUILD
index b5866d1..7240c88 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -65,7 +65,10 @@ build() {
   export GOFLAGS="-buildmode=pie -trimpath -ldflags=-linkmode=external -mod=readonly -modcacherw"
 
   cd gitlab-runner
-  make out/binaries/gitlab-runner-linux-amd64
+  go mod edit -replace=github.com/cilium/ebpf=github.com/cilium/ebpf@v0.16.0
+  go mod tidy
+  export CGO_ENABLED=1
+  make out/binaries/gitlab-runner-linux-loong64
 }
 
 package() {
@@ -75,7 +78,7 @@ package() {
   install -Dm644 "${srcdir}/gitlab-runner.service" "${pkgdir}/usr/lib/systemd/system/gitlab-runner.service"
   install -Dm644 "${srcdir}/gitlab-runner.sysusers" "${pkgdir}/usr/lib/sysusers.d/gitlab-runner.conf"
   install -Dm644 "${srcdir}/gitlab-runner.tmpfiles" "${pkgdir}/usr/lib/tmpfiles.d/gitlab-runner.conf"
-  install -Dm755 out/binaries/gitlab-runner-linux-amd64 "${pkgdir}/usr/bin/gitlab-runner"
+  install -Dm755 out/binaries/gitlab-runner-linux-loong64 "${pkgdir}/usr/bin/gitlab-runner"
   install -Dm755 packaging/root/usr/share/gitlab-runner/clear-docker-cache "${pkgdir}/usr/share/gitlab-runner/clear-docker-cache"
 
   # Move prebuilt Docker images to hard-coded canonical location
