diff --git a/PKGBUILD b/PKGBUILD
index 9f7989e..c24a938 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,6 +27,8 @@ options=(!lto)
 
 prepare() {
     patch -d "${_dirname}" -Np1 < "${srcdir}/hsa-rocr-6.4-fix-missing-include.patch"
+    patch -d "${_dirname}" -Np1 < "${srcdir}/rocr-unsupport.patch"
+    export CC=clang CXX=clang++
 }
 
 build() {
@@ -49,3 +51,9 @@ package() {
   DESTDIR="$pkgdir" cmake --install build
   install -Dm644 "$_dirname/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+# Based on https://github.com/felixonmars/archriscv-packages/pull/4749
+source+=("rocr-unsupport.patch")
+sha256sums+=('8e22758297071f239929dd746d293d6b76744fc6e698644a2a9d4c8d63ad570d')
+# gcc doesn't support `mm_malloc.h` on loong64
+makedepends+=(clang)
