--- a/PKGBUILD
+++ b/PKGBUILD
@@ -24,6 +24,10 @@ sha256sums=('a700912d102b35c9336b7352d09d56b9d19e4957daf53779983c3d00ce744c9d')
 _dirname="$(basename "$url")-rocm-$pkgver"/projects/rocr-runtime/
 options=(!lto)
 
+prepare() {
+    patch -d "${_dirname}" -Np1 < "${srcdir}/rocr-unsupport.patch"
+    export CC=clang CXX=clang++
+}
 build() {
   # Silence warnings on optional libraries with -DNDEBUG,
   # https://github.com/RadeonOpenCompute/ROCR-Runtime/issues/89#issuecomment-613788944
@@ -44,3 +48,9 @@ package() {
   DESTDIR="$pkgdir" cmake --install build
   install -Dm644 "$_dirname/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+# Based on https://github.com/felixonmars/archriscv-packages/pull/4749
+source+=("rocr-unsupport.patch")
+sha256sums+=('a725b4bee0e9c6f5d7cd71fdf942b84703f5f9bb19333aad82ab97a3a3c9a7a4')
+# gcc doesn't support `mm_malloc.h` on loong64
+makedepends+=(clang)
