--- a/src/compiler/loongarch64/vm.lisp
+++ b/src/compiler/loongarch64/vm.lisp
@@ -82,7 +82,7 @@
 
   (defregset non-descriptor-regs ra nl0 nl1 nl2 nl3 nl4 nl5 nargs nfp cfunc)
-  (defregset descriptor-regs a0 a1 a2 a3 a4 l0 l1 #-sb-thread l2 lexenv ocfp)
+  (defregset descriptor-regs a0 a1 a2 a3 a4 a5 l0 l1 #-sb-thread l2 lexenv ocfp)
   (defregset reserve-descriptor-regs lexenv)
   (defregset reserve-non-descriptor-regs cfunc)
   (defregset boxed-regs a0 a1 a2 a3 a4 a5 l0 l1 #-sb-thread l2 ocfp lexenv code)
