diff --git a/PKGBUILD b/PKGBUILD
index 8718e3a..0df0fe3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -28,8 +28,6 @@ makedepends=(
   graphviz
   texlive-bin
   texlive-latexextra
-  # -cuda
-  cuda
   # -hip
   hip-runtime-amd
   hipblas
@@ -66,7 +64,7 @@ build() {
     -DGINKGO_BUILD_BENCHMARKS=ON
     -DGINKGO_BUILD_EXAMPLES=ON
     -DGINKGO_BUILD_DOC=ON
-    -DGINKGO_BUILD_TESTS=ON
+    -DGINKGO_BUILD_TESTS=OFF # Workaround for CMake error with misssing test name
   )
   # In general, we want to list all real archs (sm_XX) and the latest virtual arch (compute_XX) for future PTX compatibility.
   # Valid values can be discovered from nvcc --help
@@ -81,6 +79,7 @@ build() {
     -DGINKGO_BUILD_SYCL=OFF
   cmake --build build
 
+  : <<COMMENT_SEPARATOR
   # -cuda package
   # Compile source code for supported GPU archs in parallel
   export CUDAFLAGS="--threads $(nproc)"
@@ -90,6 +89,7 @@ build() {
     -DGINKGO_BUILD_HIP=OFF \
     -DGINKGO_BUILD_SYCL=OFF
   cmake --build build-cuda
+COMMENT_SEPARATOR
 
   # -hip package
   # LTO does not work with HIP
@@ -192,4 +192,6 @@ package_ginkgo-hpc-hip() {
   _package -hip
 }
 
+pkgname=(${pkgname[@]/ginkgo-hpc-cuda})
+
 # vim:set ts=2 sw=2 et:
