--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,7 +21,7 @@ makedepends=(
   'qt6-base'
   'qt6-declarative'
   'python'
-  'pandoc'
+  'asciidoctor'
   'cmake'
   'clang'
 )
@@ -53,9 +53,9 @@ build() {
   pushd "$pkgname"
   xsversion=$(pacman -Qi docbook-xsl | grep ^Version | sed -e 's/.*: //' -e 's/-[0-9]$//')
   make DB2MAN=/usr/share/xml/docbook/xsl-stylesheets-${xsversion}-nons/manpages/docbook.xsl man
-  pandoc man/manual.md -o man/manual.html -s --number-sections --toc
-  pandoc man/reference-cfg-format.md -o man/reference-cfg-format.html -s --number-sections --toc
-  pandoc man/writing-addons.md -o man/writing-addons.html -s --number-sections --toc
+  asciidoctor man/manual.md -o man/manual.html -a toc
+  asciidoctor man/reference-cfg-format.md -o man/reference-cfg-format.html -a toc
+  asciidoctor man/writing-addons.md -o man/writing-addons.html -a toc
   popd
 
   cd "$pkgname"
