diff --git a/PKGBUILD b/PKGBUILD
index 1d765f3..29d84be 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -91,6 +91,14 @@ build() {
   echo kernel/x86/microcode/AuthenticAMD.bin |
     bsdtar --uid 0 --gid 0 -cnf - -T - |
     bsdtar --null -cf - --format=newc @- > amd-ucode.img
+
+  # Add gsgpu firmware
+  echo 'File: loongson/lg100_cp.bin' >> "${pkgbase}/WHENCE"
+  install -Dm644 "${srcdir}/lg100_cp.bin" -t "${pkgbase}/loongson"
+  # It use git file to find firmwares
+  # https://gitlab.com/kernel-firmware/linux-firmware/-/blob/e1d95775eb3db64ab3740701bf9682e5856aab2e/check_whence.py#L157
+  cd "${pkgbase}"
+  git add "loongson/lg100_cp.bin"
 }
 
 _pick() {
@@ -390,4 +398,9 @@ package_linux-firmware-realtek() {
     -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '^(amd-ucode)$'))
+source+=("lg100_cp.bin::https://github.com/cl91/linux-gsgpu-archpkg/raw/master/gsgpu-firmware/lg100_cp.bin")
+b2sums+=('68301cdbdbb7b03c8a8c3c051ad415f2e6313306ea725664628a6606c049b1f69200464ef16fc74e0462c9b9ac926342646e7518f90a6424ad82f240925d903a')
+noextract=('lg100_cp.bin')
+
 # vim:set sw=2 sts=-1 et:
