--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,7 +20,9 @@ build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
   CFLAGS+=' -ffat-lto-objects'
   CXXFLAGS+=' -ffat-lto-objects'
-  ./configure --prefix /usr --mandir /usr/share/man -enable-frame-pointers
+  patch -Np1 -i "${srcdir}/add-loongarch-architecture-support.patch"
+  autoconf -fiv
+  ./configure --prefix /usr --mandir /usr/share/man
   make --debug=v world.opt
 }
 
@@ -53,3 +55,6 @@ optdepends=()
   install -m755 -d "${pkgdir}/usr/share/licenses/${pkgname}"
   install -m644 LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/"
 }
+
+source+=("add-loongarch-architecture-support.patch::https://github.com/loongson/ocaml/commit/799ae3a9f3e54e9ec399e31ce5ecfe10b2034d99.patch")
+sha256sums+=('e6126aa4dea48b09f4f7bfa4ce0e8f2f2688199267e4e68f12b798694317180e')
