diff --git a/PKGBUILD b/PKGBUILD
index 12d6b0a..96362e8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,7 +20,9 @@ build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
   CFLAGS+=' -ffat-lto-objects'
   CXXFLAGS+=' -ffat-lto-objects'
-  ./configure --prefix /usr --mandir /usr/share/man -enable-frame-pointers
+  patch -Np1 -i "${srcdir}/add-loongarch-architecture-support.patch"
+  autoconf -fiv
+  ./configure --prefix /usr --mandir /usr/share/man
   make --debug=v world.opt
 }
 
@@ -53,3 +55,6 @@ optdepends=()
   install -m755 -d "${pkgdir}/usr/share/licenses/${pkgname}"
   install -m644 LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/"
 }
+
+source+=("add-loongarch-architecture-support.patch")
+sha256sums+=('4af2f8e7dd7e91e252a6bd5c5c1ed53c2c266056bf675e772417f555d7b3a0f9')
