diff --git a/PKGBUILD b/PKGBUILD
index 0001e6e..8cc0710 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,6 +37,7 @@ prepare() {
 
   git -c protocol.file.allow=always submodule update --init --recursive
 
+  patch -Np1 -d "${srcdir}/${pkgname}-rocmlir" -i "${srcdir}/fix-mlir-llvm-ConstantFolding-ConstantFoldFP128-type.patch"
   # Disable tests as they require an AMD GPU at build time
   sed -i '/add_subdirectory(test)/d' CMakeLists.txt
 }
@@ -97,3 +98,7 @@ package() {
 
   install -Dm644 "$pkgname/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+# I don't known why upstream llvm doesn't need it
+source+=("fix-mlir-llvm-ConstantFolding-ConstantFoldFP128-type.patch")
+sha256sums+=('9c9d1fba221dcbf8abb00ed55dd1482d5330073a3239a7ca4be0861597acecfc')
