diff --git a/PKGBUILD b/PKGBUILD
index 0001e6e..acd1f74 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,6 +37,7 @@ prepare() {
 
   git -c protocol.file.allow=always submodule update --init --recursive
 
+  patch -Np1 -d "${srcdir}/${pkgname}" -i "${srcdir}/miopen-loong64-support.patch"
   # Disable tests as they require an AMD GPU at build time
   sed -i '/add_subdirectory(test)/d' CMakeLists.txt
 }
@@ -97,3 +98,6 @@ package() {
 
   install -Dm644 "$pkgname/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=("miopen-loong64-support.patch::https://raw.githubusercontent.com/loongarch-moe/rocm-loongarch/228273601fa054e40056547339012e6d698c2855/stage5/5.rocm-MIOpen/miopen.patch")
+sha256sums+=('4dcc44677baa71f1d6d0fc7f371d3fd55a05c4db1b55ff654df471bdfff536f5')
