diff --git a/PKGBUILD b/PKGBUILD
index 8daf8c8..be83008 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -119,9 +119,8 @@ link-shared = true
 [build]
 description = "Arch Linux $pkgbase $epoch:$pkgver-$pkgrel"
 target = [
-  "x86_64-unknown-linux-gnu",
-  "i686-unknown-linux-gnu",
-  "x86_64-unknown-linux-musl",
+  "$(uname -m)-unknown-linux-gnu",
+  "$(uname -m)-unknown-linux-musl",
   "aarch64-unknown-linux-gnu",
   "aarch64-unknown-linux-musl",
   "wasm32-unknown-unknown",
@@ -172,24 +171,20 @@ jemalloc = true
 compression-formats = ["gz"]
 compression-profile = "fast"
 
-[target.x86_64-unknown-linux-gnu]
+[target.$(uname -m)-unknown-linux-gnu]
 cc = "/usr/bin/gcc"
 cxx = "/usr/bin/g++"
 ar = "/usr/bin/gcc-ar"
 ranlib = "/usr/bin/gcc-ranlib"
 llvm-config = "/usr/bin/llvm-config"
 
-[target.i686-unknown-linux-gnu]
+# Failed with musl-gcc: ld.lld: error: unable to find library -lgcc_s
+# Switch from musl-gcc to gcc
+[target.$(uname -m)-unknown-linux-musl]
 cc = "/usr/bin/gcc"
 cxx = "/usr/bin/g++"
 ar = "/usr/bin/gcc-ar"
 ranlib = "/usr/bin/gcc-ranlib"
-
-[target.x86_64-unknown-linux-musl]
-cc = "/usr/bin/musl-gcc"
-cxx = "/usr/bin/g++"
-ar = "/usr/bin/gcc-ar"
-ranlib = "/usr/bin/gcc-ranlib"
 sanitizers = false
 musl-root = "/usr/lib/musl"
 
@@ -304,12 +299,9 @@ build() {
 
   # rustbuild always installs copies of the shared libraries to /usr/lib,
   # overwrite them with symlinks to the per-architecture versions
-  mkdir -pv usr/lib32
-  ln -srvft usr/lib   usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/*.so
-  ln -srvft usr/lib32 usr/lib/rustlib/i686-unknown-linux-gnu/lib/*.so
+  ln -srvft usr/lib   usr/lib/rustlib/$(uname -m)-unknown-linux-gnu/lib/*.so
 
-  _pick dest-i686 usr/lib/rustlib/i686-unknown-linux-gnu usr/lib32
-  _pick dest-musl usr/lib/rustlib/x86_64-unknown-linux-musl
+  _pick dest-musl usr/lib/rustlib/$(uname -m)-unknown-linux-musl
   _pick dest-aarch64-gnu usr/lib/rustlib/aarch64-unknown-linux-gnu
   _pick dest-aarch64-musl usr/lib/rustlib/aarch64-unknown-linux-musl
   _pick dest-wasm usr/lib/rustlib/wasm32{,v1}-*
@@ -418,4 +410,7 @@ package_rust-src() {
     rustc-$pkgver-src/{COPYRIGHT,LICENSE-MIT}
 }
 
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '^(lib32-rust-libs)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(lib32-gcc-libs|lib32-glibc)$'))
+
 # vim:set ts=2 sw=2 et:
