--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,9 +7,6 @@
 pkgbase=rust
 pkgname=(
   rust
-  lib32-rust-libs
-  rust-musl
-  rust-wasm
   rust-src
 )
 epoch=1
@@ -37,18 +34,13 @@ depends=(
 makedepends=(
   clang
   cmake
-  lib32-gcc-libs
-  lib32-glibc
   libffi
   lld
   llvm
-  musl
   ninja
   perl
   python
   rust
-  wasi-libc
-  wasm-component-ld
 )
 checkdepends=(
   gdb
@@ -99,14 +91,7 @@ link-shared = true
 
 [build]
 target = [
-  "x86_64-unknown-linux-gnu",
-  "i686-unknown-linux-gnu",
-  "x86_64-unknown-linux-musl",
-  "wasm32-unknown-unknown",
-  "wasm32-wasi",
-  "wasm32-wasip1",
-  "wasm32-wasip1-threads",
-  "wasm32-wasip2",
+  "`uname -m`-unknown-linux-gnu",
 ]
 cargo = "/usr/bin/cargo"
 rustc = "/usr/bin/rustc"
@@ -148,46 +133,13 @@ jemalloc = true
 [dist]
 compression-formats = ["gz"]
 
-[target.x86_64-unknown-linux-gnu]
+[target.`uname -m`-unknown-linux-gnu]
 cc = "/usr/bin/gcc"
 cxx = "/usr/bin/g++"
 ar = "/usr/bin/gcc-ar"
 ranlib = "/usr/bin/gcc-ranlib"
 llvm-config = "/usr/bin/llvm-config"
 
-[target.i686-unknown-linux-gnu]
-cc = "/usr/bin/gcc"
-cxx = "/usr/bin/g++"
-ar = "/usr/bin/gcc-ar"
-ranlib = "/usr/bin/gcc-ranlib"
-
-[target.x86_64-unknown-linux-musl]
-sanitizers = false
-musl-root = "/usr/lib/musl"
-
-[target.wasm32-unknown-unknown]
-sanitizers = false
-profiler = false
-
-[target.wasm32-wasi]
-sanitizers = false
-profiler = false
-wasi-root = "/usr/share/wasi-sysroot"
-
-[target.wasm32-wasip1]
-sanitizers = false
-profiler = false
-wasi-root = "/usr/share/wasi-sysroot"
-
-[target.wasm32-wasip1-threads]
-sanitizers = false
-profiler = false
-wasi-root = "/usr/share/wasi-sysroot"
-
-[target.wasm32-wasip2]
-sanitizers = false
-profiler = false
-wasi-root = "/usr/share/wasi-sysroot"
 END
 }
 
@@ -229,13 +181,8 @@ build() {
 
   # rustbuild always installs copies of the shared libraries to /usr/lib,
   # overwrite them with symlinks to the per-architecture versions
-  mkdir -pv usr/lib32
-  ln -srvft usr/lib   usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/*.so
-  ln -srvft usr/lib32 usr/lib/rustlib/i686-unknown-linux-gnu/lib/*.so
+  ln -srvft usr/lib   usr/lib/rustlib/`uname -m`-unknown-linux-gnu/lib/*.so
 
-  _pick dest-i686 usr/lib/rustlib/i686-unknown-linux-gnu usr/lib32
-  _pick dest-musl usr/lib/rustlib/x86_64-unknown-linux-musl
-  _pick dest-wasm usr/lib/rustlib/wasm32-*
   _pick dest-src  usr/lib/rustlib/src
 }
 
