--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,8 +14,18 @@ depends=(gcc-libs sh)
 source=(https://download.oracle.com/berkeley-db/db-${pkgver}.tar.gz)
 b2sums=('d3254fba1300d6c7dda5d872c5fa349e8704d71be9c4d37ccf3357782adeb879ce30e066cf083bd97554405c2e94071a96cee16090ad9f797a4fa657ff7e8c46')
 
+source+=('add_cstddef.patch')
+b2sums+=('3246b3155a251a9a4fce43a9b75be7e8120673a15c7edebe60ec0d942daa1a9f0adff04e324232385a08b351300c61eefbf71253f801577d905f27e8b9ae320b')
+
+prepare() {
+  cd ${pkgname}-${pkgver}
+  patch -Np1 -i ../add_cstddef.patch
+}
+
 build() {
-  cd ${pkgname}-${pkgver}/build_unix
+  cd ${pkgname}-${pkgver}
+  cp /usr/share/autoconf/build-aux/config.{guess,sub} dist/
+  cd build_unix
   ../dist/configure \
     --prefix=/usr \
     --enable-compat185 \
