--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,13 +21,16 @@ prepare() {
 
   # fix dbstl_container build error
   patch -Np1 -i ../db-4.8.30-tls-configure.patch
+  patch -Np1 -i ../add_cstddef.patch
 
   cd dist
   ./s_config
 }
 
 build() {
-  cd ${pkgname}-${pkgver}/build_unix
+  cd ${pkgname}-${pkgver}
+  cp /usr/share/autoconf/build-aux/config.{guess,sub} dist/
+  cd build_unix
   export CFLAGS+=" -std=gnu99"
   ../dist/configure \
     --prefix=/usr \
@@ -44,3 +47,6 @@ package() {
   make -C ${pkgname}-${pkgver}/build_unix DESTDIR="${pkgdir}" install
   rm -r "${pkgdir}"/usr/docs
 }
+
+source+=('add_cstddef.patch')
+b2sums+=('3246b3155a251a9a4fce43a9b75be7e8120673a15c7edebe60ec0d942daa1a9f0adff04e324232385a08b351300c61eefbf71253f801577d905f27e8b9ae320b')
