--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,8 +13,12 @@ checkdepends=('python-hypothesis' 'python-pytest')
 source=("git+https://github.com/mehcode/python-xmlsec.git#tag=$pkgver")
 sha512sums=('809168e0b2501a3220640a2f092e38ee749636e4daeed5ba9eb247e0ef88aeedc3cd9c08d54db1a3a073303b940c592d4d0a9a6df4999fa55c2abf3745bc698d')
 
+source+=('cast-pointer-type.patch::https://github.com/xmlsec/python-xmlsec/commit/67cd4ac73e4fceac4b4eb6a320067cad33f79213.patch')
+sha512sums+=('12f8ab47afa14a0c73e7db502710cb2620ae9817be50aa3aa3dfd9aad48fa742a73f47e73281ee3a8428da812c84fa27814f196540bddcb866fa009325d5647f')
+
 build() {
   cd python-xmlsec
+  patch -Np1 -i ../cast-pointer-type.patch
   python -m build --wheel --no-isolation
 }
 
