--- a/PKGBUILD
+++ b/PKGBUILD
@@ -77,11 +77,13 @@ build() {
   cmake "${_cpu_args[@]}"
   cmake --build build
 
+  : <<COMMENT_SEPARATOR
   echo "Building torchvision (GPU version with CUDA)"
   cd "${srcdir}/${_pkgname}-cuda-${pkgver}"
   local _gpu_args=("${_common_cmake[@]}" -DWITH_CUDA=ON)
   cmake "${_gpu_args[@]}"
   cmake --build build
+COMMENT_SEPARATOR
 
   # build python-torchvision
   echo "Building torchvision python bindings (CPU version)"
@@ -91,6 +93,7 @@ build() {
   TORCHVISION_USE_NVJPEG=0 \
   python -m build --wheel --no-isolation
 
+  : <<COMMENT_SEPARATOR
   # build python-torchvision-cuda
   echo "Building torchvision python bindings (GPU version with CUDA)"
   cd "${srcdir}/python-${_pkgname}-cuda-${pkgver}"
@@ -99,6 +102,7 @@ build() {
   FORCE_CUDA=1 \
   TORCH_CUDA_ARCH_LIST="${_CUDA_ARCH_LIST}" \
   python -m build --wheel --no-isolation
+COMMENT_SEPARATOR
 }
 
 package_python-torchvision() {
@@ -158,4 +162,8 @@ package_torchvision-cuda() {
   install -vDm 644 LICENSE -t "$pkgdir"/usr/share/licenses/$pkgname/
 }
 
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '^(torchvision-cuda|python-torchvision-cuda)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|cudnn|python-pytorch-opt-cuda)$'))
+makedepends+=(python-pytorch-opt python-fsspec)
+
 # vim:set ts=2 sw=2 et:
