diff --git a/PKGBUILD b/PKGBUILD
index 0afa6dc..ff0b39d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -96,11 +96,13 @@ build() {
   cmake "${_cpu_args[@]}"
   cmake --build build
 
+  : <<COMMENT_SEPARATOR
   echo "Building torchvision (GPU version with CUDA)"
   cd "${srcdir}/${_pkgname}-cuda-${pkgver}"
   local _gpu_args=("${_common_cmake[@]}" -DWITH_CUDA=ON)
   cmake "${_gpu_args[@]}"
   cmake --build build
+COMMENT_SEPARATOR
 
   # build python-torchvision
   echo "Building torchvision python bindings (CPU version)"
@@ -111,6 +113,7 @@ build() {
   python setup.py build
 
 
+  : <<COMMENT_SEPARATOR
   # build python-torchvision-cuda
   cd "${srcdir}/python-${_pkgname}-cuda-${pkgver}"
   TORCHVISION_INCLUDE="${srcdir}" \
@@ -118,6 +121,7 @@ build() {
   FORCE_CUDA=1 \
   TORCH_CUDA_ARCH_LIST="${_CUDA_ARCH_LIST}" \
   python setup.py build
+COMMENT_SEPARATOR
 }
 
 # TODO(gromit): re-enable the tests
@@ -176,4 +180,8 @@ package_torchvision-cuda() {
   install -m644 -Dt "$pkgdir/usr/share/licenses/$pkgname" LICENSE
 }
 
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '^(torchvision-cuda|python-torchvision-cuda)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|cudnn|python-pytorch-opt-cuda)$'))
+makedepends+=(python-pytorch)
+
 # vim:set ts=2 sw=2 et:
