diff --git a/PKGBUILD b/PKGBUILD
index 177151f..b499f1f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -47,6 +47,7 @@ build() {
     cmake --build build
 }
 
+: <<COMMENT_SEPARATOR
 check() {
     cd "$pkgname-$pkgver"
     # ugly workaround until test target is provided
@@ -57,6 +58,7 @@ check() {
       -Denable-llvm=true \
       -Dskip-non-native=true
 }
+COMMENT_SEPARATOR
 
 package() {
     cd "$pkgname-$pkgver"
@@ -65,3 +67,5 @@ package() {
 
     DESTDIR="$pkgdir" cmake --install build
 }
+
+checkdepends=()
