diff --git a/PKGBUILD b/PKGBUILD
index 7ba736a..3df7a38 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,6 +20,7 @@ source=("https://github.com/intel/media-driver/archive/intel-media-${pkgver}/${p
 sha256sums=('a23185fca05c8fa8bdec09ea9ee8a20361163c87035ec978de4e2bb048a55534')
 
 build() {
+    patch -Np1 -d media-driver-intel-media-${pkgver} -i "${srcdir}/add-loongarch64-build-support.patch"
     cmake -B build -S "media-driver-intel-media-${pkgver}" \
         -G 'Unix Makefiles' \
         -DCMAKE_INSTALL_PREFIX:PATH='/usr' \
@@ -35,3 +36,7 @@ package() {
     DESTDIR="$pkgdir" cmake --install build
     install -D -m644 "media-driver-intel-media-${pkgver}/LICENSE.md" -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
+
+source+=("add-loongarch64-build-support.patch")
+sha256sums+=('576d2122e61a47a3a020ba4c2e04ea091c83217cb53a1677c507521e0ab194bb')
+makedepends+=(simde)
