--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,6 +20,7 @@ source=("https://github.com/intel/media-driver/archive/intel-media-${pkgver}/${p
 sha256sums=('ef9a7a0881b250d23df7d4676341a3729b28887bcf404a74d48db5f44ffa0220')
 
 build() {
+    patch -Np1 -d media-driver-intel-media-${pkgver} -i "${srcdir}/add-loongarch64-build-support.patch"
     cmake -B build -S "media-driver-intel-media-${pkgver}" \
         -G 'Unix Makefiles' \
         -DCMAKE_INSTALL_PREFIX:PATH='/usr' \
@@ -35,3 +36,7 @@ package() {
     DESTDIR="$pkgdir" cmake --install build
     install -D -m644 "media-driver-intel-media-${pkgver}/LICENSE.md" -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
+
+source+=("add-loongarch64-build-support.patch::https://patch-diff.githubusercontent.com/raw/intel/media-driver/pull/1936.diff")
+sha256sums+=('f3ac703356ba6689080c981a8dd6b38fc7e155e8c7a976b1f07363d97e77e502')
+makedepends+=(simde)
