diff --git a/PKGBUILD b/PKGBUILD
index 4472d68..05ff52b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,7 +10,7 @@ pkgdesc="A particle physics package for storing collision events from Monte Carl
 arch=('x86_64')
 url="https://hepmc.web.cern.ch/"
 license=('GPL3')
-makedepends=('cmake' 'doxygen' 'gcc-fortran' 'graphviz' 'hepmc2' 'pythia8' 'python' 'root' 'protobuf' 'cern-vdt')
+makedepends=('cmake' 'doxygen' 'gcc-fortran' 'graphviz' 'hepmc2' 'python' 'root' 'protobuf' 'cern-vdt')
 source=("${pkgbase}-${pkgver}.tar.gz::https://hepmc.web.cern.ch/${pkgbase}/releases/${_pkgname}-${pkgver}.tar.gz")
 sha256sums=('248f3b5b36dd773844cbe73d51f60891458334b986b259754c59dbf4bbf1d525')
 
@@ -36,8 +36,7 @@ build() {
     -D HEPMC3_INSTALL_INTERFACES=ON
     -D HEPMC3_PYTHON_VERSIONS='3.X'
     -D HEPMC3_ENABLE_PROTOBUFIO=ON
-    -D USE_INTERFACE_FROM_PYTHIA8=ON
-    -D PYTHIA8_XMLDOC_DIR=/usr/share/pythia8/xmldoc
+    -D USE_INTERFACE_FROM_PYTHIA8=OFF
     -S $_pkgname-$pkgver
     -W no-dev
   )
