--- a/PKGBUILD
+++ b/PKGBUILD
@@ -29,9 +29,6 @@ build() {
       -DKERNELLIB_HOST_CPU_VARIANTS=distro
       # Should be provided by opencl-headers, avoid conflict
       -DINSTALL_OPENCL_HEADERS=OFF
-      -DENABLE_CUDA=ON
-      -DENABLE_CUDNN=ON
-      -DCUDA_TOOLKIT_ROOT_DIR=/opt/cuda
   )
   cmake "${cmake_options[@]}"
   ninja -C build
@@ -49,3 +46,5 @@ package() {
   DESTDIR="$pkgdir"/ ninja -C build install
   install -v -Dm 644 -t "$pkgdir"/usr/share/licenses/"$pkgname" LICENSE
 }
+makedepends=(${makedepends[@]/cuda})
+makedepends=(${makedepends[@]/cudnn})
