--- a/PKGBUILD
+++ b/PKGBUILD
@@ -60,8 +60,6 @@ makedepends=(
   cfitsio
   chromium
   cmake
-  cuda
-  cudnn
   gcc-fortran
   git
   go
@@ -189,8 +187,11 @@ build() {
   cmake --build build
 
   ## ROOT with CUDA
+  # Use a "multi-line comment" to keep patch from rotting
+  : <<COMMENT_SEPARATOR
   cmake -B build-cuda "${common_cmake_options[@]}" "${cuda_cmake_options[@]}"
   cmake --build build-cuda
+COMMENT_SEPARATOR
 }
 
 _package() {
@@ -306,4 +307,5 @@ package_root-cuda() {
   _package build-cuda
 }
 
+pkgname=(root)
 # vim: ts=2 sw=2 et:
