--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,7 +6,7 @@
 # Contributor: Sebastian Voecking < voeck at web dot de >
 
 pkgbase=root
-pkgname=(root root-cuda)
+pkgname=(root)
 pkgver=6.34.02
 pkgrel=3
 pkgdesc='C++ data analysis framework and interpreter from CERN'
@@ -60,8 +60,6 @@ makedepends=(
   cfitsio
   chromium
   cmake
-  cuda
-  cudnn
   gcc-fortran
   git
   go
@@ -186,8 +184,11 @@ build() {
   cmake --build build
 
   ## ROOT with CUDA
+  # Use a "multi-line comment" to keep patch from rotting
+  : <<COMMENT_SEPARATOR
   cmake -B build-cuda "${common_cmake_options[@]}" "${cuda_cmake_options[@]}"
   cmake --build build-cuda
+COMMENT_SEPARATOR
 }
 
 _package() {
@@ -303,4 +304,7 @@ package_root-cuda() {
   _package build-cuda
 }
 
+source+=("add-loong64-support.patch::https://github.com/root-project/root/commit/0c716a13bcdb05a2b60c7c60e053f06e70aae49f.patch")
+b2sums+=('224448d2036a304ad0d129b931e0265d3dcff79589c102a46ef24fb50221c7dd04f41ee770810a466f7c14a59399a976857e373effda6c24ca7d7d5eb1bd799a')
+
 # vim: ts=2 sw=2 et:
