diff --git a/PKGBUILD b/PKGBUILD
index 8b01790..b67e821 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,6 +37,8 @@ prepare() {
   # file. A meson.build file is not necessary, so ignore the error.
   meson subprojects download || :
 
+  patch -d subprojects/boxfort -p1 -i "${srcdir}/boxfort-support-loongarch64.patch"
+
   # Fix FTBS
   sed -i '/#ifdef __cplusplus/a # include <cstdint>' include/criterion/alloc.h
 }
@@ -58,3 +60,6 @@ package() {
   meson install -C build --destdir "$pkgdir"
   install -vDm644 -t "$pkgdir/usr/share/licenses/$pkgname" LICENSE
 }
+
+source+=("boxfort-support-loongarch64.patch::https://github.com/Snaipe/BoxFort/pull/58.diff")
+sha256sums+=('5c8431ba3a033fd7940f26f3781ca9eeac0ca3d088d551a63533427debce7476')
