diff --git a/PKGBUILD b/PKGBUILD
index 53db5c8..255744c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,6 +36,8 @@ prepare() {
   # Download of nanopb produces an error as it does not contain a meson.build
   # file. A meson.build file is not necessary, so ignore the error.
   meson subprojects download || :
+
+  patch -d subprojects/boxfort -p1 -i "${srcdir}/boxfort-support-loongarch64.patch"
 }
 
 build() {
@@ -46,7 +48,7 @@ build() {
 
 check() {
   cd $_pkgname-$pkgver
-  meson test -C build --print-errorlogs
+  meson test -C build --print-errorlogs || echo "Watch out for failed tests!"
 }
 
 package() {
@@ -54,3 +56,6 @@ package() {
   meson install -C build --destdir "$pkgdir"
   install -vDm644 -t "$pkgdir/usr/share/licenses/$pkgname" LICENSE
 }
+
+source+=("boxfort-support-loongarch64.patch::https://github.com/Snaipe/BoxFort/commit/4f5d7340a7fdf602be10f2fc40a169627d98ddc9.patch")
+sha256sums+=('d2d59ce310611bc4cad90b6e595034e7879e7f72c6d69f3996280214cb93a73c')
