--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,8 +13,11 @@ source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
 sha256sums=('81b309512fa92dd97d99c0fca9511ad50d3a0252a7d73a6a63a4b6fa2b442111')
 options=('!lto')
 
+source+=("update_lock.patch::https://github.com/axodotdev/cargo-dist/pull/1466.patch")
+sha256sums+=('4a2b5466e868b6a8d549de151924e17543a786de986af4f15afd4e1cb227affb')
 prepare() {
   cd "$pkgname-$pkgver"
+  patch -Np1 -i ../update_lock.patch
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
