--- a/PKGBUILD
+++ b/PKGBUILD
@@ -9,7 +9,7 @@ arch=('x86_64')
 url="https://docs.xfce.org/xfce/thunar/thunar-vcs-plugin"
 license=('GPL-2.0-or-later')
 depends=('thunar' 'git' 'subversion')
-makedepends=('xfce4-dev-tools')
+makedepends=('xfce4-dev-tools' 'intltool')
 source=("git+https://gitlab.xfce.org/thunar-plugins/thunar-vcs-plugin.git#tag=$pkgname-$pkgver")
 sha256sums=('86511d7433eade7f4f9867b37bd57d7d2133f05d678b48b78f71a40dc45014d7')
 
