diff --git a/PKGBUILD b/PKGBUILD
index 2734c93..37e705e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -100,8 +100,20 @@ prepare() {
     patch -Np1 < "../$src"
   done
 
+  # Ensure amdkfd is avaliable
+  sed -i '/^[[:space:]]*depends on DRM_AMDGPU && (/ {
+    /LOONGARCH/! {
+        s/)[[:space:]]*$/ || LOONGARCH)/
+    }
+  }' drivers/gpu/drm/amd/amdkfd/Kconfig
+
   echo "Setting config..."
-  cp ../config.$CARCH .config
+#  cp ../config.$CARCH .config
+  cp ../config.x86_64 .config
+  if [ $CARCH == loong64 ]; then
+    make savedefconfig
+    cat defconfig ../loong-addition.config > .config
+  fi
   make olddefconfig
   # make nconfig
   diff -u ../config.$CARCH .config || :
@@ -261,6 +273,7 @@ _package-headers() {
   local karch
   case $CARCH in
     x86_64) karch=x86 ;;
+    loong64) karch=loongarch ;;
     *) echo "Unknown CARCH $CARCH"; exit 1 ;;
   esac
 
@@ -409,4 +422,35 @@ for _p in "${pkgname[@]}"; do
   }"
 done
 
+source+=("config.x86_64"
+         "loong-addition.config"
+         "LoongArch-Add-CPU-HWMon-platform-driver.patch::https://github.com/chenhuacai/linux/commit/9367b1a79e9b7b2d09e330d1f4fc0cd195f82091.diff"
+         "0001-LOONGARCH-drm-radeon-Call-mmiowb-at-the-end-of-radeon_ring_commit.patch"
+         "0010-REVERT-Revert-Mark-xe-driver-as-BROKEN-if-kernel-pag.patch"
+         "0011-BACKPORT-FROMLIST-drm-xe-bo-fix-alignment-with-non-4.patch"
+         "0012-BACKPORT-FROMLIST-drm-xe-guc-use-SZ_4K-for-alignment.patch"
+         "0013-BACKPORT-FROMLIST-drm-xe-regs-fix-RING_CTL_SIZE-size.patch"
+         "0014-FROMLIST-drm-xe-use-4K-alignment-for-cursor-jumps.patch"
+         "0015-FROMLIST-drm-xe-query-use-PAGE_SIZE-as-the-minimum-p.patch")
+b2sums+=('SKIP'
+         '652069e50060a3d84530f004f7064d98d6ac9bb02e690a3510d092d0df17598c9b3d564ae07fdb0dbbecc68d2d12d58c4c78c02cc08fbcbb8c01ba6e6256d922'
+         'f0f3a73bd953fe1c3ec3b06519ca2bdec70231de0470f217b166a9bba44168d27e826898913e38f8045f05dc5b7d84eed6074befab9563814663f1fd9b882873'
+         '4a0b268f1a28d4bfa37d9b644477c75a5215083414fe5cc5d637e2edc4895aace9e42b3b33ac0b3e2a15a18340bec651511fa3c347c1c3e347ed3418c033582c'
+         '84cc56429a3c4efc1926486d2d2eaab997bc905bed69b72ce2e461bf63203ee35f652a26fd46eb412768519a35834f0ff6f74057146b59df5af233e6d48e98fe'
+         'c50d7972d3c35d56039b70fe33c4587222c7df4595e0025efc733e42ab2e8ba8ce4ce938828ef6183d84ac413665c8ab7f1dc7b09227c3bbe44591322849648f'
+         'beb12df9b2261bfe03f4835573a5b54f545a53350ec423b42c66ad345defc645a46fa496fc0c0fc2de38da3cbbf0125ff1bab33d4452edf7baa1df23a43a2567'
+         '79a2d78fbc6055ef17f2a816bf5edeb9ee61c7e16a6cd4d5c89711dcb9328b08a4b05194bc6fda9da60818dded0a5e3916c0cb8f9e8060fcc602f12ad9a39e91'
+         'cda91ae4f92cae590be2366a72d819e7c679d322e1a118de15e7917b75bc86f4327bc3e4d1813a422ee159e218a3dd00be5291f6bd5d939702c5ef1e3cbcf053'
+         '925d08a6656b4f34566d5a69b55c0110e32ec0c4281a0395bccca70f4f593851f12a2d5a83f3971f2188e5ac4a24ede206ba248b0af21a6ab51afd43bb92fb57')
+sha512sums+=('SKIP'
+             '1b68c3b4d12eecffeae1322a431d9b5aa3b19387c2276472390c38863e472185afff20aacaedd77bf1ca0f6a8108188db44603968e0b4fd509d51010e01026f2'
+             '2105f934a448bf2735c3a03dc1f681b26ae6cd65bd8f5fb16dd91613fbd93f74be97a632b365dfc5e28d4879117f41990dca8ec55144abf5df31b4b3e63fa475'
+             '98a749b23aad98b16f7bd29cd745bb21eb0f42d492540853a69b12d84054f0413ee08d056668fe8738e21b8bb33b0dc9878fde7ea21af1838fc3b27fc0a17aaf'
+             '5b7c9c746bf7ad6e160f2068809cb1aad100c49c9dcfb656d132247868834e78dae0631a0cd79c0c8b97ac0d4de32fe93ecad4b622cd77c979897e5880e0cae7'
+             'b8b99206df053998719f36e0b8a273ad2e8cb259ec5b65690985a795dccfd21e73d3475f6d68c38b7bd34ceeac5a93a5bdbd3e42edf533762c3c61973f79869b'
+             '7d844e78574bb21aebcfe5f3cad7eb5f75edf2261bfc57108a740d8c809ef38f10c0d3aa8aa00e93eef5dc23de1237b82f906e98b59f70a5441a2ec8173b76c7'
+             '987e3f39f069e792b44433ba8915d24da8a1ac117d1df3378f0731eafd36c2c40c2c25db4133a7de3d12a8bc39c879e98bb6a7b7b3057a83f478d01993c7a406'
+             '0092065d416c20049a5e8a8cd080e020507a6c65de891bc9684f280eda056dce3e44098f0c254f70ecc9f23e3c0dbac52d3679239dc4f423e192e3fe0c0091be'
+             '8fd041f9ee3b849e699431fb2736826a44ee9137c203acdde09b27192172b525db1c8843e26e79e8426540ad2fbf79dbaa51f6cb9aecf8a2e512556f9bf83e4c')
+
 # vim:set ts=8 sts=2 sw=2 et:
