--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,6 +26,10 @@ sha256sums=('e37902dd39ca9cbee37095ccce841e3e9f744f8b86f5935fa68f833dcd37d0ed')
 
 prepare() {
     cd "${srcdir}"/local-agent-rs/python-proton-vpn-local-agent
+    cargo update -p bindgen
+    export CFLAGS="${CFLAGS/-Wp,-D_FORTIFY_SOURCE=3/} -Wno-unterminated-string-initialization"
+    export CXXFLAGS="${CXXFLAGS/-Wp,-D_FORTIFY_SOURCE=3/} -Wno-unterminated-string-initialization"
+    export CMAKE_POLICY_VERSION_MINIMUM=3.5
     cargo fetch --locked --target "$(rustc --print host-tuple)"
 }
 
@@ -43,3 +47,4 @@ package() {
     mkdir -p "${pkgdir}"/usr/lib/python"${_pyver}"/site-packages/proton/vpn
     install -Dm755 "${srcdir}"/local-agent-rs/python-proton-vpn-local-agent/target/release/libpython_proton_vpn_local_agent.so "${pkgdir}"/usr/lib/python"${_pyver}"/site-packages/proton/vpn/local_agent.abi3.so
 }
+makedepends+=(cmake clang)
