--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,6 +22,9 @@ b2sums=('6e6f5d39c20ffbc072382c43000b73058e69a7f281337ad7b75b9682db1fa5273712097
 
 prepare() {
   cd $pkgname-$pkgver
+  # Temporarily switch to wszqkzqk's fork of hnsw_rs
+  printf '\n[patch.crates-io]\nhnsw_rs = { git = "https://github.com/wszqkzqk/hnswlib-rs.git", rev = "41d206273ee291b6e24ceca21db4aa427b573178" }\n' >> Cargo.toml
+  cargo update -p hnsw_rs
   cargo fetch --locked --target "$(rustc --print host-tuple)"
 }
 
