diff --git a/PKGBUILD b/PKGBUILD
index 0bb253b..1d834ef 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -32,3 +32,10 @@ check() {
 package() {
   DESTDIR="$pkgdir" cmake --install build
 }
+
+prepare() {
+  patch -d "${srcdir}/${pkgname}" -p1 -i "${srcdir}/add-loong64-support.patch"
+}
+
+source+=("add-loong64-support.patch::https://github.com/google/libultrahdr/commit/a70b554a41189b3ce60486a832e361838d66a8f3.patch")
+sha256sums+=('a40164842f0479b1aa02177d1be13a819cc54727d284f5c19f243ac9c67435a2')
