diff --git a/PKGBUILD b/PKGBUILD
index 0bb253b..1889da4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -32,3 +32,10 @@ check() {
 package() {
   DESTDIR="$pkgdir" cmake --install build
 }
+
+prepare() {
+  patch -d "${srcdir}/${pkgname}" -p1 -i "${srcdir}/add-loong64-support.patch"
+}
+
+source+=("add-loong64-support.patch::https://github.com/google/libultrahdr/commit/0fbd5117560cfd415e969d4f6f7a8011e9e7818d.patch")
+sha256sums+=('7513241bf31cd03d71c8b373105936e34499dc1d1d1d7725ce07b16dac263b82')
