diff --git a/PKGBUILD b/PKGBUILD
index 0bb253b..95e03e7 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -32,3 +32,10 @@ check() {
 package() {
   DESTDIR="$pkgdir" cmake --install build
 }
+
+prepare() {
+  patch -d "${srcdir}/${pkgname}" -p1 -i "${srcdir}/add-loong64-support.patch"
+}
+
+source+=("add-loong64-support.patch::https://github.com/wszqkzqk/libultrahdr/commit/96e7bc1eeda936035c082269611cce5112b25040.patch")
+sha256sums+=('857424bd386e80fa24ad0bdd8da9cf51e9e3cb1795c00d45fb0228066b7d39ae')
