--- a/PKGBUILD
+++ b/PKGBUILD
@@ -93,6 +93,7 @@ prepare() {
 #  git cherry-pick -n 142801f769ed4946a161f62738bac5a5fe56a60f # https://github.com/opencv/opencv/issues/29291
 #  git cherry-pick -n 883f8d83ae1463a5f486ace784762e8c42e7ea7d # https://github.com/opencv/opencv/issues/29471
   patch -p1 < ../fix-dnn-umat-dimensions.patch # Fix face recognition in digikam
+  git cherry-pick -n ee9c019556d7c3133aca55fa02ef2d6e20893f49 # https://github.com/opencv/opencv/issues/29422
   popd
 
   pushd opencv_contrib
@@ -142,7 +143,8 @@ build() {
   cmake -B build -S $pkgname "${cmake_options[@]}" \
     -DBUILD_WITH_DEBUG_INFO=ON
   cmake --build build
-
+  # Use a "multi-line comment" to keep patch from rotting
+  : <<'COMMENT_SEPARATOR'
   # In general, we want to list all real archs (sm_XX) and the latest virtual arch (compute_XX) for future PTX compatibility.
   # Valid values can be discovered from nvcc --help
   local cuda_archs="75;80;86;87;88;89;90;100;103;110;120;121;121-virtual"
@@ -157,6 +159,7 @@ build() {
     -DENABLE_CUDA_FIRST_CLASS_LANGUAGE=ON \
     -DCMAKE_CUDA_ARCHITECTURES="$cuda_archs"
   cmake --build build-cuda --verbose
+COMMENT_SEPARATOR
 }
 
 package_opencv() {
@@ -240,3 +243,11 @@ package_python-opencv-cuda() {
 
   DESTDIR="$pkgdir" cmake --install build-cuda/modules/python3
 }
+
+for i in "${!pkgname[@]}"; do
+  if [[ ${pkgname[$i]} == *"-cuda" ]]; then
+    unset 'pkgname[$i]'
+  fi
+done
+makedepends=(${makedepends[@]/cuda})
+makedepends=(${makedepends[@]/cudnn})
