--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,8 +22,12 @@ b2sums=('SKIP'
         'SKIP'
         'SKIP')
 
+source+=("rapidjson-gcc14-fix.patch::https://github.com/cern-fts/davix/commit/599fead02dd22f34f9dad2da7719cbff631be98c.patch")
+b2sums+=('SKIP')
+
 prepare() {
   cd "${srcdir}/${pkgname}"
+  patch -Np1 -i ../rapidjson-gcc14-fix.patch
   git submodule init
 
   git config submodule."deps/curl".url "${srcdir}/${pkgname}"-curl
