--- a/PKGBUILD
+++ b/PKGBUILD
@@ -58,7 +58,6 @@ build() {
     -DGGML_BACKEND_DIR=/usr/lib/ggml
     -DGGML_BLAS=ON
     -DGGML_BLAS_VENDOR=OpenBLAS
-    -DGGML_CPU_ALL_VARIANTS=ON
     -DGGML_VULKAN=ON
   )
 
@@ -69,6 +68,12 @@ build() {
       -DGGML_HIP=ON
       -DGGML_OPENVINO=ON
     )
+  elif [[ $CARCH == loong64 ]]; then
+    _cmake_options+=(
+      -DGGML_CUDA=OFF
+      -DGGML_HIP=ON
+      -DGGML_OPENVINO=OFF
+    )
   else
     _cmake_options+=(
       -DGGML_CUDA=OFF
@@ -124,7 +129,7 @@ package_ggml() {
 
   DESTDIR="${pkgdir}" cmake --install build-all
   rm -rf "${pkgdir}"/usr/lib/ggml
-  install -Dm 755 build-all/bin/libggml-cpu-*.so -t "${pkgdir}"/usr/lib/ggml/
+  install -Dm 755 build-all/bin/libggml-cpu.so -t "${pkgdir}"/usr/lib/ggml/
   install -Dm 644 ggml/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
 
@@ -207,3 +212,10 @@ package_ggml-vulkan() {
   install -Dm 755 build-all/bin/libggml-vulkan.so -t "${pkgdir}"/usr/lib/ggml/
   install -Dm 644 ggml/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+makedepends_loong64=(
+  rocm-hip-sdk
+  rocm-toolchain
+)
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '^(ggml-sycl|ggml-cuda|ggml-openvino)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|nccl|intel-oneapi-toolkit|openvino)$'))
