--- a/PKGBUILD
+++ b/PKGBUILD
@@ -60,10 +60,9 @@ build() {
     -DGGML_BACKEND_DIR=/usr/lib/ggml
     -DGGML_BLAS=ON
     -DGGML_BLAS_VENDOR=OpenBLAS
-    -DGGML_CPU_ALL_VARIANTS=ON
-    -DGGML_CUDA=ON
+    -DGGML_CUDA=OFF
     -DGGML_HIP=ON
-    -DGGML_OPENVINO=ON
+    -DGGML_OPENVINO=OFF
     -DGGML_VULKAN=ON
     "-DAMDGPU_TARGETS=$(rocm-supported-gfx)"
     )
@@ -113,7 +112,7 @@ package_ggml() {
 
   DESTDIR="${pkgdir}" cmake --install build-all
   rm -rf "${pkgdir}"/usr/lib/ggml
-  install -Dm 755 build-all/bin/libggml-cpu-*.so -t "${pkgdir}"/usr/lib/ggml/
+  install -Dm 755 build-all/bin/libggml-cpu.so -t "${pkgdir}"/usr/lib/ggml/
   install -Dm 644 ggml/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
 
@@ -194,3 +193,10 @@ package_ggml-vulkan() {
   install -Dm 755 build-all/bin/libggml-vulkan.so -t "${pkgdir}"/usr/lib/ggml/
   install -Dm 644 ggml/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+makedepends_loong64=(
+  rocm-hip-sdk
+  rocm-toolchain
+)
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '^(ggml-sycl|ggml-cuda|ggml-openvino)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|nccl|intel-oneapi-toolkit|openvino)$'))
