--- a/PKGBUILD
+++ b/PKGBUILD
@@ -54,14 +54,13 @@ build() {
       -DGGML_BACKEND_DIR=/usr/lib/ggml \
       -DGGML_BLAS=ON \
       -DGGML_BLAS_VENDOR=OpenBLAS \
-      -DGGML_CPU_ALL_VARIANTS=ON \
-      -DGGML_CUDA=ON \
       -DGGML_HIP=ON \
-      -DGGML_OPENVINO=ON \
       -DGGML_VULKAN=ON
   cmake --build build-all
   chrpath -d build-all/bin/libggml-*.so
 
+  # Use a "multi-line comment" to keep patch from rotting
+  : <<COMMENT_SEPARATOR
   . /opt/intel/oneapi/setvars.sh
 
   cmake -S ggml -B build-sycl -G Ninja \
@@ -79,6 +78,7 @@ build() {
       -DGGML_SYCL_F16=ON
   cmake --build build-sycl
   chrpath -d build-sycl/bin/libggml-*.so
+COMMENT_SEPARATOR
 }
 
 package_ggml() {
@@ -86,8 +86,8 @@ package_ggml() {
   optdepends=(
     'ggml-blas: BLAS backend'
     'ggml-cpu: CPU backend'
-    'ggml-cuda: CUDA backend'
-    'ggml-openvino: Openvino backend'
+#    'ggml-cuda: CUDA backend'
+#    'ggml-openvino: Openvino backend'
     'ggml-hip: HIP backend'
     'ggml-vulkan: Vulkan backend'
   )
@@ -108,7 +108,7 @@ package_ggml-blas() {
 }
 
 package_ggml-cpu() {
-  install -Dm 755 build-all/bin/libggml-cpu-*.so -t "${pkgdir}"/usr/lib/ggml/
+  install -Dm 755 build-all/bin/libggml-cpu.so -t "${pkgdir}"/usr/lib/ggml/
   install -Dm 644 ggml/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
 
@@ -176,3 +176,6 @@ package_ggml-vulkan() {
   install -Dm 755 build-all/bin/libggml-vulkan.so -t "${pkgdir}"/usr/lib/ggml/
   install -Dm 644 ggml/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+pkgname=($(printf "%s\n" "${pkgname[@]}" | grep -Ev '^(ggml-sycl|ggml-cuda|ggml-openvino)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|nccl|intel-oneapi-toolkit|openvino)$'))
