--- a/PKGBUILD
+++ b/PKGBUILD
@@ -58,6 +58,8 @@ build() {
   _set_flags
   cd node
 
+  patch -Np1 -i "${srcdir}/v8-loong64-codetype-fix.patch"
+
   ./configure \
     --prefix=/usr \
     --with-intl=system-icu \
@@ -98,7 +100,8 @@ check() {
   rm test/parallel/test-icu-env.js
 
   rm test/parallel/test-runner-run-watch.mjs
-
+  rm -f test/parallel/test-dgram-send-cb-quelches-error.js
+  rm -f test/parallel/test-net-autoselectfamily.js
   make test-only
 }
 
@@ -109,4 +112,7 @@ package() {
   install -Dm644 LICENSE -t "$pkgdir"/usr/share/licenses/$pkgname
 }
 
+source+=(v8-loong64-codetype-fix.patch)
+sha512sums+=('26c84d44083eb7e6cf82add949d29a0b00eba1d196284e84ef05cde5e7434d6270cb9a954c4e0fa9c6785156cee4ea02c266f6f0225168aaaafd791a0a6358c0')
+
 # vim:set ts=2 sw=2 et:
