diff --git a/PKGBUILD b/PKGBUILD
index fb227e2..cbfe567 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -85,6 +85,11 @@ check() {
   rm test/parallel/test-http2-client-set-priority.js
   rm test/parallel/test-http2-priority-event.js
   rm test/parallel/test-http-outgoing-end-cork.js
+  rm -f test/parallel/test-dgram-send-cb-quelches-error.js \
+    test/parallel/test-net-autoselectfamily.js \
+    test/sequential/test-cpu-prof-dir-worker.js \
+    test/sequential/test-cpu-prof-exit.js \
+    test/sequential/test-cpu-prof-kill.js
   make test-only
 }
 
