--- a/src/cpu_features.c
+++ b/src/cpu_features.c
@@ -16,7 +16,7 @@
 int cpu_supports_avx512f ()  { return 0; }
 int cpu_supports_avx512vl () { return 0; }
 
-#elif defined(__riscv)
+#elif defined(__riscv) || defined(__loongarch__)
 
 // TODO: Support RVV
 int cpu_supports_sse2 ()     { return 0; }
--- a/src/bridges/bridge_argon2id_reference.c
+++ b/src/bridges/bridge_argon2id_reference.c
@@ -18,7 +18,7 @@
 #include "core.c"
 #include "blake2/blake2b.c"
 
-#if defined(__riscv)
+#if defined(__riscv) || defined(__loongarch__)
 #include "ref.c"
 #else
 #include "opt.c"
