--- a/PKGBUILD
+++ b/PKGBUILD
@@ -24,22 +24,22 @@ makedepends=(
   python-wheel
 )
 checkdepends=(python-pytest)
-source=($pkgname-$pkgver.tar.gz::$url/archive/release-$_version.tar.gz)
-sha512sums=('983bdb17b76d10eb73cf69f0ba7590806bf6786ae0c30526593ce1ab97f3b22c967f9acae83b880dd56be41a29eae9cca5726926667eb74013d0d1078433bf05')
+source=("http://deb.debian.org/debian/pool/main/p/$pkgname/${pkgname}_$pkgver.orig.tar.gz")
+sha512sums=('d863f0937b340958ef7c8bd07d62054aead993ba31b0c2753631f2e75e48f7cea412c6501d8a94d98bf6a97a13eb0716c24a89e336787d5570e9af0386c58320')
 
 prepare() {
-  cd $pkgname-release-$_version
+  cd $_name-$pkgver
   # we rely on publicsuffix-list and do not need/want to download a new list during build
   sed -e "/'requests/d" -i setup.py
 }
 
 build() {
-  cd $pkgname-release-$_version
+  cd $_name-$pkgver
   python -m build --wheel --no-isolation
 }
 
 check() {
-  cd $pkgname-release-$_version
+  cd $_name-$pkgver
   export PYTHONPATH="build/lib:$PYTHONPATH"
   pytest -v
 }
@@ -47,9 +47,9 @@ check() {
 package() {
   local site_packages=$(python -c "import site; print(site.getsitepackages()[0])")
 
-  cd $pkgname-release-$_version
+  cd $_name-$pkgver
   python -m installer --destdir="$pkgdir" dist/*.whl
-  install -vDm 644 {AUTHORS,CHANGELOG,README}.rst -t "$pkgdir/usr/share/doc/$pkgname/"
+  install -vDm 644 {CHANGELOG,README}.rst -t "$pkgdir/usr/share/doc/$pkgname/"
   install -vDm 644 "$_name.LICENSE" -t "$pkgdir/usr/share/licenses/$pkgname/"
   # use the system provided publicsuffix list (publicsuffix-list package)
   ln -svf /usr/share/publicsuffix/public_suffix_list.dat "$pkgdir/$site_packages/$_name/public_suffix_list.dat"
