--- a/build/config/clang/BUILD.gn
+++ b/build/config/clang/BUILD.gn
@@ -166,6 +169,11 @@ template("clang_lib") {
         assert(false)  # Unhandled target platform
       }
 
+      # Bit of a hack to make this find builtins from compiler-rt >= 16
+      if (is_linux || is_chromeos) {
+        _dir = "linux"
+      }
+
       _clang_lib_dir = "$clang_base_path/lib/clang/$clang_version/lib"
       _lib_file = "${_prefix}clang_rt.${_libname}${_suffix}.${_ext}"
       libs = [ "$_clang_lib_dir/$_dir/$_lib_file" ]
