--- a/cef/tools/make_distrib.py
+++ b/cef/tools/make_distrib.py
@@ -695,6 +695,12 @@
       default=False,
       help='create an ARM binary distribution (Linux only)')
   parser.add_option(
+      '--loong64-build',
+      action='store_true',
+      dest='loong64build',
+      default=False,
+      help='create a Loong64 binary distribution (Linux only)')
+  parser.add_option(
       '--arm64-build',
       action='store_true',
       dest='arm64build',
@@ -755,10 +761,12 @@
     return '--output-dir is required.'
   if options.minimal and options.client:
     return 'Cannot specify both --minimal and --client.'
-  if options.x64build + options.armbuild + options.arm64build > 1:
+  if options.x64build + options.armbuild + options.loong64build + options.arm64build > 1:
     return 'Invalid combination of build options.'
   if options.armbuild and platform != 'linux':
     return '--arm-build is only supported on Linux.'
+  if options.loong64build and platform != 'linux':
+    return '--loong64-build is only supported on Linux.'
   if options.sandbox and not platform in ('mac', 'windows'):
     return '--sandbox is only supported on macOS and Windows.'
   if not options.ninjabuild:
@@ -779,6 +787,8 @@
     return ('arm', 'arm', '_GN_arm')
   if options.arm64build:
     return ('arm64', 'arm64', '_GN_arm64')
+  if options.loong64build:
+    return ('loong64', 'loong64', '_GN_loong64')
   return ('32', 'x86', '_GN_x86')
 
 
