--- a/cef/include/base/cef_build.h	2026-06-23 13:07:25.416797210 +0000
+++ b/cef/include/base/cef_build.h	2026-06-23 13:07:25.420000000 +0000
@@ -48,11 +48,12 @@
 ///    COMPILER_MSVC / COMPILER_GCC
 ///
 ///  Processor:
-///    ARCH_CPU_ARM64 / ARCH_CPU_ARMEL / ARCH_CPU_MIPS / ARCH_CPU_MIPS64 /
-///    ARCH_CPU_MIPS64EL / ARCH_CPU_MIPSEL / ARCH_CPU_PPC64 / ARCH_CPU_S390 /
-///    ARCH_CPU_S390X / ARCH_CPU_X86 / ARCH_CPU_X86_64
+///    ARCH_CPU_ARM64 / ARCH_CPU_ARMEL / ARCH_CPU_LOONG64 / ARCH_CPU_MIPS /
+///    ARCH_CPU_MIPS64 / ARCH_CPU_MIPS64EL / ARCH_CPU_MIPSEL / ARCH_CPU_PPC64 /
+///    ARCH_CPU_S390 / ARCH_CPU_S390X / ARCH_CPU_X86 / ARCH_CPU_X86_64
 ///  Processor family:
 ///    ARCH_CPU_ARM_FAMILY: ARMEL or ARM64
+///    ARCH_CPU_LOONG64_FAMILY: LOONG64
 ///    ARCH_CPU_MIPS_FAMILY: MIPS64EL or MIPSEL or MIPS64 or MIPS
 ///    ARCH_CPU_PPC64_FAMILY: PPC64
 ///    ARCH_CPU_S390_FAMILY: S390 or S390X
@@ -247,6 +248,11 @@
 #define ARCH_CPU_32_BITS 1
 #define ARCH_CPU_BIG_ENDIAN 1
 #endif
+#elif defined(__loongarch__)
+#define ARCH_CPU_LOONG64_FAMILY 1
+#define ARCH_CPU_LOONG64 1
+#define ARCH_CPU_64_BITS 1
+#define ARCH_CPU_LITTLE_ENDIAN 1
 #else
 #error Please add support for your architecture in include/base/cef_build.h
 #endif
