diff --git a/PKGBUILD b/PKGBUILD
index 9e456b3..4b6d4e2 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -189,6 +189,37 @@ prepare() {
   # CEF: Remove libxml_visibility patch (fails with system libxml2)
   patch -Np1 -i ../cef-no-libxml-visibility-patch.patch
 
+  # Patches for loong64
+  patch -Np1 -i ../chromium-loong64-support.patch
+  patch -Np1 -i ../allow-sched_getaffinity-in-seccomp-for-loong64.patch
+  patch -Np1 -i ../Fix-build-for-CPU-yield-on-LoongArch.patch
+  patch -Np1 -i ../tools-gn-args-loong64.patch
+  patch -Np1 -i ../cef-make-distrib-loong64.patch
+  patch -Np1 -i ../cef-build-h-loong64.patch
+  patch -Np1 -i ../cef-make-config-header-loong64.patch
+
+  pushd third_party/node/
+  sed -i -e 's/@rollup/rollup/' -e "s/'wasm-node',//" node_modules.py
+  local _rollup_ver="4.32.0" # Known version that supports loong64
+  jq ".dependencies.rollup=\"$_rollup_ver\" | .dependencies.\"@rollup/rollup-linux-loongarch64-gnu\"=\"$_rollup_ver\"" package.json > package.json.new
+  mv package.json{.new,}
+  popd
+
+  third_party/node/update_npm_deps
+
+  pushd third_party/devtools-frontend/src
+  sed -i -e 's/@rollup/rollup/' -e "s/'wasm-node',//" scripts/devtools_paths.py
+  jq ".devDependencies.rollup=\"$_rollup_ver\" | .devDependencies.\"@rollup/rollup-linux-loongarch64-gnu\"=\"$_rollup_ver\"" package.json > package.json.new
+  mv package.json{.new,}
+  sed -i /registry/d .npmrc
+  sed -i 's#@rollup/wasm-node#rollup#g' \
+    inspector_overlay/BUILD.gn \
+    front_end/models/live-metrics/web-vitals-injected/BUILD.gn \
+    front_end/Images/BUILD.gn \
+    front_end/panels/recorder/injected/BUILD.gn \
+    scripts/build/ninja/bundle.gni
+  popd
+
   # CEF: Override clang_exe to use system clang
   echo 'clang_exe = "clang"' >>cef/tools/clang_util.py
 
@@ -207,6 +238,13 @@ prepare() {
   rm -f third_party/gperf/cipd/bin/gperf
   ln -s /usr/bin/gperf third_party/gperf/cipd/bin/
 
+  local _esbuild_ver=$(jq -r '.devDependencies.esbuild // .dependencies.esbuild' third_party/devtools-frontend/src/package.json | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+  npm install "@esbuild/linux-loong64@${_esbuild_ver}"
+  mkdir -p third_party/devtools-frontend/src/third_party/esbuild/
+  mv node_modules/@esbuild/linux-loong64/bin/esbuild third_party/devtools-frontend/src/third_party/esbuild/esbuild
+  rm -r node_modules
+  python3 third_party/devtools-frontend/src/scripts/deps/manage_node_deps.py
+
   if ((!_system_clang)); then
     # Use prebuilt rust as system rust cannot be used due to the error:
     #   error: the option `Z` is only accepted on the nightly compiler
@@ -330,9 +368,22 @@ build() {
     )
   fi
 
+  if [[ $CARCH == loong64 ]]; then
+    _flags+=(
+      'target_cpu="loong64"'
+      'is_cfi=false'
+      'build_litert_with_xnnpack=false'
+      'build_tflite_with_xnnpack=false'
+    )
+  fi
+
   export GN_DEFINES="${_flags[*]}"
   # Only build Release config
-  export GN_OUT_CONFIGS="Release_GN_x64"
+  if [[ $CARCH == loong64 ]]; then
+    export GN_OUT_CONFIGS="Release_GN_loong64"
+  else
+    export GN_OUT_CONFIGS="Release_GN_x64"
+  fi
 
   # Facilitate deterministic builds (taken from build/config/compiler/BUILD.gn)
   CFLAGS+='   -Wno-builtin-macro-redefined'
@@ -363,14 +414,23 @@ build() {
 
   python3 cef/tools/gclient_hook.py
   sed -i '/__sanitizer_set_death_callback/d' v8/src/sandbox/testing.cc
-  ninja -C out/Release_GN_x64 libcef chrome_sandbox
+
+  if [[ $CARCH == loong64 ]]; then
+    local _cef_build_dir='out/Release_GN_loong64'
+    local _cef_build_flag='--loong64-build'
+  else
+    local _cef_build_dir='out/Release_GN_x64'
+    local _cef_build_flag='--x64-build'
+  fi
+
+  ninja -C "$_cef_build_dir" libcef chrome_sandbox
 
   # Build the CEF binary distribution
   python3 cef/tools/make_distrib.py \
     --distrib-subdir=distrib \
     --output-dir=.. \
     --ninja-build \
-    --x64-build \
+    "$_cef_build_flag" \
     --minimal \
     --no-docs \
     --no-archive
@@ -404,4 +464,20 @@ package() {
   install -Dm644 ../chromium-$_chromium_ver/LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE-CHROMIUM"
 }
 
+makedepends+=('jq' 'npm')
+source+=(chromium-loong64-support.patch
+         allow-sched_getaffinity-in-seccomp-for-loong64.patch
+         Fix-build-for-CPU-yield-on-LoongArch.patch
+         tools-gn-args-loong64.patch
+         cef-make-distrib-loong64.patch
+         cef-build-h-loong64.patch
+         cef-make-config-header-loong64.patch)
+sha256sums+=('c498ad298917886739606cd8fac8f526fd05218a460997c0e9100db18b1996bb'
+             'b48d40e93f020b5e8861f1f320437984d8f7d62c568ad1995af78e49e88de7a9'
+             'b88f43b6b96a6b247f4bfbb267904655f57105ffd396c73829cd07759f8b5a1b'
+             'cd1f0d2a48b250ff52802234fe1adb3e0960d27653effc0164b1b58a5dcee33c'
+             '43ca618cc2f06ae9eac6ade59eed70b1b3895bbe44fe358d1575c0c62e294fcc'
+             '64ee44c5e488d43f87b6fbcd7f366fdf69d36c9e4c6d59ba67de756581a506ac'
+             '7f5c8ca01dfa811ee3b207e7955e1d4216d30da3e0562cf384141f3a53665a80')
+
 # vim:set ts=2 sw=2 et:
