diff --git a/PKGBUILD b/PKGBUILD
index 84f85ce..e695478 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -201,6 +201,36 @@ prepare() {
   # CEF: Remove libxml_visibility patch (fails with system libxml2)
   patch -Np1 -i ../cef-no-libxml-visibility-patch.patch
 
+  # Patches for loong64
+  patch -Np1 -i ../chromium-loong64-support.patch
+  patch -Np1 -i ../allow-sched_getaffinity-in-seccomp-for-loong64.patch
+  patch -Np1 -i ../tools-gn-args-loong64.patch
+  patch -Np1 -i ../cef-make-distrib-loong64.patch
+  patch -Np1 -i ../cef-build-h-loong64.patch
+  patch -Np1 -i ../cef-make-config-header-loong64.patch
+  patch -Np1 -i ../dawn-add-loongarch64-support.patch
+  pushd third_party/node/
+  sed -i -e 's/@rollup/rollup/' -e "s/'wasm-node',//" node_modules.py
+  local _rollup_ver="4.32.0" # Known version that supports loong64
+  jq ".dependencies.rollup=\"$_rollup_ver\" | .dependencies.\"@rollup/rollup-linux-loongarch64-gnu\"=\"$_rollup_ver\"" package.json > package.json.new
+  mv package.json{.new,}
+  popd
+
+  third_party/node/update_npm_deps
+
+  pushd third_party/devtools-frontend/src
+  sed -i -e 's/@rollup/rollup/' -e "s/'wasm-node',//" scripts/devtools_paths.py
+  jq ".devDependencies.rollup=\"$_rollup_ver\" | .devDependencies.\"@rollup/rollup-linux-loongarch64-gnu\"=\"$_rollup_ver\"" package.json > package.json.new
+  mv package.json{.new,}
+  sed -i /registry/d .npmrc
+  sed -i 's#@rollup/wasm-node#rollup#g' \
+    inspector_overlay/BUILD.gn \
+    front_end/models/live-metrics/web-vitals-injected/BUILD.gn \
+    front_end/Images/BUILD.gn \
+    front_end/panels/recorder/injected/BUILD.gn \
+    scripts/build/ninja/bundle.gni
+  popd
+
   # CEF: Override clang_exe to use system clang
   echo 'clang_exe = "clang"' >>cef/tools/clang_util.py
 
@@ -224,6 +254,13 @@ prepare() {
   rm -f third_party/gperf/cipd/bin/gperf
   ln -s /usr/bin/gperf third_party/gperf/cipd/bin/
 
+  local _esbuild_ver=$(jq -r '.devDependencies.esbuild // .dependencies.esbuild' third_party/devtools-frontend/src/package.json | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+  npm install "@esbuild/linux-loong64@${_esbuild_ver}"
+  mkdir -p third_party/devtools-frontend/src/third_party/esbuild/
+  mv node_modules/@esbuild/linux-loong64/bin/esbuild third_party/devtools-frontend/src/third_party/esbuild/esbuild
+  rm -r node_modules
+  python3 third_party/devtools-frontend/src/scripts/deps/manage_node_deps.py
+
   if ((!_system_clang)); then
     # Use prebuilt rust as system rust cannot be used due to the error:
     #   error: the option `Z` is only accepted on the nightly compiler
@@ -347,11 +384,22 @@ build() {
     )
   fi
 
+  if [[ $CARCH == loong64 ]]; then
+    _flags+=(
+      'target_cpu="loong64"'
+      'is_cfi=false'
+      'build_litert_with_xnnpack=false'
+      'build_tflite_with_xnnpack=false'
+    )
+  fi
+
   export GN_DEFINES="${_flags[*]}"
   # Only build Release config
   if [[ $CARCH == "aarch64" ]]; then
     export GN_DEFINES+=' target_cpu="arm64"'
     export GN_OUT_CONFIGS="Release_GN_arm64"
+  elif [[ $CARCH == loong64 ]]; then
+    export GN_OUT_CONFIGS="Release_GN_loong64"
   else
     export GN_OUT_CONFIGS="Release_GN_x64"
   fi
@@ -394,6 +442,8 @@ build() {
 
   if [[ $CARCH == "aarch64" ]]; then
     ninja -C out/Release_GN_arm64 libcef chrome_sandbox
+  elif [[ $CARCH == loong64 ]]; then
+    ninja -C out/Release_GN_loong64 libcef chrome_sandbox
   else
     ninja -C out/Release_GN_x64 libcef chrome_sandbox
   fi
@@ -401,6 +451,8 @@ build() {
   # Build the CEF binary distribution
   if [[ $CARCH == "aarch64" ]]; then
     local _distrib_arch_flag="--arm64-build"
+  elif [[ $CARCH == loong64 ]]; then
+    local _distrib_arch_flag="--loong64-build"
   else
     local _distrib_arch_flag="--x64-build"
   fi
@@ -443,4 +495,20 @@ package() {
   install -Dm644 ../chromium-$_chromium_ver/LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE-CHROMIUM"
 }
 
+makedepends+=('jq' 'npm')
+source+=(chromium-loong64-support.patch
+         allow-sched_getaffinity-in-seccomp-for-loong64.patch
+         tools-gn-args-loong64.patch
+         cef-make-distrib-loong64.patch
+         cef-build-h-loong64.patch
+         cef-make-config-header-loong64.patch
+         dawn-add-loongarch64-support.patch)
+sha256sums+=('941b1b63ae4e4432c9b9b6a3fea9a6f5352235c51bf7f980e5e249023fe84866'
+             'b48d40e93f020b5e8861f1f320437984d8f7d62c568ad1995af78e49e88de7a9'
+             'cd1f0d2a48b250ff52802234fe1adb3e0960d27653effc0164b1b58a5dcee33c'
+             '19ed7042f7b2f35c05f70fe97937ed32bd602e9b961d77da1692e5554f823b45'
+             '64ee44c5e488d43f87b6fbcd7f366fdf69d36c9e4c6d59ba67de756581a506ac'
+             '7f5c8ca01dfa811ee3b207e7955e1d4216d30da3e0562cf384141f3a53665a80'
+             'cfdb92dbd54a2e4f113603ac876debd050d5e54492e516666f71bb73252d9425')
+
 # vim:set ts=2 sw=2 et:
