--- a/cef/tools/gn_args.py	2026-06-22 23:42:54.000000000 +0000
+++ b/cef/tools/gn_args.py	2026-06-22 23:59:22.636363840 +0000
@@ -428,8 +428,8 @@
                           'arm64'), 'target_cpu must be "x86", "x64" or "arm64"'
   elif platform == 'linux':
     assert target_cpu in (
-        'x86', 'x64', 'arm',
-        'arm64'), 'target_cpu must be "x86", "x64", "arm" or "arm64"'
+        'x86', 'x64', 'arm', 'arm64',
+        'loong64'), 'target_cpu must be "x86", "x64", "arm", "arm64" or "loong64"'
 
   # ASAN requires Release builds.
   if is_asan:
@@ -611,6 +611,8 @@
     sysroot_name = 'debian_%s_armhf-sysroot' % release
   elif cpu == 'arm64':
     sysroot_name = 'debian_%s_arm64-sysroot' % release
+  elif cpu == 'loong64':
+    sysroot_name = 'debian_%s_loong64-sysroot' % release
   else:
     raise Exception('Unrecognized sysroot CPU: %s' % cpu)
 
