--- a/PKGBUILD
+++ b/PKGBUILD
@@ -35,6 +35,7 @@ prepare() {
   # fix libbpf build
   patch -Np1 -i "$srcdir/0001-Fix-libbpf-build-workaround-for-netdata-libbpf-4.patch"
 
+  sed -i "s|OBJDIR=build/|OBJDIR=build/ LIBSUBDIR=lib|g" packaging/cmake/Modules/NetdataLibBPF.cmake
   sed -i "s/CMAKE_CXX_STANDARD 14/CMAKE_CXX_STANDARD 17/" CMakeLists.txt
   sed -i "s|usr/sbin|usr/bin|g" CMakeLists.txt
   sed -i "s|usr/libexec|usr/lib|g" CMakeLists.txt
