diff --git a/PKGBUILD b/PKGBUILD
index 5127652..79fa190 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -29,3 +29,11 @@ package() {
   cd brook/cli/brook
   install -Dm755 brook -t "$pkgdir"/usr/bin/
 }
+
+prepare() {
+  cd brook/cli/brook
+  go mod edit -replace=golang.org/x/sys=golang.org/x/sys@v0.26.0
+  go mod edit -replace=golang.org/x/net=golang.org/x/net@v0.30.0
+  go mod edit -replace=github.com/phuslu/iploc=github.com/phuslu/iploc@v1.0.20250801
+  go mod tidy
+}
