diff --git a/PKGBUILD b/PKGBUILD
index 252211c..25dddc7 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -111,11 +111,18 @@ prepare() {
   patch -Np1 -d third_party/skia <../skia-fix-cfi-icall-failure-with-use_system_libjpeg-true.patch
 
   # Allow libclang_rt.builtins from compiler-rt >= 16 to be used
-  patch -Np1 -i ../compiler-rt-adjust-paths.patch
+  # Use another version to support loong64
+  #patch -Np1 -i ../compiler-rt-adjust-paths.patch
+  patch -Np1 -i "${srcdir}/compiler-rt-adjust-paths-loong64.patch"
 
   # Increase _FORTIFY_SOURCE level to match Arch's default flags
   patch -Np1 -i ../increase-fortify-level.patch
 
+  # Patches for loong64
+  patch -Np1 -i "${srcdir}/chromium-loong64-support.patch"
+  patch -Np1 -i "${srcdir}/allow-sched_getaffinity-in-seccomp-for-loong64.patch"
+  patch -Np1 -d third_party/swiftshader -i "${srcdir}/swiftshader-cpp-fix.patch"
+
   # Fixes for building with libstdc++ instead of libc++
 
   # Link to system tools required by the build
@@ -207,7 +214,7 @@ build() {
       'clang_base_path="/usr"'
       'clang_use_chrome_plugins=false'
       "clang_version=\"$_clang_version\""
-      #'chrome_pgo_phase=0' # needs newer clang to read the bundled PGO profile
+      'chrome_pgo_phase=0' # Disable PGO since the data is not for loong64
     )
 
     # Allow the use of nightly features with stable Rust compiler
@@ -324,4 +331,13 @@ package() {
   install -Dm644 LICENSE "$pkgdir/usr/share/licenses/chromium/LICENSE"
 }
 
+source+=("chromium-loong64-support.patch"
+         "compiler-rt-adjust-paths-loong64.patch"
+         "allow-sched_getaffinity-in-seccomp-for-loong64.patch"
+         "swiftshader-cpp-fix.patch::https://raw.githubusercontent.com/riscv-forks/electron/27a4ff46397145e1cec1ff34ebf36a5bdfa8788b/patches/swiftshader/0001-Fix-LLVM-s-AlignOf-after-previous-change-to-support-.patch")
+sha256sums+=('e849f525f61464ddd1ae622a3b07c3633e01e982ccd00d5accfaa69163819285'
+             '56e8d50b7c744f51953990aefceeae5b7dd08063baaf06df98ddeec02a2d4690'
+             'b48d40e93f020b5e8861f1f320437984d8f7d62c568ad1995af78e49e88de7a9'
+             '046219fd2287dd199cb03981b5ad73f08e99d517bd3f6f6f265d5ce37f2fba1e')
+
 # vim:set ts=2 sw=2 et:
