diff --git a/PKGBUILD b/PKGBUILD
index 740f343..5dcc502 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -54,6 +54,7 @@ optdepends=('pipewire: WebRTC desktop sharing under Wayland'
             'org.freedesktop.secrets: password storage backend on GNOME, KDE and Xfce'
             'upower: Battery Status API support')
 options=('!lto') # Chromium adds its own flags for ThinLTO
+_manual_clone=0
 source=(https://commondatastorage.googleapis.com/chromium-browser-official/chromium-$pkgver-lite.tar.xz
         https://github.com/foutrelis/chromium-launcher/archive/v$_launcher_ver/chromium-launcher-$_launcher_ver.tar.gz
         chromium-138-nodejs-version-check.patch
@@ -83,6 +84,8 @@ if (( _manual_clone )); then
   source[0]=fetch-chromium-release
   sha256sums[0]='2e2f36e3cd1ebc4ad57fd310774a5e5e9db77883d5f9374fedeaabd3c103b819'
   makedepends+=('python-httplib2' 'python-pyparsing' 'python-six' 'npm' 'rsync')
+else
+  sha256sums[0]="$(curl -sL "${source[0]}.hashes" | awk '$1 == "sha256" {print $2; exit}')"
 fi
 
 # Possible replacements are listed in build/linux/unbundle/replace_gn_files.py
@@ -211,6 +214,46 @@ prepare() {
   python3 build/util/lastchange.py -m DAWN_COMMIT_HASH \
     -s third_party/dawn --revision gpu/webgpu/DAWN_VERSION \
     --header gpu/webgpu/dawn_commit_hash.h
+
+  # Patches for loong64
+  patch -Np1 -i "${srcdir}/chromium-loong64-support.patch"
+  patch -Np1 -i "${srcdir}/allow-sched_getaffinity-in-seccomp-for-loong64.patch"
+
+  pushd third_party/node/
+  sed -i -e 's/@rollup/rollup/' -e "s/'wasm-node',//" node_modules.py
+  #local _rollup_ver="$(jq -r .dependencies.\"@rollup/wasm-node\" package.json)"
+  local _rollup_ver="4.32.0" # Known version that supports loong64
+  jq ".dependencies.rollup=\"$_rollup_ver\" | .dependencies.\"@rollup/rollup-linux-loongarch64-gnu\"=\"$_rollup_ver\"" package.json > package.json.new
+  mv package.json{.new,}
+  popd
+
+  # https://gitlab.archlinux.org/archlinux/packaging/packages/electron32/-/issues/1
+  third_party/node/update_npm_deps
+
+  pushd third_party/devtools-frontend/src
+  sed -i -e 's/@rollup/rollup/' -e "s/'wasm-node',//" scripts/devtools_paths.py
+  #local _rollup_ver="$(jq -r .devDependencies.\"@rollup/wasm-node\" package.json)"
+  jq ".devDependencies.rollup=\"$_rollup_ver\" | .devDependencies.\"@rollup/rollup-linux-loongarch64-gnu\"=\"$_rollup_ver\""  package.json > package.json.new
+  mv package.json{.new,}
+  # Chromium hosts a custom registry at https://npm.skia.org/chrome-devtools/
+  # and rejects some packages:
+  # Package fs-extra with version 11.3.0 was created 108h0m0s time ago. This is less than 1 week and so failed the audit.
+  sed -i /registry/d .npmrc
+  # Replace direct invocation of wasm rollup
+  sed -i 's\@rollup/wasm-node\rollup\' \
+    inspector_overlay/BUILD.gn \
+    front_end/models/live-metrics/web-vitals-injected/BUILD.gn \
+    front_end/Images/BUILD.gn \
+    front_end/panels/recorder/injected/BUILD.gn \
+    scripts/build/ninja/bundle.gni
+  popd
+  # Get the binary from the npm package
+  local _esbuild_ver=$(jq -r '.devDependencies.esbuild // .dependencies.esbuild' third_party/devtools-frontend/src/package.json | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+  npm install "@esbuild/linux-loong64@${_esbuild_ver}"
+  mkdir -p third_party/devtools-frontend/src/third_party/esbuild/
+  mv node_modules/@esbuild/linux-loong64/bin/esbuild third_party/devtools-frontend/src/third_party/esbuild/esbuild
+  rm -r node_modules # Cleaning
+  python3 third_party/devtools-frontend/src/scripts/deps/manage_node_deps.py
 }
 
 build() {
@@ -318,6 +361,9 @@ build() {
     CXXFLAGS="${CXXFLAGS/-march=*([^ ]) }"
   fi
 
+  # Disable xnnpack on loong64
+  _flags+=('build_tflite_with_xnnpack=false')
+
   gn gen out/Release --args="${_flags[*]}"
   ninja -C out/Release chrome chrome_sandbox chromedriver.unstripped
 }
@@ -395,4 +441,10 @@ package() {
   install -Dvm644 LICENSE "$pkgdir/usr/share/licenses/chromium/LICENSE"
 }
 
+source+=("chromium-loong64-support.patch"
+         "allow-sched_getaffinity-in-seccomp-for-loong64.patch")
+sha256sums+=('c498ad298917886739606cd8fac8f526fd05218a460997c0e9100db18b1996bb'
+             'b48d40e93f020b5e8861f1f320437984d8f7d62c568ad1995af78e49e88de7a9')
+makedepends+=(jq npm rsync)
+
 # vim:set ts=2 sw=2 et:
