diff --git a/PKGBUILD b/PKGBUILD
index 01d4648..4604296 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -112,7 +112,13 @@ prepare() {
   patch -Np1 -i ../chromium-138-nodejs-version-check.patch
 
   # Allow libclang_rt.builtins from compiler-rt >= 16 to be used
-  patch -Np1 -i ../compiler-rt-adjust-paths.patch
+  # Use another version to support loong64
+  #patch -Np1 -i ../compiler-rt-adjust-paths.patch
+  patch -Np1 -i "${srcdir}/compiler-rt-adjust-paths-loong64.patch"
+
+  # Patches for loong64
+  patch -Np1 -i "${srcdir}/chromium-loong64-support.patch"
+  patch -Np1 -i "${srcdir}/allow-sched_getaffinity-in-seccomp-for-loong64.patch" 
 
   # Increase _FORTIFY_SOURCE level to match Arch's default flags
   patch -Np1 -i ../increase-fortify-level.patch
@@ -205,7 +211,7 @@ build() {
       'clang_base_path="/usr"'
       'clang_use_chrome_plugins=false'
       "clang_version=\"$_clang_version\""
-      #'chrome_pgo_phase=0' # needs newer clang to read the bundled PGO profile
+      'chrome_pgo_phase=0' # Disable PGO since the data is not for loong64
     )
 
     # Allow the use of nightly features with stable Rust compiler
@@ -246,6 +252,8 @@ build() {
   # https://crbug.com/957519#c122
   CXXFLAGS=${CXXFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS}
 
+  # Disable xnnpack on loong64
+  _flags+=('build_tflite_with_xnnpack=false')
   gn gen out/Release --args="${_flags[*]}"
   ninja -C out/Release chrome chrome_sandbox chromedriver.unstripped
 }
@@ -322,4 +330,14 @@ package() {
   install -Dm644 LICENSE "$pkgdir/usr/share/licenses/chromium/LICENSE"
 }
 
+source+=("chromium-loong64-support.patch"
+         "compiler-rt-adjust-paths-loong64.patch"
+         "allow-sched_getaffinity-in-seccomp-for-loong64.patch")
+sha256sums+=('9d96b078934e41b6d14645ead011a06fb7a079f5b0725e5b07811be5b76e0e5d'
+             '640ea1fa9cd6eac337afc41520184061fe60dd994613da7ba992e0e73f88057d'
+             'b48d40e93f020b5e8861f1f320437984d8f7d62c568ad1995af78e49e88de7a9')
+# nodejs 24 is broken upstream
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(nodejs)$'))
+makedepends+=(nodejs-lts-jod)
+
 # vim:set ts=2 sw=2 et:
