diff --git a/PKGBUILD b/PKGBUILD
index b3db102..5366a0e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,6 +27,7 @@ optdepends=('pipewire: WebRTC desktop sharing under Wayland'
             'kwallet: support for storing passwords in KWallet on Plasma'
             'upower: Battery Status API support')
 options=('!lto') # Chromium adds its own flags for ThinLTO
+_manual_clone=0
 source=(https://commondatastorage.googleapis.com/chromium-browser-official/chromium-$pkgver-lite.tar.xz
         https://github.com/foutrelis/chromium-launcher/archive/v$_launcher_ver/chromium-launcher-$_launcher_ver.tar.gz
         chromium-138-nodejs-version-check.patch
@@ -47,6 +48,8 @@ sha256sums=('720a1196410080056cd97a1f5ec34d68ba216a281d9b5157b7ea81ea018ec661'
 if (( _manual_clone )); then
   source[0]=fetch-chromium-release
   makedepends+=('python-httplib2' 'python-pyparsing' 'python-six' 'npm' 'rsync')
+else
+  sha256sums[0]="$(curl -sL "${source[0]}.hashes" | awk '$1 == "sha256" {print $2; exit}')"
 fi
 
 # Possible replacements are listed in build/linux/unbundle/replace_gn_files.py
@@ -124,6 +127,10 @@ prepare() {
   # Allow libclang_rt.builtins from compiler-rt >= 16 to be used
   patch -Np1 -i ../compiler-rt-adjust-paths.patch
 
+  # Patches for loong64
+  patch -Np1 -i "${srcdir}/chromium-loong64-support.patch"
+  patch -Np1 -i "${srcdir}/allow-sched_getaffinity-in-seccomp-for-loong64.patch"
+
   # Increase _FORTIFY_SOURCE level to match Arch's default flags
   patch -Np1 -i ../increase-fortify-level.patch
 
@@ -263,6 +270,9 @@ build() {
   # https://crbug.com/957519#c122
   CXXFLAGS=${CXXFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS}
 
+  # Disable xnnpack on loong64
+  _flags+=('build_tflite_with_xnnpack=false')
+
   gn gen out/Release --args="${_flags[*]}"
   ninja -C out/Release chrome chrome_sandbox chromedriver.unstripped
 }
@@ -338,4 +348,9 @@ package() {
   install -Dm644 LICENSE "$pkgdir/usr/share/licenses/chromium/LICENSE"
 }
 
+source+=("chromium-loong64-support.patch"
+         "allow-sched_getaffinity-in-seccomp-for-loong64.patch")
+sha256sums+=('65da85f63d4554958fd6166a4b0cb7fbeea91ab6253c4f016249efec0ddda4cd'
+             'b48d40e93f020b5e8861f1f320437984d8f7d62c568ad1995af78e49e88de7a9')
+
 # vim:set ts=2 sw=2 et:
