diff --git a/PKGBUILD b/PKGBUILD
index 15a7da1..3730628 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -116,7 +116,13 @@ prepare() {
   patch -Np1 -i ../chromium-136-drop-nodejs-ver-check.patch
 
   # Allow libclang_rt.builtins from compiler-rt >= 16 to be used
-  patch -Np1 -i ../compiler-rt-adjust-paths.patch
+  # Use another version to support loong64
+  #patch -Np1 -i ../compiler-rt-adjust-paths.patch
+  patch -Np1 -i "${srcdir}/compiler-rt-adjust-paths-loong64.patch"
+
+  # Patches for loong64
+  patch -Np1 -i "${srcdir}/chromium-loong64-support.patch"
+  patch -Np1 -i "${srcdir}/allow-sched_getaffinity-in-seccomp-for-loong64.patch" 
 
   # Increase _FORTIFY_SOURCE level to match Arch's default flags
   patch -Np1 -i ../increase-fortify-level.patch
@@ -211,7 +217,7 @@ build() {
       'clang_base_path="/usr"'
       'clang_use_chrome_plugins=false'
       "clang_version=\"$_clang_version\""
-      #'chrome_pgo_phase=0' # needs newer clang to read the bundled PGO profile
+      'chrome_pgo_phase=0' # Disable PGO since the data is not for loong64
     )
 
     # Allow the use of nightly features with stable Rust compiler
@@ -328,4 +334,11 @@ package() {
   install -Dm644 LICENSE "$pkgdir/usr/share/licenses/chromium/LICENSE"
 }
 
+source+=("chromium-loong64-support.patch"
+         "compiler-rt-adjust-paths-loong64.patch"
+         "allow-sched_getaffinity-in-seccomp-for-loong64.patch")
+sha256sums+=('eb2ff43f15d569b6a91eec2a0a12836638e549b82d0bd72edcc448ea8095c328'
+             '56e8d50b7c744f51953990aefceeae5b7dd08063baaf06df98ddeec02a2d4690'
+             'b48d40e93f020b5e8861f1f320437984d8f7d62c568ad1995af78e49e88de7a9')
+
 # vim:set ts=2 sw=2 et:
