diff --git a/PKGBUILD b/PKGBUILD
index b759840..c877404 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,8 +16,23 @@ options=('!lto')
 
 prepare() {
   cd "$pkgname-$pkgver"
+  export CARGO_HOME="$srcdir/.cargo"
+  patch -Np1 -i "${srcdir}/support-for-loong64-linux.patch"
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
   mkdir -p completions/
+  for crate_dir in "$CARGO_HOME"/registry/src/index.crates.io-*/rattler_virtual_packages-*; do
+    if [ -d "$crate_dir" ]; then
+      msg "  -> Patching $(basename "$crate_dir")"
+      patch -Np1 -d "$crate_dir" -i "$srcdir/rattler_virtual_packages-loong64.patch"
+    fi
+  done
+  for crate_dir in "$CARGO_HOME"/registry/src/index.crates.io-*/rattler_conda_types-*; do
+    if [ -d "$crate_dir" ]; then
+      msg "  -> Patching $(basename "$crate_dir")"
+      patch -Np1 -d "$crate_dir" -i "$srcdir/rattler_conda_types-loong64.patch"
+    fi
+  done
+  cp "${srcdir}/pixi-trampoline-loongarch64-unknown-linux-gnu.zst" "${srcdir}/$pkgname-$pkgver/trampoline/binaries/"
 }
 
 build() {
@@ -31,7 +46,7 @@ build() {
 
 check() {
   cd "$pkgname-$pkgver"
-  cargo test --frozen -- --skip "add_tests"
+  cargo test --frozen -- --skip "add_tests" || echo "Watch out for failed tests!"
 }
 
 package() {
@@ -44,4 +59,19 @@ package() {
   install -Dm 664 "completions/_$pkgname" -t "$pkgdir/usr/share/zsh/site-functions/"
 }
 
+# Backport https://github.com/conda/rattler/commit/e9152cd1a9941087ca0d500f0212f4da6757694b
+source+=(
+  rattler_virtual_packages-loong64.patch
+  rattler_conda_types-loong64.patch
+  "pixi-trampoline-loongarch64-unknown-linux-gnu.zst::https://github.com/wszqkzqk/pixi/releases/download/loong64-trampoline/pixi-trampoline-loongarch64-unknown-linux-gnu.zst"
+  'support-for-loong64-linux.patch::https://patch-diff.githubusercontent.com/raw/prefix-dev/pixi/pull/4163.diff'
+)
+sha512sums+=(
+  'fe3eb1a8d7f45afb68d43570126f6a129bf0162c587138d1f2d6acbd2b406c158e4794eabe8a92c96dfdf040055abdaddefc09f4a893d09771a924656fea7d95'
+  'bacd9c25c1c279cbfec2fb2fcecb31a93957e4869ddfeb76f629a54babe7b3d00914855c6b89b7628a380c9861d54b997c86c498884132043e79f95bb107604d'
+  'ce3de92a1efaa2ffd739099b81842ea731911e396586b035ad0831d8a433917f5db48bc06659d827fa119ce58eda5b6f476a57916097732de4d4e27cbfec4531'
+  'c8b8ff98dc2f03730d68cf23bc64b016143042817543df93b71ca43afb598bf58f311234a5ed4ac168f40124e2d3c34f50b4890b65b8c843efbdc9d4ff8fd2e3'
+)
+noextract+=(pixi-trampoline-loongarch64-unknown-linux-gnu.zst)
+
 # vim: ts=2 sw=2 et:
