--- a/src/lib.rs
+++ b/src/lib.rs
@@ -591,6 +591,7 @@ impl Archspec {
             Platform::Win32 | Platform::Linux32 => "x86",
             Platform::Win64 | Platform::Osx64 | Platform::Linux64 => "x86_64",
             Platform::LinuxAarch64 | Platform::LinuxArmV6l | Platform::LinuxArmV7l => "aarch64",
+            Platform::LinuxLoong64 => "loong64",
             Platform::LinuxPpc64le => "ppc64le",
             Platform::LinuxPpc64 => "ppc64",
             Platform::LinuxPpc => "ppc",
