--- a/PKGBUILD
+++ b/PKGBUILD
@@ -33,6 +33,9 @@ prepare() {
 
   # Fix build with GCC 15
   git apply -3 ../0001-Fix-build-with-GCC-15.patch
+
+  # Fix loongarch64 multi-bit-depth build
+  git apply -3 ../0002-Fix-loongarch64-build-with-multiple-bit-depths.patch
 }
 
 build() {
@@ -73,4 +76,6 @@ package() {
   DESTDIR="${pkgdir}" cmake --install build
 }
 
+b2sums+=('eb5e022096ca9c9707c9cf010b19b811f91fb8ddd5fd296927b4c98dde3379106a3ecc44f9c662a3f8e4871d37ff6d9a73a41e162244c690287bccdc07df1163')
+source+=(0002-Fix-loongarch64-build-with-multiple-bit-depths.patch)
 # vim: ts=2 sw=2 et:
