diff --git a/PKGBUILD b/PKGBUILD
index c07c678..21e19e4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -50,8 +50,9 @@ build() {
     -DBUILD_SHARED_LIBS=BOTH \
     -DBUILD_LTO_LIBS=ON \
     -DLDC_WITH_LLD=OFF \
-    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=lld --flto=thin" \
+    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=lld --flto=thin -L--plugin-opt=-mattr=+d" \
     -DADDITIONAL_DEFAULT_LDC_SWITCHES="\"-link-defaultlib-shared\"," \
+    -DLD_FLAGS="-Wl,--no-as-needed -latomic -Wl,--as-needed" \
     ..
     ninja
 }
