diff --git a/PKGBUILD b/PKGBUILD
index 4c6549b..81ea438 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,6 +26,9 @@ b2sums=('0bae800496db6ffd0a9b24b356df86627a8049ff75e9f8a38d8594182f80f24a98d59d9
         'e8f2b19e2474f30a4f984b45787950012668bf0acb5ad1ebb25cd9776925ab4a6aa927f8131ed53e35b1c71b32c504c700fe5b5145ecd25c7a8284373bb951ed')
 
 build() {
+  # Enable LASX to workaround the compile error with LSX
+  export CFLAGS="${CFLAGS} -mlasx"
+  export CXXFLAGS="${CXXFLAGS} -mlasx"
   export CGO_CPPFLAGS="${CPPFLAGS}"
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
@@ -89,3 +92,7 @@ package_ollama-docs() {
   cp -r ollama/docs "$pkgdir/usr/share/doc/ollama"
   install -Dm644 ollama/LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+# Remove cuda
+pkgname=(${pkgname[@]/ollama-cuda})
+makedepends=(${makedepends[@]/cuda})
