--- a/PKGBUILD
+++ b/PKGBUILD
@@ -28,7 +28,6 @@ checkdepends=(
   boost
   clang
   cmake
-  cuda
   cython
   doxygen
   gcc-fortran
@@ -42,7 +41,7 @@ checkdepends=(
   graphviz
   gtest
   gtk-doc
-  gtk-sharp-2
+#  gtk-sharp-2
   gtk3
   gtkmm3
   hdf5
@@ -55,13 +54,13 @@ checkdepends=(
   libwmf
   llvm
   mercurial
-  mono
+#  mono
   nasm
   netcdf-fortran
   openmpi
   openssh
   protobuf
-  pypy3
+#  pypy3
   python-gobject
   python-lxml
   python-pytest-xdist
@@ -74,7 +73,7 @@ checkdepends=(
   rust-bindgen
   sdl2
   vala
-  valgrind
+#  valgrind
   vulkan-headers
   vulkan-icd-loader
   vulkan-validation-layers
