diff --git a/PKGBUILD b/PKGBUILD
index 4a03ce2..3536bf4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,7 +25,6 @@ checkdepends=(
   boost
   clang
   cmake
-  cuda
   cython
   doxygen
   gcc-fortran
@@ -39,7 +38,7 @@ checkdepends=(
   graphviz
   gtest
   gtk-doc
-  gtk-sharp-2
+#  gtk-sharp-2
   gtk3
   gtkmm3
   hotdoc
@@ -50,7 +49,7 @@ checkdepends=(
   libwmf
   llvm
   mercurial
-  mono
+#  mono
   nasm
   netcdf-fortran
   openmpi
@@ -64,7 +63,7 @@ checkdepends=(
   rust-bindgen
   sdl2
   vala
-  valgrind
+#  valgrind
   vulkan-validation-layers
   wxgtk3
 )
