--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,11 +21,10 @@ makedepends=(
   python-setuptools
   python-wheel
 )
-checkdepends=(
+makedepends+=(
   boost
   clang
   cmake
-  cuda
   cython
   doxygen
   gcc-fortran
@@ -39,18 +38,18 @@ checkdepends=(
   graphviz
   gtest
   gtk-doc
-  gtk-sharp-2
+#  gtk-sharp-2
   gtk3
   gtkmm3
   hotdoc
   itstool
   java-environment=8
-  ldc
+#  ldc
   libelf
   libwmf
   llvm
   mercurial
-  mono
+#  mono
   nasm
   netcdf-fortran
   openmpi
@@ -64,7 +63,7 @@ checkdepends=(
   rust-bindgen
   sdl2
   vala
-  valgrind
+#  valgrind
   vulkan-validation-layers
   wxgtk3
 )
