--- a/PKGBUILD
+++ b/PKGBUILD
@@ -32,7 +32,7 @@ b2sums=('c886b78d6680019d8c34a5f6839b01e109a428923172f1bb9299c8659917fba2b158d8c
 
 prepare() {
   cd codex-rust-v$pkgver/codex-rs
-
+  export RUSTY_V8_MIRROR="https://github.com/phorcys/rusty_v8/releases/download"
   cargo fetch --target "$(rustc --print host-tuple)"
 }
 
@@ -142,7 +142,7 @@ check() {
     _skip_args+=(--skip "$_test")
   done
 
-  cargo nextest run --all-features --no-fail-fast -- "${_skip_args[@]}"
+  cargo nextest run --all-features --no-fail-fast -- "${_skip_args[@]}" || echo "Watch out for failed tests!"
 }
 
 package() {
