diff --git a/PKGBUILD b/PKGBUILD
index b446af8..aa61a30 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -99,7 +99,7 @@ package_elfutils() {
   make DESTDIR="$pkgdir" install -C $pkgbase
 
   # set the default DEBUGINFOD_URLS environment variable to the distribution's debuginfod URL
-  echo "https://debuginfod.archlinux.org" > "$pkgdir/etc/debuginfod/archlinux.urls"
+  echo "https://loongarchlinux.lcpu.dev/debuginfod" > "$pkgdir/etc/debuginfod/archlinux.urls"
 
   (
     cd "$pkgdir"
