diff --git a/PKGBUILD b/PKGBUILD
index a5411a1..05f154c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -77,7 +77,7 @@ check() {
 	local python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
 	python -m venv --system-site-packages test-env
 	test-env/bin/python -m installer dist/*.whl
-	pushd build/lib.linux-$CARCH-cpython-${python_version}
+	pushd build/lib.linux-$(uname -m)-cpython-${python_version}
 	ln -sv ../../tests .
 	PYTHONPATH="$PWD" PATH="../../test-env/bin:$PATH" ../../test-env/bin/python -m twisted.trial -j$(nproc) tests
 	rm -r tests _trial_temp
