diff --git a/PKGBUILD b/PKGBUILD
index f431ee6..0b5d88e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,12 +21,13 @@ b2sums=('2483d2b833d566508680a5b9b3949aef187c290ab5602ef423ede214cb57f4b8f61dfeb
 build() {
   cd findlib-$pkgver
   ./configure -config /etc/findlib.conf -mandir /usr/share/man -sitelib /usr/lib/ocaml
-  make all opt
+  make all #opt
 }
 
 package () {
   cd findlib-$pkgver
   make prefix="$pkgdir" install
-  install -Dm755 src/findlib/ocamlfind_opt "$pkgdir/usr/bin/ocamlfind_opt"
+  # ocamlfind_opt is not supported
+  #install -Dm755 src/findlib/ocamlfind_opt "$pkgdir/usr/bin/ocamlfind_opt"
   install -Dm644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
