From 0353d8225644784eb5838d9d02c6e9fb4c9bb697 Mon Sep 17 00:00:00 2001
From: wszqkzqk <wszqkzqk@qq.com>
Date: Mon, 25 Nov 2024 22:42:56 +0800
Subject: [PATCH] BLI: Add loong64 support in BLI_build_config.h

Signed-off-by: Zhou Qiankang <wszqkzqk@qq.com>
---
 source/blender/blenlib/BLI_build_config.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/source/blender/blenlib/BLI_build_config.h b/source/blender/blenlib/BLI_build_config.h
index c45d5b065f4..2ccfcb71787 100644
--- a/source/blender/blenlib/BLI_build_config.h
+++ b/source/blender/blenlib/BLI_build_config.h
@@ -354,6 +354,11 @@
 #  else
 #    define ARCH_CPU_BIG_ENDIAN 1
 #  endif
+#elif defined(__loongarch_lp64)
+#  define ARCH_CPU_LOONG_FAMILY 1
+#  define ARCH_CPU_LOONG64 1
+#  define ARCH_CPU_64_BITS 1
+#  define ARCH_CPU_LITTLE_ENDIAN 1
 #else
 #  error Please add support for your architecture in BLI_build_config.h
 #endif
@@ -396,6 +401,9 @@
 #if !defined(ARCH_CPU_RISCV_FAMILY)
 #  define ARCH_CPU_RISCV_FAMILY 0
 #endif
+#if !defined(ARCH_CPU_LOONG_FAMILY)
+#  define ARCH_CPU_LOONG_FAMILY 0
+#endif
 
 #if !defined(ARCH_CPU_ARM64)
 #  define ARCH_CPU_ARM64 0
@@ -439,5 +447,8 @@
 #if !defined(ARCH_CPU_RISCV128)
 #  define ARCH_CPU_RISCV128 0
 #endif
+#if !defined(ARCH_CPU_LOONG64)
+#  define ARCH_CPU_LOONG64 0
+#endif
 
 /** \} */
