diff --git a/PKGBUILD b/PKGBUILD
index 49a49f6..e516dba 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -133,6 +133,11 @@ prepare() {
 
   # Fix build with older oneapi (we really should upgrade soon!)
   git revert -n 49414a72f607ccd15f8b71b81edc9aff040d581e
+  # Adding mia ssing include of the "state_util.h" header
+  git cherry-pick -n d0938c1cfba8868e58af27f8a4e16b3594d41765
+  # Headers removed in ROCm 7.0
+  sed -i -E '/hiprt.*impl.*(Geometry|Scene)\.h/d' intern/cycles/kernel/CMakeLists.txt
+  sed -i -E '/hiprt.*impl.*(Geometry|Scene)\.h/d' tools/check_source/check_cmake_consistency_config.py
 }
 
 _get_pyver() {
@@ -154,8 +159,6 @@ build() {
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_PREFIX=/usr
     -D WITH_LINKER_MOLD=ON
-    -D CUDA_HOST_COMPILER="$NVCC_CCBIN"
-    -D CYCLES_CUDA_BINARIES_ARCH="$cuda_archs"
     -D HIP_ROOT_DIR=/opt/rocm
     -D HIPRT_INCLUDE_DIR=/opt/rocm/include
     -D HIPRT_COMPILER_PARALLEL_JOBS=8
@@ -169,6 +172,11 @@ build() {
     -D WITH_CYCLES_OSL=ON
     -D WITH_INSTALL_PORTABLE=OFF
     -D WITH_PYTHON_INSTALL=OFF
+    -D WITH_CYCLES_EMBREE=OFF
+    -D WITH_CYCLES_DEVICE_CUDA=OFF
+    -D WITH_CYCLES_DEVICE_OPTIX=OFF
+    -D WITH_CYCLES_CUDA_BINARIES=OFF
+    -D WITH_CUDA_DYNLOAD=OFF
     -G Ninja
     -S "$pkgname"
     -W no-dev
@@ -190,7 +198,11 @@ package() {
 
   # Move OneAPI AOT lib to proper place
   mkdir "${pkgdir}"/usr/lib/
-  mv "${pkgdir}"/usr/share/blender/lib/libcycles_kernel_oneapi_aot.so "${pkgdir}"/usr/lib/
+  # mv "${pkgdir}"/usr/share/blender/lib/libcycles_kernel_oneapi_aot.so "${pkgdir}"/usr/lib/
 
   install -vDm 644 doc/license/{BSD-{2,3}-Clause,MIT,Zlib}-license.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+depends=($(printf "%s\n" "${depends[@]}" | grep -Ev '^(embree|intel-oneapi-compiler-dpcpp-cpp-runtime-libs|intel-oneapi-compiler-shared-runtime-libs|openimagedenoise|usd)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|intel-oneapi-compiler-shared-runtime|intel-oneapi-dpcpp-cpp|intel-compute-runtime)$'))
+makedepends+=(vulkan-headers)
