diff --git a/PKGBUILD b/PKGBUILD
index 71a9b37..9e81342 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -140,8 +140,10 @@ sha512sums=('e113ddf9deb48ff1d53ddd7137677f6fd661b8315363fd55189a5b4fc44fe5a0f1c
 prepare() {
   cd "$pkgname"
 
+  export GIT_TERMINAL_PROMPT=0
   git lfs install --local
   git remote add network-origin https://projects.blender.org/blender/blender
+  git config --local lfs.https://projects.blender.org/blender/blender.git/info/lfs.access none
   git lfs fetch network-origin
   git lfs checkout
 
@@ -174,8 +176,11 @@ build() {
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_PREFIX=/usr
     -D WITH_LINKER_MOLD=ON
-    -D CUDA_HOST_COMPILER="$NVCC_CCBIN"
-    -D CYCLES_CUDA_BINARIES_ARCH="$cuda_archs"
+    -D WITH_CYCLES_EMBREE=OFF
+    -D WITH_CYCLES_DEVICE_CUDA=OFF
+    -D WITH_CYCLES_DEVICE_OPTIX=OFF
+    -D WITH_CYCLES_CUDA_BINARIES=OFF
+    -D WITH_CUDA_DYNLOAD=OFF
     -D HIP_ROOT_DIR=/opt/rocm
     -D HIPRT_INCLUDE_DIR=/opt/rocm/include
     -D HIPRT_COMPILER_PARALLEL_JOBS=8
@@ -232,3 +237,7 @@ package() {
 
   install -vDm 644 doc/license/{BSD-{2,3}-Clause,MIT,Zlib}-license.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+depends=($(printf "%s\n" "${depends[@]}" | grep -Ev '^(embree|intel-oneapi-compiler-dpcpp-cpp-runtime-libs|intel-oneapi-compiler-shared-runtime-libs|openimagedenoise|usd)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|intel-oneapi-compiler-shared-runtime|intel-oneapi-dpcpp-cpp|intel-compute-runtime)$'))
+makedepends+=(vulkan-headers)
