--- a/PKGBUILD
+++ b/PKGBUILD
@@ -30,7 +30,6 @@ depends=(
   'bash'
   'boost-libs'
   'draco'  # seems to use static lib and should probably only be makedepends
-  'embree'
   'expat'
   'ffmpeg'
   'fftw'
@@ -41,8 +40,6 @@ depends=(
   'gmp'
   'hicolor-icon-theme'
   'imath'
-  'intel-oneapi-compiler-dpcpp-cpp-runtime-libs'
-  'intel-oneapi-compiler-shared-runtime-libs'
   'jack'  # dlopen'ed
   'jemalloc'
   'level-zero-loader'  # dlopen'ed
@@ -67,7 +64,6 @@ depends=(
   'openal'
   'opencolorio'
   'openexr'
-  'openimagedenoise'
   'openimageio'
   'openjpeg2'
   'openpgl'
@@ -83,7 +79,6 @@ depends=(
   'python-requests'
   'sdl2'  # dlopen'ed
   'shared-mime-info'
-  'usd'
   'xdg-utils'
   'yaml-cpp'
   'zlib'
@@ -92,14 +87,8 @@ depends=(
 makedepends=(
   'boost'
   'cmake'
-  'cuda'
   'git'
   'git-lfs'
-  'hip-runtime-amd'
-  'hiprt'
-  'intel-oneapi-compiler-shared-runtime'
-  'intel-oneapi-dpcpp-cpp'
-  'intel-compute-runtime'
   'level-zero-headers'
   'libdecor'
   'mold'
@@ -109,11 +98,6 @@ makedepends=(
   'subversion'
   'wayland-protocols'
 )
-optdepends=('cuda: Cycles renderer CUDA support'
-            'intel-compute-runtime: Cycles renderer Intel OneAPI support'
-            'hip-runtime-amd: Cycles renderer AMD ROCm support'
-            'hiprt: Ray tracing AMD ROCm support'
-            'libdecor: wayland support')
 # We're using !lto here as otherwise we get stuff such as FS#77557
 options=('!lto')
 source=("git+https://projects.blender.org/blender/blender.git#tag=v$pkgver"
@@ -139,6 +123,9 @@ prepare() {
   git -C release/datafiles/assets lfs checkout
 
   patch -p1 -i "$srcdir"/hiprt-lib-path.patch
+
+  # Support loong64
+  patch -p1 -i "$srcdir"/add-loong64-support.patch
 }
 
 _get_pyver() {
@@ -159,9 +146,6 @@ build() {
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_PREFIX=/usr
     -D WITH_LINKER_MOLD=ON
-    -D CUDA_HOST_COMPILER="$NVCC_CCBIN"
-    -D HIP_ROOT_DIR=/opt/rocm
-    -D HIPRT_INCLUDE_DIR=/opt/rocm/include
     -D OCLOC_INSTALL_DIR=/usr
     -D OPTIX_ROOT_DIR="$srcdir"
     -D PYTHON_VERSION="$(_get_pyver)"
@@ -172,6 +156,13 @@ build() {
     -D WITH_CYCLES_OSL=ON
     -D WITH_INSTALL_PORTABLE=OFF
     -D WITH_PYTHON_INSTALL=OFF
+    -D WITH_CYCLES_EMBREE=OFF
+    -D WITH_CYCLES_DEVICE_CUDA=OFF
+    -D WITH_CYCLES_DEVICE_OPTIX=OFF
+    -D WITH_CYCLES_CUDA_BINARIES=OFF
+    -D WITH_CUDA_DYNLOAD=OFF
+    -D WITH_CYCLES_DEVICE_HIP=OFF
+    -D WITH_CYCLES_HIP_BINARIES=OFF
     -G Ninja
     -S "$pkgname"
     -W no-dev
@@ -193,7 +184,11 @@ package() {
 
   # Move OneAPI AOT lib to proper place
   mkdir "${pkgdir}"/usr/lib/
-  mv "${pkgdir}"/usr/share/blender/lib/libcycles_kernel_oneapi_aot.so "${pkgdir}"/usr/lib/
+  # mv "${pkgdir}"/usr/share/blender/lib/libcycles_kernel_oneapi_aot.so "${pkgdir}"/usr/lib/
 
   install -vDm 644 doc/license/{BSD-{2,3}-Clause,MIT,Zlib}-license.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+source+=("add-loong64-support.patch::https://projects.blender.org/blender/blender/commit/d18a0cf277e3d7403eb18718b9cb7a6ec036819f.patch")
+sha512sums+=('4bcbe578638dec0d7d3808835d7191f890c7060adbd6b4006e1e1fc4eee70ae403527419c6583913c43254d5ca8dbf6427ba9b445d45ac5f4af7e5a26b88c98b')
+makedepends+=(vulkan-headers)
