--- a/PKGBUILD
+++ b/PKGBUILD
@@ -30,7 +30,6 @@ depends=(
   'bash'
   'boost-libs'
   'draco'  # seems to use static lib and should probably only be makedepends
-  'embree'
   'expat'
   'ffmpeg'
   'fftw'
@@ -41,8 +40,6 @@ depends=(
   'gmp'
   'hicolor-icon-theme'
   'imath'
-  'intel-oneapi-compiler-dpcpp-cpp-runtime-libs'
-  'intel-oneapi-compiler-shared-runtime-libs'
   'jack'  # dlopen'ed
   'jemalloc'
   'level-zero-loader'  # dlopen'ed
@@ -67,10 +64,8 @@ depends=(
   'openal'
   'opencolorio'
   'openexr'
-  'openimagedenoise'
   'openimageio'
   'openjpeg2'
-  'openpgl'
   'openshadinglanguage'
   'opensubdiv'
   'openvdb'
@@ -83,7 +78,6 @@ depends=(
   'python-requests'
   'sdl2'  # dlopen'ed
   'shared-mime-info'
-  'usd'
   'xdg-utils'
   'yaml-cpp'
   'zlib'
@@ -92,13 +86,8 @@ depends=(
 makedepends=(
   'boost'
   'cmake'
-  'cuda'
   'git'
   'git-lfs'
-  'hip-runtime-amd'
-  'intel-oneapi-compiler-shared-runtime'
-  'intel-oneapi-dpcpp-cpp'
-  'intel-compute-runtime'
   'level-zero-headers'
   'libdecor'
   'mold'
@@ -108,9 +97,6 @@ makedepends=(
   'subversion'
   'wayland-protocols'
 )
-optdepends=('cuda: Cycles renderer CUDA support'
-            'intel-compute-runtime: Cycles renderer Intel OneAPI support'
-            'libdecor: wayland support')
 # We're using !lto here as otherwise we get stuff such as FS#77557
 options=('!lto')
 source=("git+https://projects.blender.org/blender/blender.git#tag=v$pkgver"
@@ -167,14 +153,10 @@ build() {
   local cmake_options=(
     -B build
     -C "$pkgname/build_files/cmake/config/blender_release.cmake"
-    -C "$srcdir/precache"
+    # -C "$srcdir/precache"
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_PREFIX=/usr
     -D WITH_LINKER_MOLD=ON
-    -D CUDA_HOST_COMPILER="$NVCC_CCBIN"
-    -D HIP_ROOT_DIR=/opt/rocm
-    # Skip targets gfx115{0,1} as they not supported by ROCm 6.0.2
-    -D CYCLES_HIP_BINARIES_ARCH="gfx900;gfx90c;gfx902;gfx1010;gfx1011;gfx1012;gfx1030;gfx1031;gfx1032;gfx1034;gfx1035;gfx1036;gfx1100;gfx1101;gfx1102;gfx1103"
     -D OCLOC_INSTALL_DIR=/usr
     -D OPTIX_ROOT_DIR="$srcdir"
     -D PYTHON_VERSION="$(_get_pyver)"
@@ -185,6 +167,13 @@ build() {
     -D WITH_CYCLES_OSL=ON
     -D WITH_INSTALL_PORTABLE=OFF
     -D WITH_PYTHON_INSTALL=OFF
+    -D WITH_CYCLES_EMBREE=OFF
+    -D WITH_CYCLES_DEVICE_CUDA=OFF
+    -D WITH_CYCLES_DEVICE_OPTIX=OFF
+    -D WITH_CYCLES_CUDA_BINARIES=OFF
+    -D WITH_CUDA_DYNLOAD=OFF
+    -D WITH_CYCLES_DEVICE_HIP=OFF
+    -D WITH_CYCLES_HIP_BINARIES=OFF
     -G Ninja
     -S "$pkgname"
     -W no-dev
