diff --git a/PKGBUILD b/PKGBUILD
index bab1baf..5994774 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -180,8 +180,11 @@ build() {
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_PREFIX=/usr
     -D WITH_LINKER_MOLD=ON
-    -D CUDA_HOST_COMPILER="$NVCC_CCBIN"
-    -D CYCLES_CUDA_BINARIES_ARCH="$cuda_archs"
+    -D WITH_CYCLES_EMBREE=OFF
+    -D WITH_CYCLES_DEVICE_CUDA=OFF
+    -D WITH_CYCLES_DEVICE_OPTIX=OFF
+    -D WITH_CYCLES_CUDA_BINARIES=OFF
+    -D WITH_CUDA_DYNLOAD=OFF
     -D HIP_ROOT_DIR=/opt/rocm
     -D HIPRT_INCLUDE_DIR=/opt/rocm/include
     -D HIPRT_COMPILER_PARALLEL_JOBS=8
@@ -238,3 +241,7 @@ package() {
 
   install -vDm 644 doc/license/{BSD-{2,3}-Clause,MIT,Zlib}-license.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+depends=($(printf "%s\n" "${depends[@]}" | grep -Ev '^(embree|intel-oneapi-compiler-dpcpp-cpp-runtime-libs|intel-oneapi-compiler-shared-runtime-libs|openimagedenoise|usd)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|intel-oneapi-compiler-shared-runtime|intel-oneapi-dpcpp-cpp|intel-compute-runtime)$'))
+makedepends+=(vulkan-headers)
