--- a/PKGBUILD
+++ b/PKGBUILD
@@ -67,7 +67,7 @@ makedepends=(
   gl2ps                   # libvtkRenderingGL2PSOpenGL2.so, libvtkIOExportGL2PS.so (2 direct libs, 4 total libs)
   proj                    # libvtkGeovisCore.so, libvtkIOCesium3DTiles.so (2 direct libs, 4 total libs)
   alembic                 # libvtkIOAlembic.so (1 direct lib, 2 total libs)
-  anari-sdk               # libvtkRenderingAnari.so (1 direct lib, 2 total libs)
+#  anari-sdk               # libvtkRenderingAnari.so (1 direct lib, 2 total libs)
   ffmpeg                  # libvtkIOFFMPEG.so (1 direct lib, 2 total libs)
   imath                   # libvtkIOOpenVDB.so (1 direct lib, 2 total libs)
   libharu                 # libvtkIOExportPDF.so (1 direct lib, 2 total libs)
@@ -76,11 +76,11 @@ makedepends=(
   libtheora               # libvtkIOOggTheora.so (1 direct lib, 2 total libs)
   mariadb-libs            # libvtkIOMySQL.so (1 direct lib, 2 total libs)
   opencascade             # libvtkIOOCCT.so (1 direct lib, 2 total libs)
-  openimagedenoise        # libvtkRenderingRayTracing.so (1 direct lib, 2 total libs)
+#  openimagedenoise        # libvtkRenderingRayTracing.so (1 direct lib, 2 total libs)
   openvdb                 # libvtkIOOpenVDB.so (1 direct lib, 2 total libs)
   openvr                  # libvtkRenderingOpenVR.so (1 direct lib, 2 total libs)
   openxr                  # libvtkRenderingOpenXR.so (1 direct lib, 2 total libs)
-  ospray                  # libvtkRenderingRayTracing.so (1 direct lib, 2 total libs)
+#  ospray                  # libvtkRenderingRayTracing.so (1 direct lib, 2 total libs)
   pdal                    # libvtkIOPDAL.so (1 direct lib, 2 total libs)
   postgresql-libs         # libvtkIOPostgreSQL.so (1 direct lib, 2 total libs)
   unixodbc                # libvtkIOODBC.so (1 direct lib, 2 total libs)
@@ -189,7 +189,9 @@ build() {
     -DVTK_VERSIONED_INSTALL=OFF \
     -DVTK_WRAP_JAVA=ON \
     -DVTK_WRAP_PYTHON=ON \
-    -DVTKOSPRAY_ENABLE_DENOISER=ON \
+    -DVTKOSPRAY_ENABLE_DENOISER=OFF \
+    -DVTK_ENABLE_OSPRAY=OFF \
+    -DVTK_MODULE_ENABLE_VTK_RenderingAnari=NO \
     -DVTK_USE_EXTERNAL=ON \
     -DVTK_MODULE_USE_EXTERNAL_VTK_exprtk=OFF \
     -DVTK_MODULE_USE_EXTERNAL_VTK_ioss=OFF \
@@ -214,8 +216,8 @@ package() {
   install -dv "${pkgdir}"/usr/share/java/vtk
   mv -v "${pkgdir}"/usr/lib/java/vtk.jar "${pkgdir}"/usr/share/java/vtk
   # …and the libs to the proper place
-  mv "${pkgdir}"/usr/lib/java/vtk-Linux-${CARCH}/*.so "${pkgdir}"/usr/lib/
-  rmdir "${pkgdir}"/usr/lib/java/{vtk-Linux-${CARCH}/,}
+  mv "${pkgdir}"/usr/lib/java/vtk-Linux-`uname -m`/*.so "${pkgdir}"/usr/lib/
+  rmdir "${pkgdir}"/usr/lib/java/{vtk-Linux-`uname -m`/,}
 
   # byte-compile python modules since the CMake build does not do it
   local site_packages=$(python -c "import site; print(site.getsitepackages()[0])")
