diff --git a/PKGBUILD b/PKGBUILD
index 2e35761..d1d2bf3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,7 +20,8 @@ makedepends=(cmake
 groups=(qt6)
 _pkgfn=${pkgname/6-/}
 source=(git+https://code.qt.io/qt/$_pkgfn#tag=v$pkgver)
-sha256sums=('576a5e14bf90760dcf53f541112113329a957eb6d7010f706be894260b087701')
+sha256sums=('576a5e14bf90760dcf53f541112113329a957eb6d7010f706be894260b087701'
+            '391ef157b440bff095db5a3876f293bd7c32149bada9b49b884a7e8983005e8e')
 
 build() {
   cmake -B build -S $_pkgfn -G Ninja \
@@ -34,3 +35,9 @@ package() {
   install -d "$pkgdir"/usr/share/licenses
   ln -s /usr/share/licenses/qt6-base "$pkgdir"/usr/share/licenses/$pkgname
 }
+
+prepare() {
+  patch -d $_pkgfn -p1 -i "$srcdir/0001-physx-loong64.patch"
+}
+
+source+=("0001-physx-loong64.patch")
