diff --git a/PKGBUILD b/PKGBUILD
index 271b2ec..5c935f5 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,6 +16,11 @@ source=(https://xorg.freedesktop.org/releases/individual/lib/${pkgname}-${pkgver
 sha512sums=('3ea293cd308a4faeb600939df2c98148154cf1faf8e435b9f33df46c33acfeb32572cb0c1e148ee9f6f39f9e8a49fbda627edcac03da2000d89e22941849a04d')
 #validpgpkeys=('') # Maarten Lankhorst <maarten.lankhorst@linux.intel.com>
 
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -p1 -i "${srcdir}/LoongArch-LSX-and-LASX-SIMD-implementation.patch"
+}
+
 build() {
   arch-meson $pkgname-$pkgver build \
     -D loongson-mmi=disabled \
@@ -25,7 +30,11 @@ build() {
     -D a64-neon=disabled \
     -D rvv=disabled \
     -D mips-dspr2=disabled \
-    -D gtk=disabled
+    -D gtk=disabled \
+    -D lsx=enabled \
+    -D mmx=disabled \
+    -D ssse3=disabled \
+    -D sse2=disabled
   ninja -C build
 }
 
@@ -37,3 +46,6 @@ package() {
   DESTDIR="$pkgdir" meson install -C build
   install -Dt "$pkgdir/usr/share/licenses/$pkgname" -m644 $pkgname-$pkgver/COPYING
 }
+
+source+=('LoongArch-LSX-and-LASX-SIMD-implementation.patch')
+sha512sums+=('41f127f43370d0cc3f2d6754845a871bacde7d2822c8422b15aeaf74df87eda7041c1e17e6f8271113ffa66166a5195abbaf4a73e14a438356e6ac20ff442f5e')
