diff --git a/PKGBUILD b/PKGBUILD
index e651ca5..12ec09b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,6 +16,11 @@ source=(https://xorg.freedesktop.org/releases/individual/lib/${pkgname}-${pkgver
 sha512sums=('b24907f9a0bf6a9aa3278d31986c562c5f2d68f8c178021497c8648aca9f91e4c6f1ecfacf93eb01ba4f03ce4cfc5970fc40bb0fe6724bac9210fc57da09cba4')
 #validpgpkeys=('') # Maarten Lankhorst <maarten.lankhorst@linux.intel.com>
 
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -p1 -i "${srcdir}/LoongArch-LSX-and-LASX-SIMD-implementation.patch"
+}
+
 build() {
   arch-meson $pkgname-$pkgver build \
     -D loongson-mmi=disabled \
@@ -25,7 +30,11 @@ build() {
     -D a64-neon=disabled \
     -D rvv=disabled \
     -D mips-dspr2=disabled \
-    -D gtk=disabled
+    -D gtk=disabled \
+    -D lsx=enabled \
+    -D mmx=disabled \
+    -D ssse3=disabled \
+    -D sse2=disabled
   ninja -C build
 }
 
@@ -37,3 +46,6 @@ package() {
   DESTDIR="$pkgdir" meson install -C build
   install -Dt "$pkgdir/usr/share/licenses/$pkgname" -m644 $pkgname-$pkgver/COPYING
 }
+
+source+=('LoongArch-LSX-and-LASX-SIMD-implementation.patch')
+sha512sums+=('1afd3cc6209f38f807d0e0167790e3c6e6f7f58d2b64753ed7a25d221cc7c880a44ba5c41bd3eb93ca0f6113cae17c2fb631fc51277632c6c72fe6503007ed36')
