--- a/PKGBUILD
+++ b/PKGBUILD
@@ -54,12 +54,12 @@ build() {
   export CARGO_PROFILE_RELEASE_LTO=true CARGO_PROFILE_RELEASE_CODEGEN_UNITS=1
   # Dynamically link system libraries
   export ZSTD_SYS_USE_PKG_CONFIG=1
-  cargo build --release --frozen --features all,dist-server
+  cargo build --release --frozen --features all
 }
 
 package() {
   cd sccache
-  install -Dm 755 target/release/sccache{,-dist} -t "${pkgdir}/usr/bin/"
+  install -Dm 755 target/release/sccache -t "${pkgdir}/usr/bin/"
   install -dm 755 "${pkgdir}/usr/share/doc/${pkgname}"
   install -m 644 README.md docs/* -t "${pkgdir}/usr/share/doc/${pkgname}/"
   install -dm 755 "${pkgdir}/usr/lib/sccache/bin"
