--- a/PKGBUILD
+++ b/PKGBUILD
@@ -15,11 +15,11 @@ pkgname=(
 )
 _majorver=11
 _minorver=0
-_securityver=32.1
-_updatever=1
+_securityver=32
+_updatever=9
 pkgver=${_majorver}.${_minorver}.${_securityver}.u${_updatever}
 pkgrel=1
-_git_tag=jdk-${_majorver}.${_minorver}.${_securityver}+${_updatever}
+_git_tag=jdk-${_majorver}.${_minorver}.${_securityver}+${_updatever}-ls-ga
 arch=('x86_64')
 url='https://openjdk.java.net/'
 license=('LicenseRef-Java')
@@ -67,6 +67,7 @@ sha256sums=('d9a11caf637121ac7a034d882443da4119ae32c4d5b192a2350d51167ad98008'
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   aarch64)_JARCH='aarch64';;
+  loong64) _JARCH='loongarch64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -434,5 +435,7 @@ package_openjdk11-doc() {
   install -dm 755 "${pkgdir}/usr/share/licenses"
   ln -s ${pkgbase} "${pkgdir}/usr/share/licenses/${pkgname}"
 }
+source[0]="https://github.com/loongson/jdk${_majorver}u/archive/${_git_tag}.tar.gz"
+sha256sums[0]="40f8cf67f944c27bf21a968ac32b56fd14936386f42c7c4e47926ba8ab271167"
 
 # vim: ts=2 sw=2 et:
