diff --git a/PKGBUILD b/PKGBUILD
index dd6a5b8..36a1fd4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -38,6 +38,11 @@ _pick() {
   done
 }
 
+prepare() {
+  cd AIDA-X-$pkgver
+  patch -p1 -d "modules/dpf" -i "${srcdir}/add-loongarch-support.patch"
+}
+
 build() {
   local cmake_options=(
     -B build
@@ -174,3 +179,7 @@ package_aida-x-vst3() {
   # mv -v $pkgname/* "$pkgdir"
   install -vDm 755 build/bin/$_name.vst3/Contents/$CARCH-linux/*.so -t "$pkgdir/usr/lib/vst3/$_name.vst3/Contents/$CARCH-linux/"
 }
+
+source+=("add-loongarch-support.patch::https://github.com/DISTRHO/DPF/commit/9c11acb038e4c8494519a4e271e046b49bd2eaea.patch")
+sha512sums+=('d94c7a95c0602755e79624511b1862cdb4985986522990274f48373cb261210a2b558030692a290a755ee153799874bbb7cf83edd3754616cb72121d1d4e5ec8')
+b2sums+=('c214508c3d2b759a490cce4b8ce26b66a173425d52fd1ff7b9323f877cff2614acdd46a8be352a6f36c18db1f7bafee6e4466f15cc1134be2ee679379db7ce9c')
