--- a/PKGBUILD
+++ b/PKGBUILD
@@ -38,6 +38,11 @@ _pick() {
   done
 }
 
+prepare() {
+  cd AIDA-X-$pkgver
+  patch -p1 -d "modules/dpf" -i "${srcdir}/add-loongarch-support.patch"
+}
+
 build() {
   local cmake_options=(
     -B build
@@ -178,5 +183,9 @@ package_aida-x-vst3() {
   )
 
   # mv -v $pkgname/* "$pkgdir"
-  install -vDm 755 build/bin/$_name.vst3/Contents/$CARCH-linux/*.so -t "$pkgdir/usr/lib/vst3/$_name.vst3/Contents/$CARCH-linux/"
+  install -vDm 755 build/bin/$_name.vst3/Contents/$(uname -m)-linux/*.so -t "$pkgdir/usr/lib/vst3/$_name.vst3/Contents/$(uname -m)-linux/"
 }
+
+source+=("add-loongarch-support.patch::https://github.com/DISTRHO/DPF/commit/8ee7f96f4ebcccf28d4df7c6226aace5f966722e.diff")
+sha512sums+=('d968bdf5bf50a2784bd0b2bb633630d1a5416834cfa46d2e2b28597c8232e9381351e47994b6095d72abc9fefd3a46bc2ff44fbbef26757f28106e9b4be738d8')
+b2sums+=('d423812a4bd9f716997b9f3f5f10002ad115126633d8164f50873e920828bfab924c2416da15a674bfb96116bd7f9193d00791657680689655c379c49d93db3e')
