diff --git a/PKGBUILD b/PKGBUILD
index dd6a5b8..104de93 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -4,7 +4,7 @@ pkgbase=aida-x
 _name="${pkgbase^^}"
 pkgname=(
   aida-x
-  aida-x-{clap,lv2,standalone,vst,vst3}
+  aida-x-{clap,lv2,standalone,vst}
 )
 pkgver=1.1.0
 pkgrel=1
@@ -38,6 +38,11 @@ _pick() {
   done
 }
 
+prepare() {
+    cd AIDA-X-$pkgver
+    patch -p1 -i ../disable-vst3.patch
+}
+
 build() {
   local cmake_options=(
     -B build
@@ -174,3 +179,7 @@ package_aida-x-vst3() {
   # mv -v $pkgname/* "$pkgdir"
   install -vDm 755 build/bin/$_name.vst3/Contents/$CARCH-linux/*.so -t "$pkgdir/usr/lib/vst3/$_name.vst3/Contents/$CARCH-linux/"
 }
+
+source+=("disable-vst3.patch")
+sha512sums+=('3673abb1162be4425376da68f57d22a6d8db6cc77c64fc7f576b610864b49ccba8f3bc0c4421c8e0341d5508e2c8a72f7eab2cb7a2678658e634a464dae15acd')
+b2sums+=('1d807b77a7f292cc92f7caa572808b4e9f2e84c4bd2d067beab6a90462b341c128dd90a73792f4d201f7f91ec681078fa4ba7ad2936b6497e6f7a651ff590d5b')

