--- a/PKGBUILD
+++ b/PKGBUILD
@@ -69,6 +69,8 @@ prepare() {
 
   # Remove CMake find module for zstd; breaks if out of sync with upstream zstd
   rm cmake/modules/Findzstd.cmake
+
+  patch -p2 -i "${srcdir}/LoongArch-Allow-f16-codegen-with-expansion-to-libcalls.patch"
 }
 
 build() {
@@ -153,4 +155,7 @@ package_llvm18-libs() {
     "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
 
+source+=("LoongArch-Allow-f16-codegen-with-expansion-to-libcalls.patch::https://github.com/llvm/llvm-project/pull/94456/commits/bbb1854a5a76a8199c50d60a2ae5809163f9d1c5.patch")
+sha256sums+=('c485df407ab98dd299c80aa2b26f59c18c728b326a8ae32348bec57418515dcb')
+
 # vim:set ts=2 sw=2 et:
