diff --git a/PKGBUILD b/PKGBUILD
index 67edd7b..1ae2815 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,6 +20,8 @@ source=(
 sha256sums=('bf1ad7f7476bb8f3076b9b404b0152259e55763a39fa313a3cc19a89846163a4')
 
 build() {
+  patch -Np1 -d "$pkgname-$pkgver" -i "${srcdir}/correct-narrowing-conversion-in-lsx-string-parsing.patch"
+  python "$pkgname-$pkgver/singleheader/amalgamate.py"
   cmake -B build -S "$pkgname-$pkgver" \
     -DBUILD_SHARED_LIBS='On' \
     -DCMAKE_BUILD_TYPE='None' \
@@ -51,4 +53,10 @@ package() {
   DESTDIR="$pkgdir/" cmake --install build
 }
 
+# Backport https://github.com/simdjson/simdjson/pull/2481 (merged)
+source+=("correct-narrowing-conversion-in-lsx-string-parsing.patch")
+sha256sums+=('c1c98b952966a1f93ae63b80cacac24baaec13526bee754c2d6bf462e96970c8')
+# To run amalgamate.py after applying "correct-narrowing-conversion-in-lsx-string-parsing.patch"
+makedepends+=(python)
+
 # vim: tabstop=2 shiftwidth=2 expandtab:
