--- a/PKGBUILD
+++ b/PKGBUILD
@@ -52,3 +52,13 @@ package() {
 
     install -Dm644 "$_dir_name/LICENSE.md" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+prepare() {
+  patch -Np1 -d "$_dir_name" -i "$srcdir/0002-support-arm64-loongarch64-riscv64.patch"
+  patch -Np1 -d "$_dir_name" -i "$srcdir/0004-replace-llvm-mc.patch"
+}
+
+source+=("0002-support-arm64-loongarch64-riscv64.patch"
+         "0004-replace-llvm-mc.patch")
+sha256sums+=('d43329acf9f632b3aa8749086197552e6ea0789392f54262c8979802a4993ed5'
+             '932c06c6d7630d389149eccbdde84d228c2ab2bccf0ffc7f0f63de2dd3bc0a26')
