--- a/PKGBUILD
+++ b/PKGBUILD
@@ -49,3 +49,13 @@ package() {
     echo '/opt/rocm/lib/libamdocl64.so' > 'amdocl64.icd'
     install -Dm644 'amdocl64.icd' "$pkgdir/etc/OpenCL/vendors/amdocl64.icd"
 }
+
+prepare() {
+  patch -Np1 -d "$_dir_name" -i "$srcdir/0002-support-arm64-loongarch64-riscv64.patch"
+  patch -Np1 -d "$_dir_name" -i "$srcdir/0004-replace-llvm-mc.patch"
+}
+
+source+=("0002-support-arm64-loongarch64-riscv64.patch"
+         "0004-replace-llvm-mc.patch")
+sha256sums+=('7bfa3814a41f3d6195664fe844e06cc208c9ccc7dea223e973eac88667aa1968'
+             '1779ab6d3128135fa3eb1fe8f24836ec019ccaf5f17bb91bf84824a671d83c30')
