diff --git a/PKGBUILD b/PKGBUILD
index 9e050a0..3aa1146 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,6 +17,7 @@ b2sums=('706a2a58b58c2051142a984de31cfce46b25d4bf3c2152a5644d65a3130506eff3d6f47
 
 prepare() {
   cd $pkgname
+  patch -p1 -i "${srcdir}/update-carge.patch"
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
@@ -36,3 +37,6 @@ package() {
   install -Dm644 -t "$pkgdir"/usr/share/man/man1 doc/$pkgname.1
   install -Dm644 -t "$pkgdir"/usr/share/licenses/$pkgname LICENSE-MIT
 }
+
+source+=("update-carge.patch::https://github.com/sharkdp/diskus/commit/e77f2083728d3bf24eb511f9c276cfdf81990ab4.patch")
+b2sums+=('f92f513137e2d5d8ae05e6e847fa40f9d3782863aa06e089c6192f0006862c274a9c7603b130e7a0f034d4506fa82427eb0a39f8232030dcf7f1c5ae7a3a51fe')
