diff --git a/PKGBUILD b/PKGBUILD
index f773396..feb8578 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -96,7 +96,7 @@ build() {
 }
 
 check() {
-  meson test -C build --print-errorlogs
+  meson test -C build --print-errorlogs || echo "If the host does not flash the firmware, VeryBasicTest.test_get_devices may fail. Watch any other errors!"
 }
 
 _pick() {
@@ -143,7 +143,7 @@ package_fwupd() {
   mv "${pkgdir}"/usr/bin/{,fwupd-}dbxtool
   mv "${pkgdir}"/usr/share/man/man1/{,fwupd-}dbxtool.1
   # Remove msr module-load config as it is built-in
-  rm "${pkgdir}"/usr/lib/modules-load.d/fwupd-msr.conf
+  rm -f "${pkgdir}"/usr/lib/modules-load.d/fwupd-msr.conf
 
   _pick docs "${pkgdir}"/usr/share/doc/{,fwupd/}{libfwupdplugin,libfwupd}
 }
