diff --git a/PKGBUILD b/PKGBUILD
index fba9569..0057ae3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -77,6 +77,9 @@ prepare() {
   echo "Setting config..."
   cp ../config .config
   make olddefconfig
+  if [ $CARCH == loong64 ]; then
+    patch .config -i ../config.loong64.diff
+  fi
   diff -u ../config .config || :
 
   make -s kernelrelease > version
@@ -141,20 +144,28 @@ _package-headers() {
   install -Dt "$builddir" -m644 .config Makefile Module.symvers System.map \
     localversion.* version vmlinux tools/bpf/bpftool/vmlinux.h
   install -Dt "$builddir/kernel" -m644 kernel/Makefile
-  install -Dt "$builddir/arch/x86" -m644 arch/x86/Makefile
+  if [ $CARCH == x86_64 ]; then
+    install -Dt "$builddir/arch/x86" -m644 arch/x86/Makefile
+  elif [ $CARCH == loong64 ]; then
+    install -Dt "$builddir/arch/loongarch" -m644 arch/loongarch/Makefile
+  fi
   cp -t "$builddir" -a scripts
 
   # required when STACK_VALIDATION is enabled
-  install -Dt "$builddir/tools/objtool" tools/objtool/objtool
+  #install -Dt "$builddir/tools/objtool" tools/objtool/objtool
 
   # required when DEBUG_INFO_BTF_MODULES is enabled
   install -Dt "$builddir/tools/bpf/resolve_btfids" tools/bpf/resolve_btfids/resolve_btfids
 
   echo "Installing headers..."
   cp -t "$builddir" -a include
-  cp -t "$builddir/arch/x86" -a arch/x86/include
-  install -Dt "$builddir/arch/x86/kernel" -m644 arch/x86/kernel/asm-offsets.s
-
+  if [ $CARCH == x86_64 ]; then
+    cp -t "$builddir/arch/x86" -a arch/x86/include
+    install -Dt "$builddir/arch/x86/kernel" -m644 arch/x86/kernel/asm-offsets.s
+  elif [ $CARCH == loong64 ]; then
+    cp -t "$builddir/arch/loongarch" -a arch/loongarch/include
+    install -Dt "$builddir/arch/loongarch/kernel" -m644 arch/loongarch/kernel/asm-offsets.s
+  fi
   install -Dt "$builddir/drivers/md" -m644 drivers/md/*.h
   install -Dt "$builddir/net/mac80211" -m644 net/mac80211/*.h
 
@@ -175,7 +186,11 @@ _package-headers() {
   echo "Removing unneeded architectures..."
   local arch
   for arch in "$builddir"/arch/*/; do
-    [[ $arch = */x86/ ]] && continue
+    if [ $CARCH == x86_64 ]; then
+      [[ $arch = */x86/ ]] && continue
+    elif [ $CARCH == loong64 ]; then
+      [[ $arch = */loongarch/ ]] && continue
+    fi
     echo "Removing $(basename "$arch")"
     rm -r "$arch"
   done
@@ -243,4 +258,8 @@ for _p in "${pkgname[@]}"; do
   }"
 done
 
+source+=('config.loong64.diff')
+sha256sums+=('3a9608c93058f25bcda6a5d27f6be89416bb18d86c68b0f1705995d73dcf5d2b')
+b2sums+=('2e86b04c62d4d38efcaf460dc5e4b7e91db2758d42c4dfcd05c2ffed3544b0ef98b3bd2b0d35b5cbee7031fbc2268cd098f9e5c279ab163d50e08664008d3971')
+
 # vim:set ts=8 sts=2 sw=2 et:
