--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,7 +11,7 @@ url="https://docs.xfce.org/xfce/thunar/archive"
 license=('GPL-2.0-only')
 groups=('xfce4-goodies')
 depends=('thunar' 'hicolor-icon-theme')
-makedepends=('git' 'xfce4-dev-tools')
+makedepends=('git' 'xfce4-dev-tools' 'intltool')
 optdepends=('file-roller' 'engrampa' 'ark' 'xarchiver')
 source=("git+https://gitlab.xfce.org/thunar-plugins/thunar-archive-plugin.git#tag=$pkgname-$pkgver")
 sha256sums=('147372fe3b114d8abb4549ae054094c4354247c4495c1927b9e6771f6f3bc1e1')
