--- a/PKGBUILD
+++ b/PKGBUILD
@@ -239,7 +239,7 @@ build() {
   local meson_options=(
     -D android-libbacktrace=disabled
     -D b_ndebug=true
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,iris,crocus,i915,zink,d3d12,asahi,freedreno
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,iris,zink,d3d12,asahi,freedreno
     -D gallium-extra-hud=true
     -D gallium-mediafoundation=disabled
     -D gallium-rusticl=true
@@ -251,7 +251,7 @@ build() {
     -D sysprof=true
     -D valgrind=enabled
     -D video-codecs=all
-    -D vulkan-drivers=amd,gfxstream,intel,intel_hasvk,nouveau,swrast,virtio,microsoft-experimental,asahi,freedreno
+    -D vulkan-drivers=amd,gfxstream,intel,nouveau,swrast,virtio,microsoft-experimental,asahi,freedreno
     -D vulkan-layers=device-select,intel-nullhw,overlay,screenshot,vram-report-limit
   )
 
@@ -340,8 +340,8 @@ package_mesa() {
     _pick vkgfxstr $icddir/gfxstream_vk_icd.*.json
     _pick vkgfxstr $libdir/libvulkan_gfxstream.so
 
-    _pick vkintel $icddir/intel{,_hasvk}_icd.*.json
-    _pick vkintel $libdir/libvulkan_intel{,_hasvk}.so
+    _pick vkintel $icddir/intel_icd.*.json
+    _pick vkintel $libdir/libvulkan_intel.so
 
     _pick vknvidia $icddir/nouveau_icd.*.json
     _pick vknvidia $libdir/libvulkan_nouveau.so
@@ -689,4 +689,14 @@ package_mesa-docs() {
   install -Dm644 mesa-$_pkgver/docs/license.rst -t "$pkgdir/usr/share/licenses/$pkgname"
 }
 
+source+=("enable-build-on-other-64bit-processors.patch::https://gitlab.freedesktop.org/mesa/mesa/-/commit/3d62be64f55fa8455f1f9f179490fe4ccf053df7.diff"
+         "anv-Use-mem_alignment-for-mmap-offset-alignment.patch::https://gitlab.freedesktop.org/mesa/mesa/-/commit/9142a019b40b5c7a5c3582f50b2f3f96cfad86e9.diff"
+         "llvm21-support.patch::https://gitlab.freedesktop.org/mesa/mesa/-/commit/cd129dbf8af2d16b1243f2ce287ff69c6a5dc557.patch")
+b2sums+=('4cf6c062c305be9766dee3a0aeff0b3395b62368cde67368f26dc7905a66a263d1acc7010b95f92d5269ac2f14de5864b46c1d3e63c9879594b704d6e3ec31c1'
+         '2cddb5e793dc9381afdc52b60021809214ca70eacefc9d05c0061bf535893b6f8286eed19e523ef33a3d44a14fd7c46c2c02d70e9350680c41f1ee0cb2218055'
+         'a2fe345a6f745d548ca3bc605c5d5bf038729bc82fffba3a7d763a62d2ecd401ad06ee86f6340c2dc13b73384b9e738336f7ac28ba9f45f1c0c6fd0c70100f18')
+sha256sums+=('74036abdb2d8ccb07e83e66f2da97993d3dc56dc22c0817405bec48a2c6fe7ed'
+             '29b683d66efb4bb843d2cda5ed518f1337ffdcd70bc2afa8058b642361355307'
+             '410169419f87baa5444882b170c07c3f206325218e94e6dadbbe2bea56f5ec94')
+
 # vim:set sw=2 sts=-1 et:
