diff --git a/PKGBUILD b/PKGBUILD
index fe00ace..ff21074 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,19 +16,14 @@ arch=(x86_64)
 url='https://www.open-mpi.org'
 license=('BSD-3-Clause AND LicenseRef-MPICH')
 makedepends=(
-  cuda
-  nvidia-utils  # for libcuda.so
   gcc-fortran
   gcc-libs
   glibc
-  hip-runtime-amd
   hwloc
   libevent
   libfabric
   libnl
   openpmix
-  openucc
-  openucx
   prrte
   valgrind
   zlib
@@ -70,13 +65,6 @@ build() {
     --with-pmix=external
     --with-prrte=external
     --with-valgrind
-    --with-ucc=/usr
-    --with-ucx=/usr
-    --with-cuda=/opt/cuda
-    # this tricks the configure script to look for /usr/lib/pkgconfig/cuda.pc
-    # instead of /opt/cuda/lib/pkgconfig/cuda.pc
-    --with-cuda-libdir=/usr/lib
-    --with-rocm=/opt/rocm
     # all components that link to libraries provided by optdepends must be run-time loadable
     --enable-mca-dso=accelerator_cuda,accelerator_rocm,btl_smcuda,rcache_gpusm,rcache_rgpusm,coll_ucc,scoll_ucc
     # mpirun should not warn on MCA component load failures by default - usually caused by missing optdepends, which is ok
@@ -110,16 +98,12 @@ package_openmpi() {
     libfabric
     libnl
     openpmix libpmix.so
-    openucx
     prrte
     zlib
   )
   optdepends=(
-    'cuda: cuda support'
-    'hip-runtime-amd: ROCm support'
     'gcc-fortran: fortran support'
     'openssh: for execution on remote hosts via plm_ssh_agent'
-    'openucc: for UCC accelerated collectives'
   )
   provides=(
     libmpi.so
