diff --git a/PKGBUILD b/PKGBUILD
index 71ea10a..58cdb6b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -71,12 +71,7 @@ build() {
     --with-pmix=external
     --with-prrte=external
     --with-valgrind
-    --with-ucc=/usr
     --with-ucx=/usr
-    --with-cuda=/opt/cuda
-    # this tricks the configure script to look for /usr/lib/pkgconfig/cuda.pc
-    # instead of /opt/cuda/lib/pkgconfig/cuda.pc
-    --with-cuda-libdir=/usr/lib
     --with-rocm=/opt/rocm
     # all components that link to libraries provided by optdepends must be run-time loadable
     --enable-mca-dso=accelerator_cuda,accelerator_rocm,btl_smcuda,rcache_gpusm,rcache_rgpusm,coll_ucc,scoll_ucc
@@ -115,11 +110,9 @@ package_openmpi() {
     zlib
   )
   optdepends=(
-    'cuda: cuda support'
     'hip-runtime-amd: ROCm support'
     'gcc-fortran: fortran support'
     'openssh: for execution on remote hosts via plm_ssh_agent'
-    'openucc: for UCC accelerated collectives'
   )
   provides=(
     libmpi.so
@@ -146,4 +139,6 @@ package_openmpi-docs() {
   install -vDm 644 $pkgbase-$pkgver/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
 
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|nvidia-utils|openucc)$'))
+
 # vim: ts=2 sw=2 et:
