diff --git a/PKGBUILD b/PKGBUILD
index 5ed54a7..3e55962 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -40,3 +40,11 @@ package() {
 
   DESTDIR="${pkgdir}" ninja -C build install
 }
+
+prepare() {
+  cd android-tools-$_tag
+  patch -d vendor -p1 -i "${srcdir}/boringssl-add-loong64-support.patch"
+}
+
+source+=("boringssl-add-loong64-support.patch")
+sha256sums+=('b37c782b1aced1d671da08be743a8fd870672238169395cd68a514860ea33c60')
