--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,6 +25,8 @@ prepare() {
 
 	# Keep adb's LOG/VLOG macros from being overwritten by protobuf 36's Abseil headers.
 	patch -Np1 -i ../protobuf-36-absl-log-macros.patch
+	# loong64: add loong64 support to the vendored boringssl
+	patch -Np 1 -d "vendor" < ../boringssl-add-loong64-support.patch
 }
 
 build() {
@@ -48,3 +50,6 @@ package() {
 	install -Dm644 vendor/fmtlib/LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/MIT.txt"
 	install -Dm644 vendor/avb/LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/BSD-3-Clause.txt"
 }
+
+source+=("boringssl-add-loong64-support.patch")
+sha256sums+=('b37c782b1aced1d671da08be743a8fd870672238169395cd68a514860ea33c60')
