diff --git a/PKGBUILD b/PKGBUILD
index 7d139da..d0995c8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -39,6 +39,10 @@ sha512sums=('ecbeee1d495a6a733585cecf512161a9f861e784578e2713ad10b55f69968dd2bde
 build() {
   cd binaryen
 
+  export CC=clang CXX=clang++
+  # clang does not support it
+  CXXFLAGS=($(printf "%s\n" "${CXXFLAGS[@]}" | grep -Ev '^(-fstack-clash-protection|-fno-plt)$'))
+  CFLAGS=($(printf "%s\n" "${CFLAGS[@]}" | grep -Ev '^(-fstack-clash-protection|-fno-plt)$'))
   export CXXFLAGS+=" -Wno-error=array-bounds"
   cmake \
       -Bbuild \
@@ -114,3 +118,5 @@ package() {
   install -Dm644 "$srcdir"/emscripten.sh "$pkgdir"/etc/profile.d/emscripten.sh
   install -Dm644 LICENSE "$pkgdir"/usr/share/licenses/$pkgname/LICENSE
 }
+
+makedepends+=(clang lld)
