diff --git a/PKGBUILD b/PKGBUILD
index 2f003e4..2361490 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -56,4 +56,11 @@ package() {
   install -m755 -d "${pkgdir}/etc/ananicy.d"
 }
 
+prepare() {
+  # Fix missing header `vmlinux.h`
+  install -Dm644 /usr/lib/modules/*.*-arch*-*/build/vmlinux.h -t "${srcdir}/${pkgname}/libananicycpp_bpf/loongarch"
+}
+
+makedepends+=(linux-headers)
+
 # vim:set sw=2 sts=2 et:
