--- a/PKGBUILD
+++ b/PKGBUILD
@@ -65,6 +65,9 @@ prepare() {
   patch -p1 -i ../lua-root.patch
 # Fix build with GCC 16
   patch -p1 -i ../b50e042b.patch
+# loong64-specific patches
+  patch -p1 -i $srcdir/texlive-bin-loong64.patch
+  patch -p1 -d libs/luajit/LuaJIT-src -i $srcdir/texlive-luajit-loong.patch
 }
 
 build() {
@@ -198,3 +201,8 @@ package_texlive-bin() {
   rm "$pkgdir"/usr/lib/libsynctex.*
   rm "$pkgdir"/usr/lib/pkgconfig/synctex.pc
 }
+
+source+=(texlive-bin-loong64.patch
+         texlive-luajit-loong.patch)
+sha256sums+=('3b91a0c022fc3195b58f9c03c4fcd81562539f57c19a0b3638720b63ab37329d'
+             '5508b6d15cc956c30e6eb4e48647196d402110b291147742ba2d1bc437a414be')
