diff --git a/PKGBUILD b/PKGBUILD
index ba83315..9504942 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,8 +46,10 @@ makedepends=(bash
 url='https://tug.org/texlive/'
 _commit=cf0f89c44f5d409c6d1bc0592a808af278611428 # tags/texlive-2024.2
 source=(git+https://github.com/Tex-Live/texlive-source.git#commit=$_commit
+        texlive-bin-loong64.patch
         ptex-debug-print.patch)
 sha256sums=('bf83f4db38318468e86795c7416e972a010f46d1dcd22cc20064141dae5a952f'
+            '0c09915ccb8bad1792f11a69cc7057d775cdf54a443427d9ca77f5602432f1aa' 
             'aa838f09003c62c2efb5770a8de66f99b409df049fbd65098d80fd1957d06c50')
 
 prepare() {
@@ -60,6 +62,7 @@ prepare() {
   sed -i s/SELFAUTOPARENT/TEXMFROOT/ texk/tex4htk/t4ht.c
 # remove spurious ptex "guessed encoding" print
   patch -p1 -i ../ptex-debug-print.patch
+  patch -p1 -i $srcdir/texlive-bin-loong64.patch
 }
 
 build() {
