diff --git a/PKGBUILD b/PKGBUILD
index a6e86e3..3788d1c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -60,6 +60,9 @@ prepare() {
   sed -i s/SELFAUTOPARENT/TEXMFROOT/ texk/tex4htk/t4ht.c
 # remove spurious ptex "guessed encoding" print
   patch -p1 -i ../ptex-debug-print.patch
+# loong64-specific patches
+  patch -p1 -i $srcdir/texlive-bin-loong64.patch
+  patch -p1 -d libs/luajit/LuaJIT-src -i $srcdir/texlive-luajit-loong.patch
 }
 
 build() {
@@ -194,3 +197,8 @@ package_texlive-bin() {
   rm "$pkgdir"/usr/lib/libsynctex.*
   rm "$pkgdir"/usr/lib/pkgconfig/synctex.pc
 }
+
+source+=(texlive-bin-loong64.patch
+         texlive-luajit-loong.patch)
+sha256sums+=('3b91a0c022fc3195b58f9c03c4fcd81562539f57c19a0b3638720b63ab37329d'
+             '5508b6d15cc956c30e6eb4e48647196d402110b291147742ba2d1bc437a414be')
