diff --git a/libs/luajit/configure b/libs/luajit/configure
index 7d798f8..ea7d3c1 100755
--- a/libs/luajit/configure
+++ b/libs/luajit/configure
@@ -15094,6 +15094,11 @@ then :
   echo '-D__AARCH64EB__=1' >>native_flags
 fi
 
+elif grep 'LJ_TARGET_LOONGARCH64 ' conftest.i >/dev/null 2>&1
+then :
+  LJARCH=loongarch64
+  echo '-D__loongarch__=1' >>native_flags
+
 elif grep 'LJ_TARGET_PPC ' conftest.i >/dev/null 2>&1
 then :
   LJARCH=ppc
diff --git a/libs/luajit/m4/lj-system.m4 b/libs/luajit/m4/lj-system.m4
index 73ba282..7005664 100644
--- a/libs/luajit/m4/lj-system.m4
+++ b/libs/luajit/m4/lj-system.m4
@@ -29,6 +29,9 @@ AS_IF([grep 'LJ_TARGET_X64 ' conftest.i >/dev/null 2>&1],
          AS_IF([grep '__AARCH64EB__' conftest.i >/dev/null 2>&1],
                  [echo '-D__AARCH64EB__=1' >>native_flags])
         ],
+      [grep 'LJ_TARGET_LOONGARCH64 ' conftest.i >/dev/null 2>&1],
+        [LJARCH=loongarch64
+        ],
       [grep 'LJ_TARGET_PPC ' conftest.i >/dev/null 2>&1],
         [LJARCH=ppc
          AS_IF([grep 'LJ_LE 1' conftest.i >/dev/null 2>&1],
