--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,11 +27,15 @@ sha256sums=('7f7650d1d3f9d1a136c966ca532d5e5fd7b08f704e8a0b2e3d86442f3c0be1e2')
 
 prepare() {
   cd "$pkgbase"
+  patch -Np1 -i ../update_cargo_lock.patch
   local tripple="$(rustc -vV | sed -n 's/host: //p')"
   cargo fetch --locked --target "$tripple"
   mkdir completions
 }
 
+source+=(update_cargo_lock.patch)
+sha256sums+=('6272ab1ebe13a5ada3b409f6d9d2644d87d93d517f6bd4216710e15da76288ed')
+
 # Note --frozen doesn't work here because cargo fetch didn't get everything
 # maturin ends up trying to use so we make do with --locked ...
 build() {
