diff --git a/PKGBUILD b/PKGBUILD
index 4e57a0c..8f56cb5 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,6 +25,7 @@ _tensile_dir="$(basename "$_tensile")-$(basename "${source[1]}" ".tar.gz")"
 
 
 prepare() {
+  export LDFLAGS="${LDFLAGS} -fuse-ld=mold"
   cd "$_dirname"
   # https://github.com/ROCm/Tensile/issues/2044
   patch -Np1 -i ../remove-mf16c-flag-as-f16-intrinsics.patch
@@ -60,3 +61,5 @@ package() {
 
   install -Dm644 "$_dirname/LICENSE.md" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+makedepends+=(mold)
