--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,7 +14,7 @@ source=("https://downloads.sourceforge.net/project/libsquish/libsquish-$pkgver.t
 b2sums=('d2cdf274baf9cf8890ee4c5c434448a34bc6d3d8967df6e2e9334fe1eff66ce5371597396c564c80a128709a8849f1f622d90aaf470eacc1ad67811cef38bd60')
 
 prepare() {
-  for x in USE_OPENMP=0 USE_SHARED=1 USE_SSE=1; do
+  for x in USE_OPENMP=0 USE_SHARED=1 USE_SSE=0; do
     setconf config $x
   done
   setconf libsquish.pc.in Version "$pkgver"
@@ -27,6 +27,7 @@ build() {
     -D CMAKE_INSTALL_PREFIX=/usr \
     -D CMAKE_POLICY_VERSION_MINIMUM=3.5 \
     -D BUILD_SHARED_LIBS=ON \
+    -D BUILD_SQUISH_WITH_SSE2=OFF \
     -S .
   make -C build
 }
