--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,6 +26,7 @@ validpgpkeys=(
 
 prepare() {
   cd ${pkgbase}-${pkgver}
+  patch -Np1 -i ../libseccomp-loong64-support.patch
   autoreconf -fiv
 }
 
@@ -56,3 +57,6 @@ package_python-libseccomp() {
 }
 
 # vim: ts=2 sw=2 et:
+source+=('libseccomp-loong64-support.patch')
+sha256sums+=('ca7fadad1c3fc167904ca44ae4cc417dc186e7703f7fd8d6221355cb604c4386')
+b2sums+=('c6654141850444e43c3c19961a7192fa04a0264568359fddff65ab4839d6c8f0ecd83c08b113eb6aab0892d7b0b764112e35dcaca94db35af15f8fecabe1125b')
