--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,13 +6,12 @@
 pkgbase=libseccomp
 pkgname=(libseccomp python-libseccomp)
 pkgver=2.5.5
-pkgrel=3
+pkgrel=3.1
 pkgdesc='Enhanced seccomp library'
 arch=('x86_64')
 license=('LGPL2.1')
 url="https://github.com/seccomp/libseccomp"
 depends=('glibc')
-checkdepends=('valgrind')
 makedepends=('gperf' 'cython' 'python-setuptools')
 source=(https://github.com/seccomp/${pkgname}/releases/download/v${pkgver}/${pkgname}-${pkgver}.tar.gz{,.asc})
 sha256sums=('248a2c8a4d9b9858aa6baf52712c34afefcf9c9e94b76dce02c1c9aa25fb3375'
@@ -26,6 +25,7 @@ validpgpkeys=(
 
 prepare() {
   cd ${pkgbase}-${pkgver}
+  patch -Np1 -i ../libseccomp-loong64-support.patch
   autoreconf -fiv
 }
 
@@ -56,3 +56,6 @@ package_python-libseccomp() {
 }
 
 # vim: ts=2 sw=2 et:
+source+=('libseccomp-loong64-support.patch')
+sha256sums+=('ca7fadad1c3fc167904ca44ae4cc417dc186e7703f7fd8d6221355cb604c4386')
+b2sums+=('c6654141850444e43c3c19961a7192fa04a0264568359fddff65ab4839d6c8f0ecd83c08b113eb6aab0892d7b0b764112e35dcaca94db35af15f8fecabe1125b')
