diff --git a/PKGBUILD b/PKGBUILD
index f660a69..dbe0f7d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -30,6 +30,9 @@ sha256sums=('f67c16fe40d078be6bf40006b086068951ab885ad815dfe8fa96c0a546aac57f'
 prepare() {
   mkdir -p src/github.com/opencontainers
   cp -r runc-${pkgver} src/github.com/opencontainers/runc
+  cd "src/github.com/opencontainers/runc"
+  go mod edit -replace=github.com/seccomp/libseccomp-golang=github.com/seccomp/libseccomp-golang@v0.11.1
+  go mod tidy
 }
 
 build() {
