diff --git a/PKGBUILD b/PKGBUILD
index 812890d..3eb252e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,7 +22,6 @@ pkgname=(
   qemu-ui-{curses,dbus,egl-headless,gtk,opengl,sdl,spice-{app,core}}
   qemu-user{,-static}{,-binfmt}
   qemu-vhost-user-gpu
-  qemu-vmsr-helper
   qemu-{base,desktop,emulators-full,full}
 )
 pkgver=9.2.3
@@ -219,7 +218,6 @@ _qemu_desktop_optdepends=(
   'qemu-tests: for QEMU tests'
   'qemu-tools: for QEMU tools'
   'qemu-user: for user mode emulation of QEMU targets'
-  'qemu-vmsr-helper: for accessing the Running Average Power Limit (RAPL) MSR'
 )
 
 _qemu_base_optdepends=(
@@ -609,8 +607,6 @@ package_qemu-common() {
 
     _pick qemu-vhost-user-gpu usr/lib/qemu/vhost-user-gpu
     _pick qemu-vhost-user-gpu usr/share/qemu/vhost-user/50-qemu-gpu.json
-
-    _pick qemu-vmsr-helper usr/bin/qemu-vmsr-helper
   )
 }
 
@@ -863,7 +859,7 @@ package_qemu-system-hppa-firmware() {
 
 package_qemu-system-loongarch64() {
   pkgdesc="QEMU system emulator for LoongArch64"
-  depends=("${_qemu_system_deps[@]}" dtc systemd-libs)
+  depends=("${_qemu_system_deps[@]}" dtc systemd-libs edk2-loongarch64)
   mv -v $pkgname/* "$pkgdir"
   _install_licenses
 }
@@ -1178,7 +1174,7 @@ package_qemu-base() {
   depends=(
     qemu-common=$pkgver-$pkgrel
     qemu-img=$pkgver-$pkgrel
-    qemu-system-x86=$pkgver-$pkgrel
+    qemu-system-loongarch64=$pkgver-$pkgrel
     virtiofsd
   )
   optdepends=("${_qemu_base_optdepends[@]}")
@@ -1227,11 +1223,12 @@ package_qemu-full() {
     qemu-tests=$pkgver-$pkgrel
     qemu-tools=$pkgver-$pkgrel
     qemu-user=$pkgver-$pkgrel
-    qemu-vmsr-helper=$pkgver-$pkgrel
   )
   optdepends=("${_qemu_full_optdepends[@]}")
   provides=(qemu=$pkgver)
   _install_licenses
 }
 
+_qemu_desktop_optdepends+=("qemu-system-x86: for x86 system emulator")
+
 # vim:set ts=2 sw=2 et:
