--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,7 +10,7 @@ pkgname=(
   qemu-chardev-{baum,spice}
   qemu-docs
   qemu-guest-agent
-  qemu-hw-display-{qxl,virtio-{gpu{,-{gl,rutabaga,pci,pci-gl,pci-rutabaga}},vga{,-gl,-rutabaga}}}
+  qemu-hw-display-{qxl,virtio-{gpu{,-{gl,pci,pci-gl}},vga{,-gl}}}
   qemu-hw-s390x-virtio-gpu-ccw
   qemu-hw-uefi-vars
   qemu-hw-usb-{host,redirect,smartcard}
@@ -112,7 +112,6 @@ makedepends=(
   python-sphinx
   python-sphinx_rtd_theme
   rdma-core
-  rutabaga-ffi
   sdl2
   sdl2_image
   snappy
@@ -243,13 +242,10 @@ _qemu_base_optdepends=(
   'qemu-hw-display-qxl: for the QXL display device'
   'qemu-hw-display-virtio-vga: for the virtio-vga display device'
   'qemu-hw-display-virtio-vga-gl: for the virtio-vga-gl display device'
-  'qemu-hw-display-virtio-vga-rutabaga: for the virtio-vga-rutabaga display device'
   'qemu-hw-display-virtio-gpu: for the virtio-gpu display device'
   'qemu-hw-display-virtio-gpu-gl: for the virtio-gpu-gl display device'
-  'qemu-hw-display-virtio-gpu-rutabaga: for the virtio-gpu-rutabaga display device'
   'qemu-hw-display-virtio-gpu-pci: for the virtio-gpu-pci display device'
   'qemu-hw-display-virtio-gpu-pci-gl: for the virtio-gpu-pci-gl display device'
-  'qemu-hw-display-virtio-gpu-pci-rutabaga: for the virtio-gpu-pci-rutabaga display device'
   'qemu-hw-uefi-vars: for UEFI variable support'
   'qemu-hw-usb-host: for host USB support'
   'qemu-hw-usb-redirect: for USB redirect support'
@@ -323,7 +319,6 @@ build() {
     --enable-sdl
     --enable-slirp
     --enable-tpm
-    --enable-rutabaga-gfx
     --smbd=/usr/bin/smbd
     --with-coroutine=ucontext
   )
@@ -488,13 +483,10 @@ package_qemu-common() {
     _pick qemu-hw-display-qxl usr/lib/qemu/hw-display-qxl.so
     _pick qemu-hw-display-virtio-gpu usr/lib/qemu/hw-display-virtio-gpu.so
     _pick qemu-hw-display-virtio-gpu-gl usr/lib/qemu/hw-display-virtio-gpu-gl.so
-    _pick qemu-hw-display-virtio-gpu-rutabaga usr/lib/qemu/hw-display-virtio-gpu-rutabaga.so
     _pick qemu-hw-display-virtio-gpu-pci usr/lib/qemu/hw-display-virtio-gpu-pci.so
     _pick qemu-hw-display-virtio-gpu-pci-gl usr/lib/qemu/hw-display-virtio-gpu-pci-gl.so
-    _pick qemu-hw-display-virtio-gpu-pci-rutabaga usr/lib/qemu/hw-display-virtio-gpu-pci-rutabaga.so
     _pick qemu-hw-display-virtio-vga usr/lib/qemu/hw-display-virtio-vga.so
     _pick qemu-hw-display-virtio-vga-gl usr/lib/qemu/hw-display-virtio-vga-gl.so
-    _pick qemu-hw-display-virtio-vga-rutabaga usr/lib/qemu/hw-display-virtio-vga-rutabaga.so
 
     _pick qemu-hw-usb-host usr/lib/qemu/hw-usb-host.so
     _pick qemu-hw-usb-redirect usr/lib/qemu/hw-usb-redirect.so
@@ -898,7 +890,7 @@ package_qemu-system-hppa-firmware() {
 
 package_qemu-system-loongarch64() {
   pkgdesc="QEMU system emulator for LoongArch64"
-  depends=("${_qemu_system_deps[@]}" dtc systemd-libs)
+  depends=("${_qemu_system_deps[@]}" dtc systemd-libs edk2-loongarch64)
   mv -v $pkgname/* "$pkgdir"
   _install_licenses
 }
@@ -1222,7 +1214,7 @@ package_qemu-base() {
   depends=(
     qemu-common=$pkgver-$pkgrel
     qemu-img=$pkgver-$pkgrel
-    qemu-system-x86=$pkgver-$pkgrel
+    qemu-system-loongarch64=$pkgver-$pkgrel
     virtiofsd
   )
   optdepends=("${_qemu_base_optdepends[@]}")
@@ -1238,8 +1230,8 @@ package_qemu-desktop() {
     qemu-audio-{alsa,dbus,jack,oss,pa,pipewire,sdl,spice}=$pkgver-$pkgrel
     qemu-block-{curl,dmg,nfs,ssh}=$pkgver-$pkgrel
     qemu-chardev-spice=$pkgver-$pkgrel
-    qemu-hw-display-{qxl,virtio-gpu{,-{gl,rutabaga,pci,pci-gl,pci-rutabaga}}}=$pkgver-$pkgrel
-    qemu-hw-display-virtio-vga{,-gl,-rutabaga}=$pkgver-$pkgrel
+    qemu-hw-display-{qxl,virtio-gpu{,-{gl,pci,pci-gl}}}=$pkgver-$pkgrel
+    qemu-hw-display-virtio-vga{,-gl}=$pkgver-$pkgrel
     qemu-hw-uefi-vars=$pkgver-$pkgrel
     qemu-hw-usb-{host,redirect,smartcard}=$pkgver-$pkgrel
     qemu-ui-{curses,dbus,egl-headless,gtk,opengl,sdl,spice-{app,core}}=$pkgver-$pkgrel
@@ -1281,4 +1273,6 @@ package_qemu-full() {
   _install_licenses
 }
 
+_qemu_desktop_optdepends+=("qemu-system-x86: for x86 system emulator")
+pkgname=(${pkgname[@]/qemu-vmsr-helper})
 # vim:set ts=2 sw=2 et:
