--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,7 +37,7 @@ build() {
   python -m build --wheel --no-isolation
 # Compile attach libraries
   cd build/lib*/debugpy/_vendored/pydevd/pydevd_attach_to_process
-  g++ ${CXXFLAGS} -m64 -shared -o attach_linux_amd64.so -fPIC -nostartfiles linux_and_mac/attach.cpp ${LDFLAGS}
+  g++ ${CXXFLAGS} -shared -o attach_linux_amd64.so -fPIC -nostartfiles linux_and_mac/attach.cpp ${LDFLAGS}
 }
 
 package() {
