diff --git a/PKGBUILD b/PKGBUILD
index 7d999ae..018f762 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -50,8 +50,9 @@ build() {
     -e 's/-fcf-protection ?//'
     -e 's/-fexceptions ?//'
     -e 's/-fstack-clash-protection ?//'
-    -e 's/-march=x86-64 ?//'
-    -e 's/-mtune=generic ?//'
+    -e 's/-march=loongarch64 ?//'
+    -e 's/-mabi=lp64d ?//'
+    -e 's/-mcmodel=medium ?//'
   )
   CFLAGS="$(<<<"$CFLAGS" sed "${sanitize_flags[@]}")"
   CXXFLAGS="$(<<<"$CXXFLAGS" sed "${sanitize_flags[@]}")"
