diff --git a/src/libs/utils/hostosinfo.cpp b/src/libs/utils/hostosinfo.cpp
index 64d4244c38c..0d5ff958d56 100644
--- a/src/libs/utils/hostosinfo.cpp
+++ b/src/libs/utils/hostosinfo.cpp
@@ -113,6 +113,8 @@ OsArch binaryArchitecture()
     return OsArchArm;
 #elif defined(Q_PROCESSOR_IA64)
     return OsArchItanium;
+#elif defined(Q_PROCESSOR_LOONGARCH_64)
+    return OsArchLoongArch64;
 #else
     static_assert(false, "Unknown architecture, please add detection.");
     return OsArchUnknown;
diff --git a/src/libs/utils/osspecificaspects.h b/src/libs/utils/osspecificaspects.h
index 56b985afd81..64c0fb7216d 100644
--- a/src/libs/utils/osspecificaspects.h
+++ b/src/libs/utils/osspecificaspects.h
@@ -15,7 +15,7 @@ namespace Utils {
 // Add more as needed.
 enum OsType { OsTypeWindows, OsTypeLinux, OsTypeMac, OsTypeOtherUnix, OsTypeOther };
 
-enum OsArch { OsArchUnknown, OsArchX86, OsArchAMD64, OsArchItanium, OsArchArm, OsArchArm64 };
+enum OsArch { OsArchUnknown, OsArchX86, OsArchAMD64, OsArchItanium, OsArchArm, OsArchArm64, OsArchLoongArch64 };
 
 QTCREATOR_UTILS_EXPORT QString osTypeToString(OsType osType);
 QTCREATOR_UTILS_EXPORT Utils::Result<OsType> osTypeFromString(const QString &string);
diff --git a/src/plugins/extensionmanager/extensionsmodel.cpp b/src/plugins/extensionmanager/extensionsmodel.cpp
index fd650fc952b..37488d710dd 100644
--- a/src/plugins/extensionmanager/extensionsmodel.cpp
+++ b/src/plugins/extensionmanager/extensionsmodel.cpp
@@ -391,6 +391,8 @@ QString customOsArchToString(OsArch osArch)
         return "arm";
     case OsArchArm64:
         return "arm64";
+    case OsArchLoongArch64:
+        return "loongarch64";
     case OsArchUnknown:
         break;
     }
