diff --git a/PKGBUILD b/PKGBUILD
index d010578..0fa6307 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -67,6 +67,8 @@ prepare() {
   git submodule set-url src/libs/qlitehtml "$srcdir"/qlitehtml
   git -c protocol.file.allow=always submodule update --init src/libs/qlitehtml src/tools/perfparser
 
+# Fix build with loong64
+  patch -Np1 -i "${srcdir}/add-loongarch64-support.patch"
 # Fix doc build with system litehtml
   rm src/libs/qlitehtml/src/3rdparty/qt_attribution.json
 # Fix build with Qt 6.10
@@ -101,3 +103,6 @@ package_qtcreator-devel() {
 
   DESTDIR="$pkgdir" cmake --install build --component Devel
 }
+
+source+=("add-loongarch64-support.patch")
+sha256sums+=('2c4f8f251a85259b33c8ed8018c5a64d8353bef0700d1673239ad018c1bcfd17')
