diff --git a/PKGBUILD b/PKGBUILD
index 2e7621a..2d95a2c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,6 +14,7 @@ sha512sums=('d90f9638f6165d92867ef74b3dd95269c6f9a3cec9ddbd4018104caf5664ca9ce06
 
 prepare() {
   cd "$pkgname-$pkgver"
+  patch -p1 -i "${srcdir}/update-cargo.patch"
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
@@ -35,3 +36,5 @@ package() {
 }
 
 # vim: ts=2 sw=2 et:
+source+=("update-cargo.patch::https://github.com/rgwood/systemctl-tui/pull/27.patch")
+sha512sums+=('847c98c327cfe34f0dc8aa7c7dbfc231e49d5bcc3a56743549b1f116e973a801f7bc71b70899b95870685beaa6d69d7eaea697c926a3cdf0a1c264bcaa62f015')
