--- a/PKGBUILD
+++ b/PKGBUILD
@@ -43,11 +43,18 @@ prepare() {
   patch -Np1 -i ../db-5.3.28-mmap-high-cpu-usage.patch
   # gcc fix
   patch -Np1 -i ../db-5.3.28-atomic_compare_exchange.patch
+  patch -Np1 -i ../add_cstddef.patch
+  for c_s in $(find -type f -name config.sub -o -name configure.sub); do cp -f /usr/share/automake-1.1?/config.sub "$c_s"; done
+  for c_g in $(find -type f -name config.guess -o -name configure.guess); do cp -f /usr/share/automake-1.1?/config.guess "$c_g"; done
 }
 
+source+=('add_cstddef.patch')
+b2sums+=('3246b3155a251a9a4fce43a9b75be7e8120673a15c7edebe60ec0d942daa1a9f0adff04e324232385a08b351300c61eefbf71253f801577d905f27e8b9ae320b')
 
 build() {
   cd db-${pkgver}/build_unix
+  CFLAGS+=" -Wno-error=implicit-int -Wno-error=implicit-function-declaration"
+  BUILD_CFLAGS+=" -Wno-error=implicit-int -Wno-error=implicit-function-declaration"
   ../dist/configure \
     --prefix=/usr \
     --bindir=/usr/bin/db5.3 \
